<!DOCTYPE html>
<html>
  <head>
    <title>vatlab</title>
    
      <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
<meta name="generator" content="Hugo 0.36.1" />
<title>use :: vatlab</title>
<link rel="shortcut icon" href="https://vatlab.github.io/vat-docs/images/favicon.png" type="image/x-icon" />
<link href="https://vatlab.github.io/vat-docs/css/font-awesome.min.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/css/nucleus.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/theme-flex/style.css" rel="stylesheet">

<link rel="stylesheet" href="https://vatlab.github.io/vat-docs/css/bootstrap.min.css">
<script src="https://vatlab.github.io/vat-docs/js/jquery-2.x.min.js"></script>
<script type="text/javascript">
      var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
</script>
<meta name="description" content="">


    
  </head>
  <body data-url="/documentation/vtools-commands/use/">
    
      <header>
  <div class="logo">
    
	
  
    <a class="baselink" href="https://vatlab.github.io/vat-docs/">vatlab</a>
  

  </div>
  <div class="burger"><a href="javascript:void(0);" style="font-size:15px;">&#9776;</a></div>
</header>
<article>
  <aside>
    <ul class="menu">
          <li data-nav-id="/" class="dd-item">
          <a href="https://vatlab.github.io/vat-docs/">
            <i class="fa fa-fw fa-home"></i>
          </a>
          </li>
    <li data-nav-id="/install/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/install/">Installation</a>
      </div>
    </li>
    <li data-nav-id="/documentation/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/">Documentation</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/keyconcepts/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/">
            Key concepts
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/tutorials/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/tutorials/">
            Tutorials
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/help/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/help/">
            Help
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/vtools-commands/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/">Vtools commands</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools-commands/init/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/init/">
            Init
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/import/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/import/">
            Import
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/update/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/update/">
            Update
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/phenotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/phenotype/">
            Phenotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/liftover/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/liftover/">
            lifttover
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/show/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/show/">
            show
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/use/" class="dd-item active">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/use/">
            use
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/select/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/select/">
            select
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/exclude/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/exclude/">
            exclude
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/compare/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/compare/">
            compare
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/associate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/associate/">
            associate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/output/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/output/">
            output
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/export/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/export/">
            export
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/remove/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/remove/">
            remove
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/execute/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/execute/">
            execute
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/admin/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/admin/">
            admin
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/functions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/functions/">Functions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/functions/ref_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/ref_sequence/">
            ref_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/mut_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/mut_sequence/">
            mut_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/track/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/track/">
            track
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/genotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/genotype/">
            Genotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/samples/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/samples/">
            samples
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/in_table/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/in_table/">
            in_table
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/otherfunctions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/otherfunctions/">
            Other functions
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/vtools_report/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/">Vtools_report</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools_report/avg_depth/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/avg_depth/">
            avg_depth
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/discordance_rate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/discordance_rate/">
            discordance_rate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sam_info/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sam_info/">
            sam_info
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sequence/">
            sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/trans_ratio/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/trans_ratio/">
            trans_ratio
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/variant_stat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/variant_stat/">
            variant_stat
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/inbreeding_coefficient/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/inbreeding_coefficient/">
            inbreeding_coefficient
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/meta_analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/meta_analysis/">
            meta_analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_fields/">
            plot_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_geno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_geno_fields/">
            plot_geno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_pheno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_pheno_fields/">
            plot_pheno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_association/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_association/">
            plot_association
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/customization/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/">Customization</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/format/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/">
            Format
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/">Simulation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/custimzedsimu/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/custimzedsimu/">
            Customized simulations
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/models/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/">Models</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/models/peng1/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/peng1/">
            Peng2014_ex1
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/models/pen2/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/pen2/">
            Peng2014_ex2
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/">Pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/customizedpipeline/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/customizedpipeline/">
            Customized pipeline
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/pipelines/variant-calling/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/">Variant calling</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_hg19/">
            bwa_gatk33_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_b37/">
            bat_gatk33_b37
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_hg19/">
            bwa_gatk28_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_b37/">
            bwa_gatk28_b37
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/other-pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/">Other pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/other-pipelines/import_vcf/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/import_vcf/">
            import_vcf
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/anno_utils/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/anno_utils/">
            anno_utils
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/illumina/">
            Illumnina
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/annovar/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/annovar/">
            ANNOVAR
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/snpeff/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/snpeff/">
            snpEff
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/king/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/king/">
            KING
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/utilities/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/utilities/">Utilities</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/utilities/report_progress/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/report_progress/">
            report_progress
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/utilities/_md5sumd/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/_md5sumd/">
            md5sumd
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/">Application</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
    <li data-nav-id="/applications/annotation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/">Annotation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/tutorial/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/tutorial/">
            Tutorial
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/customized/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">
            Customized annotation databases
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/annotation/genes/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/">Genes</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/genes/ccdsgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/ccdsgene/">
            ccdsGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/hugogene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/hugogene/">
            Hugo gene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/entrezgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/entrezgene/">
            EntrezGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/knowngene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/knowngene/">
            knownGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/refgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/refgene/">
            refGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/cancergenecensus/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/cancergenecensus/">
            CancerGeneCensus
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/variants/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/">Variants</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/variants/cosmic/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/cosmic/">
            COSMIC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbnsfp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbnsfp/">
            sbNSFP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbsnp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbsnp/">
            dbSNP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/linkagephysicalmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/linkagephysicalmap/">
            linkage-physical map
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/databaseof/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/databaseof/">
            Database of Genomic Variants
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/exomesequencingpro/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/exomesequencingpro/">
            Exome Sequencing Project
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/gwas/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/gwas/">
            gwasCatalog
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/hapmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/hapmap/">
            HapMap
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/thousand/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/thousand/">
            Thousand Genomes
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/regions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/">Regions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/regions/low-complexity-regions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/low-complexity-regions/">
            Low complexity regions
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/phast-cons/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/phast-cons/">
            Phast cons
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/genomic-super-dups/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/genomic-super-dups/">
            Genomic super dups
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/cytoband/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/cytoband/">
            cytoBand
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/misc/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/">Misc</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/misc/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/illumina/">
            Illumina
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/keggpathway/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/keggpathway/">
            keggPathway
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/encode/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/encode/">
            ENCODE
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
      <li data-nav-id="/applications/discovery/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/discovery/">
            Discovery
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/">Association</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/data-explorationqc/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/data-explorationqc/">
            Data exploration/QC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/association-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/association-analysis/">
            Association analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/post-assoc-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/post-assoc-analysis/">
            Post-assoc analysis
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/basic-data-statistics/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/">Basic data statistics</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/basic-data-statistics/group-statwrite/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/group-statwrite/">
            Group stat&amp;Write
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-variant-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/">Single variant analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-variant-analysis/fisher-exact-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/fisher-exact-test/">
            Fisher exact test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-gene-association-methods/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/">Single gene association methods</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-gene-association-methods/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/cmc-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/cmc-test/">
            CMC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/c_alpha-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/c_alpha-test/">
            C-alpha test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/kbac-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/kbac-test/">
            KBAC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rbt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rbt-test/">
            RBT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rarecover-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rarecover-test/">
            RareCover test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/vt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/vt-test/">
            VT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/wss-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/wss-test/">
            WSS test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/asum-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/asum-test/">
            aSum test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/jointconditional-association-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/">Joint/Conditional association analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/collapsing-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/collapsing-methods/">
            Collapsing methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/aggregation-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/aggregation-methods/">
            Aggregation methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/weighted-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/weighted-methods/">
            Weighted methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/variable-thresholds/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/variable-thresholds/">
            Variable thresholds
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/skat-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/skat-test/">
            SKAT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/metaskat-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/metaskat-analysis/">
            MetaSKAT analysis
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/vat-stacking/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/vat-stacking/">VAT stacking</a>
      </div>
    </li>
    <li data-nav-id="/applications/association/create-your-test/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/">Create your test!</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/create-your-test/running-r-programs/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/running-r-programs/">
            Running R programs
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/create-your-test/programming-in-vat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/programming-in-vat/">
            Programming in VAT
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/view-association-result/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/">View association result</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/view-association-result/qq-plot-and-manhattan-plot/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/qq-plot-and-manhattan-plot/">
            QQ plot and Manhattan plot
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/development/">Development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/development/chagelog/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/chagelog/">
            ChangeLog
          </a>
        </div>
    </li>
      <li data-nav-id="/development/getinvolved/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/getinvolved/">
            Get Involved
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/administration/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/administration/">Administration</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/administration/allrecent/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/allrecent/">
            All recent changes
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/autolink/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/autolink/">
            Autolink
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/stylesheet/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/stylesheet/">
            Style sheet
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/trafficstat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/trafficstat/">
            Traffic statistics
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/under-development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/under-development/">Under development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/under-development/faq/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/under-development/faq/">
            FAQ
          </a>
        </div>
    </li>
        </ul>
    </li>




    </ul>
    <section>
    </section>
  </aside>
  <section class="page">
    
    <div class="nav-select">
      <center>Navigation : 
        <select onchange="javascript:location.href = this.value;">
          
    <option value="/install/" >
   Installation</option>
    <option value="/documentation/" >
   Documentation</option> 
      <option value="/documentation/keyconcepts/" >- Key concepts</option>
      <option value="/documentation/tutorials/" >- Tutorials</option>
      <option value="/documentation/help/" >- Help</option>
    <option value="/documentation/vtools-commands/" >
  - 
   Vtools commands</option> 
      <option value="/documentation/vtools-commands/init/" >-- Init</option>
      <option value="/documentation/vtools-commands/import/" >-- Import</option>
      <option value="/documentation/vtools-commands/update/" >-- Update</option>
      <option value="/documentation/vtools-commands/phenotype/" >-- Phenotype</option>
      <option value="/documentation/vtools-commands/liftover/" >-- lifttover</option>
      <option value="/documentation/vtools-commands/show/" >-- show</option>
      <option value="/documentation/vtools-commands/use/"  selected>-- use</option>
      <option value="/documentation/vtools-commands/select/" >-- select</option>
      <option value="/documentation/vtools-commands/exclude/" >-- exclude</option>
      <option value="/documentation/vtools-commands/compare/" >-- compare</option>
      <option value="/documentation/vtools-commands/associate/" >-- associate</option>
      <option value="/documentation/vtools-commands/output/" >-- output</option>
      <option value="/documentation/vtools-commands/export/" >-- export</option>
      <option value="/documentation/vtools-commands/remove/" >-- remove</option>
      <option value="/documentation/vtools-commands/execute/" >-- execute</option>
      <option value="/documentation/vtools-commands/admin/" >-- admin</option>
  
    <option value="/documentation/functions/" >
  - 
   Functions</option>
    <option value="/documentation/vtools_report/" >
  - 
   Vtools_report</option>
    <option value="/documentation/customization/" >
  - 
   Customization</option>
    <option value="/documentation/pipelines/" >
  - 
   Pipelines</option>
    <option value="/documentation/utilities/" >
  - 
   Utilities</option>
  
    <option value="/applications/" >
   Application</option>
    <option value="/development/" >
   Development</option>
    <option value="/administration/" >
   Administration</option>
    <option value="/under-development/" >
   Under development</option>



        </select>
      </center>
    </div>
      <div>
        <div class="searchbox">
          <input data-search-input id="search-by" type="text" placeholder="Search...">
        </div>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/lunr.min.js"></script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/auto-complete.js"></script>
        <link href="https://vatlab.github.io/vat-docs/css/auto-complete.css" rel="stylesheet">
        <script type="text/javascript">
          
              var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
          
        </script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/search.js"></script>
      </div>
    

    <h1>use</h1>
    
    
    
    

<h1 id="use-an-annotation-database">Use an annotation database</h1>

<h2 id="usage">Usage</h2>

<pre><code>% vtools use -h


usage: vtools use [-h] [--as NAME] [-l [LINKED_BY [LINKED_BY ...]]]
                  [--anno_type {variant,position,range,field}]
                  [--linked_fields [LINKED_FIELDS [LINKED_FIELDS ...]]]
                  [-f [FILES [FILES ...]]] [--rebuild] [-j N] [-v {0,1,2}]
                  source

Link an annotation database to the project, download it from the variant tools
website or build it from source if needed.

positional arguments:
  source                Use an annotation database ($source.DB or
                        $source.DB.gz) if it is available, download or build
                        the database if a description file ($source.ann) is
                        available. Otherwise, this command will download a
                        description file and the corresponding database from
                        web (c.f. runtime variable $search_path) and the
                        latest version of the datavase). If all means fail,
                        this command will try to download the source of the
                        annotation database (or use source files provided by
                        option --files).

optional arguments:
  -h, --help            show this help message and exit
  -v {0,1,2}, --verbosity {0,1,2}
                        Output error and warning (0), info (1) and debug (2)
                        information to standard output (default to 1).

Basic link options:
  --as NAME             An alternative name for the linked database. This
                        option allows the use of shorter field names (e.g.
                        tg.chr instead of thousandGenomes.chr) and the use of
                        multiple versions of the same database.
  -l [LINKED_BY [LINKED_BY ...]], --linked_by [LINKED_BY [LINKED_BY ...]]
                        A list of fields that are used to link the annotation
                        database to tables in the existing project. This
                        parameter is required only for 'field' type of
                        annotation databases that link to fields of existing
                        tables.

Advanced link options:
  --anno_type {variant,position,range,field}
                        This option overrides type of an existing annotation
                        database when it is attached to a project. It
                        corresponds to key anno_type of the data sources
                        section of an annotation file (with suffix .ann) but
                        does not affect the .ann file or the database built
                        from it.
  --linked_fields [LINKED_FIELDS [LINKED_FIELDS ...]]
                        An alternative set of fields that are used to link the
                        annotation database to the master variant table. It
                        should have four, two, and three values for database
                        of type variant, position, and range. Similar to
                        anno_type, this option does not affect the .ann file
                        or the database built from it.

Build database from source:
  -f [FILES [FILES ...]], --files [FILES [FILES ...]]
                        A list of source files. If specified, vtools will not
                        try to download and select source files. These source
                        files will be compiled into a local annotation
                        database. This is used only when no local annotation
                        database is specified.
  --rebuild             If set, variant tools will always rebuild the
                        annotation database from source, ignoring existing
                        local and online database. In addition to $name.DB,
                        variant tools will also create $name-$version.DB.gz
                        that can be readily distributed.
  -j N, --jobs N        If need to build database from source, maximum number
                        of processes to use.
</code></pre>

<h2 id="details">Details</h2>

<p>Command <code>vtools use</code> attaches an annotation database to the project, effectively making one or more attributes available to variants in the project. Four types of annotation databases can be used with variant tools:</p>

<ul>
<li>A <strong>variant</strong> annotation database is linked directly to the project by linking chromosome, position, reference and alternative allele to fields <strong><code>chr</code></strong>, <strong><code>pos</code></strong>, <strong><code>ref</code></strong>, and <strong><code>alt</code></strong> of the project. It provides variant-level annotation.</li>
<li>A <strong>position</strong> annotation database is linked to the project using fields <strong><code>chr</code></strong> and <strong><code>pos</code></strong>. It provides annotation information for nucleotide locations. If there are more than one variants at a locus, the annotation information are used for all of them.</li>
<li>A <strong>range</strong> annotation database is linked to the project by comparing fields <strong><code>chr</code></strong> and <strong><code>pos</code></strong> to chromosome, starting and ending positions of each range. It provides annotation information for all variants in chromosomal regions.</li>
<li>A <strong>field</strong> annotation database can be linked to any <strong>existing variant info or annotation fields</strong> and provide annotation information for these fields. For example, a gene annotation table might provide description for some genes, which can be linked to the project through field <code>refGene.name2</code> that is provided by database <code>refGene</code>. Because attribute annotation databases are anchored to a project through existing attributes, <strong>one or more fields must be specified to link the database to the project using parameter <code>--linked_by</code>.</strong></li>
</ul>

<h3 id="basic-usages-of-system-provided-annotation-databases">Basic usages of system-provided annotation databases</h3>

<p>It is easy to use system-provided annotation databases. Generally speaking, you should</p>

<ul>
<li>Use command <code>vtools show annotations</code> to see a list of annotation databases, and identify annotation databases that match the reference genome of your project.</li>
<li>Use command <code>vtools show annotation ANNODB</code> to check the details of an annotation database.</li>

<li><p>Use command <code>vtools use ANNODB</code> to download (if needed) and link the database to your project. <em>variant tools</em> will automatically link the project to the database using appropriate fields.</p></li>

<li><p>Position-aware annotation databases (variant, position, and range) are ref-genome dependent. Most such annotation databases are built for a particular build of reference genome, but some of them support multiple reference genomes.</p></li>

<li><p>Newer databases usually contain more updated annotation information and usually use more recent build of reference genome. If you are using <em>hg18</em> and would like to use annotation databases that use build <em>hg19</em> of the reference genome, you can liftover your project to add <em>hg19</em> as an alternative reference genome.</p></li>
</ul>

<p>(:toggleexample Examples: Use system-provided annotation databases:) Let us get a project</p>

<pre><code>% vtools init use --parent vt_simple
</code></pre>

<p>This project uses build <code>hg19</code> of the reference genome, as shown in the output of command <code>vtools show</code></p>

<pre><code>% vtools show


Upgrading to 1.0.7: 100% [==============================] 3 202.2/s in 00:00:00
Project name:                test
Primary reference genome:    hg19
Secondary reference genome:  None
Runtime options:             verbosity=1
Variant tables:              variant
Annotation databases:
</code></pre>

<p>We then use command <code>vtools show annotations</code> to check all available databases, using option <code>-v0</code> to suppress descriptions of databases:</p>

<pre><code>% vtools show annotations -v0 


CancerGeneCensus-20111215
CancerGeneCensus-20120315
CancerGeneCensus
CosmicCodingMuts-v61_260912
CosmicCodingMuts
CosmicMutantExport-v61_260912
CosmicMutantExport
CosmicNonCodingVariants-v61_260912
CosmicNonCodingVariants
ccdsGene-hg19_20110909
ccdsGene-hg19_20111206
ccdsGene
ccdsGene_exon-hg19_20110909
ccdsGene_exon-hg19_20111206
ccdsGene_exon
ccdsGene_exon_hg19-20111206
ccdsGene_hg19-20111206
cytoBand-hg18_20111216
cytoBand-hg19_20111216
cytoBand
dbNSFP-hg18_hg19_1.1_2
dbNSFP-hg18_hg19_1_3
dbNSFP-hg18_hg19_2_0b4
dbNSFP
dbNSFP_light-hg18_hg19_1.0_0
dbNSFP_light-hg18_hg19_1_3
dbNSFP_light
dbSNP-hg18_129
dbSNP-hg18_130
dbSNP-hg19_131
dbSNP-hg19_132
dbSNP-hg19_135-1
dbSNP-hg19_135
dbSNP-hg19_137
dbSNP
evs-6500
evs-hg19_20111107
evs
evs_5400
genomicSuperDups-hg19_20130626
genomicSuperDups
gwasCatalog-hg19_20111220
gwasCatalog
keggPathway-20110823
keggPathway
knownGene-hg18_20110909
knownGene-hg18_20121219
knownGene-hg19_20110909
knownGene-hg19_20121219
knownGene
knownGene_exon-hg18_20110909
knownGene_exon-hg19_20110909
knownGene_exon
phastCons-hg19_20110909
phastCons-hg19_20130322
phastCons
phastConsElements-hg19_20130622
phastConsElements
refGene-hg18_20110909
refGene-hg19_20110909
refGene
refGene_exon-hg18_20110909
refGene_exon-hg19_20110909
refGene_exon
thousandGenomes-hg19_20110909
thousandGenomes-hg19_201202
thousandGenomes
thousandGenomesEBI-hg19_phase1_release_v3_20101123
thousandGenomesEBI
</code></pre>

<p>Some databases uses <code>hg18</code>, but most uses <code>hg19</code>. The default databases (the ones without version number) generally refer to the latest databases that uses <code>hg19</code>, so you can use most databases as simple as</p>

<pre><code>% vtools use refGene


INFO: Downloading annotation database from annoDB/refGene.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/refGene-hg19_20110909.DB.gz
INFO: Using annotation DB refGene in project test.
INFO: refseq Genes




% vtools use refGene_exon


INFO: Downloading annotation database from annoDB/refGene_exon.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/refGene_exon-hg19_20110909.DB.gz
INFO: Using annotation DB refGene_exon in project test.
INFO: Exon locations of refseq Genes




% vtools use dbNSFP


INFO: Downloading annotation database from annoDB/dbNSFP.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/dbNSFP-hg18_hg19_2_0b4.DB.gz
INFO: Using annotation DB dbNSFP in project test.
INFO: dbNSFP version 2.0b4, maintained by Xiaoming Liu from UTSPH. Please cite &quot;Liu X, Jian X, and Boerwinkle E. 2011. dbNSFP: a lightweight database of human non-synonymous SNPs and their functional predictions. Human Mutation. 32:894-899&quot; if you find this database useful.




% vtools use thousandGenomes


INFO: Downloading annotation database from annoDB/thousandGenomes.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/thousandGenomes-hg19_201202.DB.gz
INFO: Using annotation DB thousandGenomes in project test.
INFO: 1000 Genomes VCF file (available from: ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606/VCF/v4.0/00-All.vcf.gz).
</code></pre>

<p>The CancerGeneCensus database is a bit difficult to use because it is a field database that annotate gene names so you need to link it to <code>refGene.name2</code>,</p>

<pre><code>% vtools use CancerGeneCensus --linked_by refGene.name2


INFO: Downloading annotation database from annoDB/CancerGeneCensus.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/CancerGeneCensus-20130711.DB.gz
INFO: Using annotation DB CancerGeneCensus in project test.
INFO: This database contains variants from the Cancer Genome Project. It is
an ongoing effort to catalogue those genes for which mutations have been causally
implicated in cancer. The original census and analysis was published in Nature
Reviews Cancer and supplemental analysis information related to the paper is also
available. Currently, more than 1% of all human genes are implicated via mutation
in cancer. Of these, approximately 90% have somatic mutations in cancer, 20% bear
germline mutations that predispose to cancer and 10% show both somatic and
germline mutations.
INFO: 475 out of 23242 refgene.name2 are annotated through annotation database CancerGeneCensus
WARNING: 9 out of 484 values in annotation database CancerGeneCensus are not linked to the project.
</code></pre>

<p>After you use these databases, you could get the details of them using command <code>vtools show annotation ANNODB</code>,</p>

<pre><code>% vtools show annotation refGene


WARNING: Resouce file annoDB/CancerGeneCensus.ann has been updated. Please update it using command &quot;vtools admin --update_resource existing&quot;.
Annotation database refGene (version hg19_20110909)
Description:            refseq Genes
Database type:          range
Reference genome hg19:  chr, txStart, txEnd
  name                  Gene name
  chr
  strand                which DNA strand contains the observed alleles
  txStart               Transcription start position
  txEnd                 Transcription end position
  cdsStart              Coding region start
  cdsEnd                Coding region end
  exonCount             Number of exons
  score                 Score
  name2                 Alternative name
  cdsStartStat          cds start stat, can be 'non', 'unk', 'incompl', and
                        'cmp1'
  cdsEndStat            cds end stat, can be 'non', 'unk', 'incompl', and
                        'cmp1'
</code></pre>

<ul>
<li>More detailed information such as statistics for each field are available if you specify option <code>-v2</code> to the <code>vtools show annotation</code> command</li>
<li>If you need to see a list of available fields, use command <code>vtools show fields</code>.</li>
</ul>

<p>(:exampleend:)</p>

<p>About multiple versions of the same database:</p>

<ul>
<li>The default database without version name always refers to the latest version. For example, <code>dbSNP</code> could be <code>dbSNP-hg19_135</code>, and then <code>dbSNP-hg19_138</code> after version 138 becomes available.</li>
<li>You can use a specific version of a database by specifying its full name. The database will appear without version name in your project (e.g. as <code>dbSNP</code>).</li>
<li>You can use the <code>--as</code> option to give databases different names if you would like to use different versions of the same database in the project.</li>
</ul>

<h3 id="advanced-usages-of-annotation-databases">Advanced usages of annotation databases</h3>

<p>Each database provide a number of fields and one or more default methods to link them to a project database. For example, if you look at the output of <code>vtools show annotation refGene</code>, you can see that this is a range-based database that is linked to a project using fields <code>chr</code>, <code>txStart</code>, and <code>txEnd</code>, which are chromosome name and start and ending of transcription region. The database, however, has other fields such as <code>cdsStart</code> and <code>cdsEnd</code> so you can link the database to a project using these two fields to identify variants in the coding regions of ref seq genes. You can even use regions such as <code>txStart-5000, txEnd+5000</code> to extend each region by 5k to include variants that are in vicinity of ref seq genes.</p>

<p>(:toggleexample Examples: alternative ways to link range-based annotation databases:) There are 730 variants in the regular ref seq gene regions,</p>

<pre><code>% vtools select variant 'refGene.chr is not NULL' -c


Counting variants: 13 456.6/s in 00:00:00
730
</code></pre>

<p>We can link to the refGene database using coding regions</p>

<pre><code>% vtools use refGene --linked_fields chr cdsStart cdsEnd


INFO: Downloading annotation database from annoDB/refGene.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/refGene-hg19_20110909.DB.gz
Binning ranges: 100% [===============================================] 40,843 24.3K/s in 00:00:01
INFO: Using annotation DB refGene in project test.
INFO: refseq Genes
</code></pre>

<p>only 225 variants are selected, that means most variants are not in the coding regions,</p>

<pre><code>% vtools select variant 'refGene.chr is not NULL' -c


Counting variants: 13 965.9/s in 00:00:00
225
</code></pre>

<p>You can also link the database by an expanded region of each transcription region (adding 5k before and after the region),</p>

<pre><code>% vtools use refGene --linked_fields chr 'txStart-5000' 'txEnd+5000'


INFO: Downloading annotation database from annoDB/refGene.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/refGene-hg19_20110909.DB.gz
Binning ranges: 100% [================================================] 40,843 62.2K/s in 00:00:00
INFO: Using annotation DB refGene in project test.
INFO: refseq Genes
</code></pre>

<p>This time more variants are selected by the refGene database,</p>

<pre><code>% vtools select variant 'refGene.chr is not NULL' -c


Counting variants: 14 847.2/s in 00:00:00
1170
</code></pre>

<p>(:exampleend:)</p>

<p>In addition to the fields used to link to the project, you can even change the type of annotation database by specifying the appropriate values to options <code>--anno_type</code> and <code>--linked_fieleds</code>. For example, the gawsCatalog database records is a position-based annotation database that records GWAS hits. Because of the uncertainty of the exact locations of these findings, you can link it to the project as a range-based database that find variants that are close to these GWAS hits.</p>

<p>(:toggleexample Examples: Link a position based database as a range-based database:) We would like to see if any of our variants has been identified by one of the Genome-wide association studies. <code>gwasCatalog</code> is a position-based database that records all identified variants by their positions.</p>

<pre><code>% vtools use gwasCatalog


INFO: Downloading annotation database from annoDB/gwasCatalog.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/gwasCatalog-hg19_20111220.DB.gz
INFO: Using annotation DB gwasCatalog in project test.
INFO: GWAS Catalog
</code></pre>

<p>No matching variant is found in our project</p>

<pre><code>% vtools select variant 'gwasCatalog.chr is not null' -c


Counting variants: 3 1.2K/s in 00:00:00
0
</code></pre>

<p>However, we can link the database as a range-based database to check if there is any variant that falls in the vicinity of any GWA hits.</p>

<pre><code>% vtools use gwasCatalog --anno_type range --linked_fields chr 'position - 10000' 'position + 10000'


INFO: Downloading annotation database from annoDB/gwasCatalog.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/gwasCatalog-hg19_20111220.DB.gz
Binning ranges: 100% [================================] 8,608 67.1K/s in 00:00:00
INFO: Using annotation DB gwasCatalog in project test.
INFO: GWAS Catalog
</code></pre>

<p>This time, 24 variants are identified as within 10k distance of one of the GWA hits.</p>

<p>(:exampleend:)</p>

<p>Because field-based databases can link to arbitrary fields, you do not have to use the default linking fields. For example, the <code>CancerGeneCensus</code> database by default links to the <code>refGene.name2</code> field by &lsquo;gene name&rsquo;. It can also be linked to a knownGene ID because the CancerGeneCensus database has this field.</p>

<p>(:toggleexample Examples: Linking a field database using an alternative field:) The <code>CancerGeneCensus</code> database by default links to the <code>refGene.name2</code> field by &lsquo;gene name&rsquo;.</p>

<pre><code>% vtools use CancerGeneCensus --linked_by refGene.name2


INFO: Downloading annotation database from annoDB/CancerGeneCensus.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/CancerGeneCensus-20130711.DB.gz
INFO: Using annotation DB CancerGeneCensus in project test.
INFO: This database contains variants from the Cancer Genome Project. It is
an ongoing effort to catalogue those genes for which mutations have been causally
implicated in cancer. The original census and analysis was published in Nature
Reviews Cancer and supplemental analysis information related to the paper is also
available. Currently, more than 1% of all human genes are implicated via mutation
in cancer. Of these, approximately 90% have somatic mutations in cancer, 20% bear
germline mutations that predispose to cancer and 10% show both somatic and
germline mutations.
INFO: 471 out of 23242 refgene.name2 are annotated through annotation database CancerGeneCensus
WARNING: 16 out of 487 values in annotation database CancerGeneCensus are not linked to the project.
</code></pre>

<p>From the output of <code>vtools show annotation CancerGeneCensus</code>, you can see that this database also provides <code>kgID</code> for each record. This allows you to link the database to the project using <code>kgID</code>,</p>

<pre><code>% vtools use knownGene


INFO: Downloading annotation database from annoDB/knownGene.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/knownGene-hg19_20121219.DB.gz
INFO: Using annotation DB knownGene in project test.
INFO: UCSC Known Genes


% vtools use CancerGeneCensus --linked_fields kgID --linked_by knownGene.name


INFO: Downloading annotation database from annoDB/CancerGeneCensus.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/CancerGeneCensus-20130711.DB.gz
INFO: Using annotation DB CancerGeneCensus in project test.
INFO: This database contains variants from the Cancer Genome Project. It is
an ongoing effort to catalogue those genes for which mutations have been causally
implicated in cancer. The original census and analysis was published in Nature
Reviews Cancer and supplemental analysis information related to the paper is also
available. Currently, more than 1% of all human genes are implicated via mutation
in cancer. Of these, approximately 90% have somatic mutations in cancer, 20% bear
germline mutations that predispose to cancer and 10% show both somatic and
germline mutations.
INFO: 433 out of 80922 knowngene.name are annotated through annotation database CancerGeneCensus
WARNING: 54 out of 487 values in annotation database CancerGeneCensus are not linked to the project.
</code></pre>

<p>Note that you can link <code>GeneSymbol</code> (the default link-out field in the database) to <code>knowGene.name</code> because none of the names will match</p>

<pre><code>% vtools use CancerGeneCensus --linked_by knownGene.name


INFO: Downloading annotation database from annoDB/CancerGeneCensus.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/CancerGeneCensus-20130711.DB.gz
INFO: Using annotation DB CancerGeneCensus in project test.
INFO: 0 out of 80922 knowngene.name are annotated through annotation database CancerGeneCensus
WARNING: 487 out of 487 values in annotation database CancerGeneCensus are not linked to the project.
</code></pre>

<p>(:exampleend:)</p>

<p>There is also nothing prevents you from using other annotation database as a field-based database. For example, the <code>gwasCatalog</code> database has a field <code>region</code> that records the cytoband of each GWAS signal belongs. If you link this field to database <code>cytoBand</code>, you can provide a list of GWAS hits within each cytoband.</p>

<p>(:toggleexample Examples: Use a position-based database as a field database to annotate cytoBand:) The gwasCatalog database has a field <code>region</code> that records the cytoband of each GWAS hit. If we first use <code>cytoBand</code>,</p>

<pre><code>% vtools use cytoBand


INFO: Downloading annotation database from annoDB/cytoBand.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/cytoBand-hg19_20111216.DB.gz
INFO: Using annotation DB cytoBand in project test.
INFO: Cyto Band
</code></pre>

<p>and link the <code>gwasCatalog</code> database (originally a position-based database) to it through field <code>region</code>,</p>

<pre><code>% vtools use gwasCatalog --anno_type field --linked_fields region --linked_by cytoBand.name


INFO: Downloading annotation database from annoDB/gwasCatalog.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/gwasCatalog-hg19_20111220.DB.gz
INFO: Using annotation DB gwasCatalog in project test.
INFO: GWAS Catalog
INFO: 745 out of 862 cytoband.name are annotated through annotation database gwasCatalog
WARNING: 1 out of 746 values in annotation database gwasCatalog are not linked to the project.
</code></pre>

<p>For each variant, we have cytoband information as <code>cytoBand.name</code> (as well as <code>gwasCatalog.region</code>), and all GWAS hits for that belong to that region,</p>

<pre><code>% vtools output variant chr pos cytoBand.name gwasCatalog.genes gwasCatalog.trait --all -l 10


1   4540    1p36.33 PRKCZ   Reasoning
1   5683    1p36.33 PRKCZ   Reasoning
1   5966    1p36.33 PRKCZ   Reasoning
1   6241    1p36.33 PRKCZ   Reasoning
1   9992    1p36.33 PRKCZ   Reasoning
1   9993    1p36.33 PRKCZ   Reasoning
1   10007   1p36.33 PRKCZ   Reasoning
1   10098   1p36.33 PRKCZ   Reasoning
1   14775   1p36.33 PRKCZ   Reasoning
1   16862   1p36.33 PRKCZ   Reasoning
</code></pre>

<p>Although in this case we should use option <code>--all</code> to list all GAWS hits</p>

<pre><code>% vtools output variant chr pos cytoBand.name gwasCatalog.genes gwasCatalog.trait --all -l 10


1   4540    1p36.33 NR  Body mass index
1   4540    1p36.33 PRKCZ   Height
1   4540    1p36.33 PRKCZ   Reasoning
1   5683    1p36.33 NR  Body mass index
1   5683    1p36.33 PRKCZ   Height
1   5683    1p36.33 PRKCZ   Reasoning
1   5966    1p36.33 NR  Body mass index
1   5966    1p36.33 PRKCZ   Height
1   5966    1p36.33 PRKCZ   Reasoning
1   6241    1p36.33 NR  Body mass index
</code></pre>

<p>Obviously, all variants belonging to the same cytoband will have the same gwas hits as annotations. (:exampleend:)</p>

<h3 id="create-your-own-annotation-databases">Create your own annotation databases</h3>

<p>You may create your own annotation databases to annotate, prioritize, and filter variants in your project. For example if you have a huge amount of data and you do not want to import all of them into a project (to reduce size and improve performance of the main project database), you can create annotation databases from input data.</p>

<p>The general process to create an annotation database is</p>

<ol>
<li>Write a <code>.ann</code> file that describe your annotation source (<a href="http://varianttools.sourceforge.net/Annotation/New" target="_blank">http://varianttools.sourceforge.net/Annotation/New</a>). You can usually start from one of the system <code>.ann</code> files under <code>~/.varianttools/annoDB</code>.</li>
<li>Use command <code>vtools use NAME.ann</code> to create an annotation database and use it in the project. You can specify source files using option <code>--files</code>. A option <code>--rebuild</code> will force the rebuild of database from source file and generate a versioned and compressed <code>.DB.gz</code> file.</li>
</ol>

<p>The <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Pipeline.AnnoUtils?action=edit" target="_blank">anno_utils</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Pipeline.AnnoUtils?action=edit" target="_blank">?</a> pipelines provide a few pipelines that can help you create annotation databases from various resources. For example, if you have a project with variants from unaffected individuals, you can export the variants as an annotation database. This database can be used by other projects to filter variants.</p>


    
    
        <div class="chevrons">
    <div id="navigation">
<a class="nav nav-prev" href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/show/" title="show"> <i class="fa fa-chevron-left"></i><label>show</label></a>
    <a class="nav nav-next" href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/select/" title="select" style="margin-right: 0px;"><label>select</label><i class="fa fa-chevron-right"></i></a></div>
  </div>

  </section>
</article>

<footer>

<div class="footline">
    

    

  

    <div class="github-link">
      <a href="Documentation/vtools%20Commands/use.md" target="blank"><i class="fa fa-code-fork"></i>
        Improve this page</a>
    </div>
  </div>


	<div>


  
    Create a content/_footer.md file to customize the footer content
  



	</div>
</footer>

<script src="https://vatlab.github.io/vat-docs/js/clipboard.min.js"></script>

<link href="https://vatlab.github.io/vat-docs/css/featherlight.min.css" rel="stylesheet">
<script src="https://vatlab.github.io/vat-docs/js/featherlight.min.js"></script>



<script src="https://vatlab.github.io/vat-docs/theme-flex/script.js"></script>

    

    
    

    
  </body>
</html>