<!DOCTYPE html>
<html>
  <head>
    <title>vatlab</title>
    
      <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
<meta name="generator" content="Hugo 0.36.1" />
<title>sbNSFP :: vatlab</title>
<link rel="shortcut icon" href="https://vatlab.github.io/vat-docs/images/favicon.png" type="image/x-icon" />
<link href="https://vatlab.github.io/vat-docs/css/font-awesome.min.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/css/nucleus.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/theme-flex/style.css" rel="stylesheet">

<link rel="stylesheet" href="https://vatlab.github.io/vat-docs/css/bootstrap.min.css">
<script src="https://vatlab.github.io/vat-docs/js/jquery-2.x.min.js"></script>
<script type="text/javascript">
      var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
</script>
<meta name="description" content="">


    
  </head>
  <body data-url="/applications/annotation/variants/dbnsfp/">
    
      <header>
  <div class="logo">
    
	
  
    <a class="baselink" href="https://vatlab.github.io/vat-docs/">vatlab</a>
  

  </div>
  <div class="burger"><a href="javascript:void(0);" style="font-size:15px;">&#9776;</a></div>
</header>
<article>
  <aside>
    <ul class="menu">
          <li data-nav-id="/" class="dd-item">
          <a href="https://vatlab.github.io/vat-docs/">
            <i class="fa fa-fw fa-home"></i>
          </a>
          </li>
    <li data-nav-id="/install/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/install/">Installation</a>
      </div>
    </li>
    <li data-nav-id="/documentation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/">Documentation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/keyconcepts/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/">
            Key concepts
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/tutorials/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/tutorials/">
            Tutorials
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/help/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/help/">
            Help
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/vtools-commands/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/">Vtools commands</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools-commands/init/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/init/">
            Init
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/import/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/import/">
            Import
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/update/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/update/">
            Update
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/phenotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/phenotype/">
            Phenotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/liftover/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/liftover/">
            lifttover
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/show/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/show/">
            show
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/use/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/use/">
            use
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/select/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/select/">
            select
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/exclude/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/exclude/">
            exclude
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/compare/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/compare/">
            compare
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/associate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/associate/">
            associate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/output/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/output/">
            output
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/export/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/export/">
            export
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/remove/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/remove/">
            remove
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/execute/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/execute/">
            execute
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/admin/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/admin/">
            admin
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/functions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/functions/">Functions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/functions/ref_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/ref_sequence/">
            ref_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/mut_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/mut_sequence/">
            mut_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/track/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/track/">
            track
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/genotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/genotype/">
            Genotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/samples/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/samples/">
            samples
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/in_table/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/in_table/">
            in_table
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/otherfunctions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/otherfunctions/">
            Other functions
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/vtools_report/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/">Vtools_report</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools_report/avg_depth/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/avg_depth/">
            avg_depth
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/discordance_rate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/discordance_rate/">
            discordance_rate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sam_info/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sam_info/">
            sam_info
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sequence/">
            sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/trans_ratio/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/trans_ratio/">
            trans_ratio
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/variant_stat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/variant_stat/">
            variant_stat
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/inbreeding_coefficient/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/inbreeding_coefficient/">
            inbreeding_coefficient
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/meta_analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/meta_analysis/">
            meta_analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_fields/">
            plot_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_geno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_geno_fields/">
            plot_geno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_pheno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_pheno_fields/">
            plot_pheno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_association/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_association/">
            plot_association
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/customization/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/">Customization</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/format/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/">
            Format
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/">Simulation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/custimzedsimu/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/custimzedsimu/">
            Customized simulations
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/models/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/">Models</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/models/peng1/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/peng1/">
            Peng2014_ex1
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/models/pen2/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/pen2/">
            Peng2014_ex2
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/">Pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/customizedpipeline/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/customizedpipeline/">
            Customized pipeline
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/pipelines/variant-calling/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/">Variant calling</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_hg19/">
            bwa_gatk33_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_b37/">
            bat_gatk33_b37
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_hg19/">
            bwa_gatk28_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_b37/">
            bwa_gatk28_b37
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/other-pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/">Other pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/other-pipelines/import_vcf/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/import_vcf/">
            import_vcf
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/anno_utils/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/anno_utils/">
            anno_utils
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/illumina/">
            Illumnina
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/annovar/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/annovar/">
            ANNOVAR
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/snpeff/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/snpeff/">
            snpEff
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/king/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/king/">
            KING
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/utilities/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/utilities/">Utilities</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/utilities/report_progress/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/report_progress/">
            report_progress
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/utilities/_md5sumd/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/_md5sumd/">
            md5sumd
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/">Application</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
    <li data-nav-id="/applications/annotation/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/">Annotation</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/tutorial/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/tutorial/">
            Tutorial
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/customized/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">
            Customized annotation databases
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/annotation/genes/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/">Genes</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/genes/ccdsgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/ccdsgene/">
            ccdsGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/hugogene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/hugogene/">
            Hugo gene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/entrezgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/entrezgene/">
            EntrezGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/knowngene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/knowngene/">
            knownGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/refgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/refgene/">
            refGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/cancergenecensus/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/cancergenecensus/">
            CancerGeneCensus
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/variants/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/">Variants</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/variants/cosmic/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/cosmic/">
            COSMIC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbnsfp/" class="dd-item active">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbnsfp/">
            sbNSFP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbsnp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbsnp/">
            dbSNP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/linkagephysicalmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/linkagephysicalmap/">
            linkage-physical map
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/databaseof/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/databaseof/">
            Database of Genomic Variants
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/exomesequencingpro/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/exomesequencingpro/">
            Exome Sequencing Project
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/gwas/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/gwas/">
            gwasCatalog
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/hapmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/hapmap/">
            HapMap
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/thousand/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/thousand/">
            Thousand Genomes
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/regions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/">Regions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/regions/low-complexity-regions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/low-complexity-regions/">
            Low complexity regions
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/phast-cons/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/phast-cons/">
            Phast cons
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/genomic-super-dups/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/genomic-super-dups/">
            Genomic super dups
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/cytoband/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/cytoband/">
            cytoBand
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/misc/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/">Misc</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/misc/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/illumina/">
            Illumina
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/keggpathway/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/keggpathway/">
            keggPathway
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/encode/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/encode/">
            ENCODE
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
      <li data-nav-id="/applications/discovery/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/discovery/">
            Discovery
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/">Association</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/data-explorationqc/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/data-explorationqc/">
            Data exploration/QC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/association-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/association-analysis/">
            Association analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/post-assoc-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/post-assoc-analysis/">
            Post-assoc analysis
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/basic-data-statistics/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/">Basic data statistics</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/basic-data-statistics/group-statwrite/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/group-statwrite/">
            Group stat&amp;Write
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-variant-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/">Single variant analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-variant-analysis/fisher-exact-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/fisher-exact-test/">
            Fisher exact test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-gene-association-methods/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/">Single gene association methods</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-gene-association-methods/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/cmc-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/cmc-test/">
            CMC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/c_alpha-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/c_alpha-test/">
            C-alpha test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/kbac-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/kbac-test/">
            KBAC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rbt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rbt-test/">
            RBT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rarecover-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rarecover-test/">
            RareCover test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/vt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/vt-test/">
            VT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/wss-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/wss-test/">
            WSS test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/asum-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/asum-test/">
            aSum test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/jointconditional-association-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/">Joint/Conditional association analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/collapsing-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/collapsing-methods/">
            Collapsing methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/aggregation-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/aggregation-methods/">
            Aggregation methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/weighted-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/weighted-methods/">
            Weighted methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/variable-thresholds/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/variable-thresholds/">
            Variable thresholds
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/skat-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/skat-test/">
            SKAT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/metaskat-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/metaskat-analysis/">
            MetaSKAT analysis
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/vat-stacking/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/vat-stacking/">VAT stacking</a>
      </div>
    </li>
    <li data-nav-id="/applications/association/create-your-test/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/">Create your test!</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/create-your-test/running-r-programs/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/running-r-programs/">
            Running R programs
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/create-your-test/programming-in-vat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/programming-in-vat/">
            Programming in VAT
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/view-association-result/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/">View association result</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/view-association-result/qq-plot-and-manhattan-plot/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/qq-plot-and-manhattan-plot/">
            QQ plot and Manhattan plot
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/development/">Development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/development/changelog/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/changelog/">
            ChangeLog
          </a>
        </div>
    </li>
      <li data-nav-id="/development/getinvolved/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/getinvolved/">
            Get Involved
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/administration/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/administration/">Administration</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/administration/allrecent/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/allrecent/">
            All recent changes
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/autolink/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/autolink/">
            Autolink
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/stylesheet/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/stylesheet/">
            Style sheet
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/trafficstat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/trafficstat/">
            Traffic statistics
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/under-development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/under-development/">Under development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/under-development/faq/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/under-development/faq/">
            FAQ
          </a>
        </div>
    </li>
        </ul>
    </li>




    </ul>
    <section>
    </section>
  </aside>
  <section class="page">
    
    <div class="nav-select">
      <center>Navigation : 
        <select onchange="javascript:location.href = this.value;">
          
    <option value="/install/" >
   Installation</option>
    <option value="/documentation/" >
   Documentation</option>
    <option value="/applications/" >
   Application</option> 
    <option value="/applications/annotation/" >
  - 
   Annotation</option> 
      <option value="/applications/annotation/introduction/" >-- Introduction</option>
      <option value="/applications/annotation/tutorial/" >-- Tutorial</option>
      <option value="/applications/annotation/customized/" >-- Customized annotation databases</option>
    <option value="/applications/annotation/genes/" >
  -- 
   Genes</option>
    <option value="/applications/annotation/variants/" >
  -- 
   Variants</option> 
      <option value="/applications/annotation/variants/cosmic/" >--- COSMIC</option>
      <option value="/applications/annotation/variants/dbnsfp/"  selected>--- sbNSFP</option>
      <option value="/applications/annotation/variants/dbsnp/" >--- dbSNP</option>
      <option value="/applications/annotation/variants/linkagephysicalmap/" >--- linkage-physical map</option>
      <option value="/applications/annotation/variants/databaseof/" >--- Database of Genomic Variants</option>
      <option value="/applications/annotation/variants/exomesequencingpro/" >--- Exome Sequencing Project</option>
      <option value="/applications/annotation/variants/gwas/" >--- gwasCatalog</option>
      <option value="/applications/annotation/variants/hapmap/" >--- HapMap</option>
      <option value="/applications/annotation/variants/thousand/" >--- Thousand Genomes</option>
  
    <option value="/applications/annotation/regions/" >
  -- 
   Regions</option>
    <option value="/applications/annotation/misc/" >
  -- 
   Misc</option>
  
      <option value="/applications/discovery/" >- Discovery</option>
    <option value="/applications/association/" >
  - 
   Association</option>
  
    <option value="/development/" >
   Development</option>
    <option value="/administration/" >
   Administration</option>
    <option value="/under-development/" >
   Under development</option>



        </select>
      </center>
    </div>
      <div>
        <div class="searchbox">
          <input data-search-input id="search-by" type="text" placeholder="Search...">
        </div>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/lunr.min.js"></script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/auto-complete.js"></script>
        <link href="https://vatlab.github.io/vat-docs/css/auto-complete.css" rel="stylesheet">
        <script type="text/javascript">
          
              var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
          
        </script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/search.js"></script>
      </div>
    

    <h1>sbNSFP</h1>
    
    
    
    

<h1 id="dbnsfp">dbNSFP</h1>

<p>(:skin vtools-annotation:)</p>

<p>(:toc:)</p>

<p>dbNSFP is an annotation database for non-synonymous SNPs assembled by Xiaoming Liu from the University of Texas School of Public Health (see citation below). With variant tools you can use the <strong>dbNSFP</strong> database or <strong>dbNSFP-light</strong> (a version with fewer features) - see which features are offered for each database version below.</p>

<h2 id="dbnsfp-1">dbNSFP</h2>

<p>There can be frequent changes of name and their meanings of the fields across versions. Please pay close attention to the comments of fields before you use them.</p>

<p>For the latest version dbNSFP 2.4</p>

<ul>
<li>For <code>SIFT_score</code>, lower score means more damaging.</li>
<li>For Polyphen2 scores, higher score means more damaging.</li>
<li>There are multiple scores in fields <code>SIFT_score_all</code>, <code>SIFT_pred_all</code>, <code>Polyphen2_HDIV_score_all</code>, <code>Polyphen2_HVAR_score_all</code>, <code>Polyphen2_HDIV_pred_all</code> and <code>Polyphen2_HVAR_pred_all</code>. If you need a score for selecting most damaging variants, use fields such as <code>SIFT_score</code>, <code>SIFT_pred</code>, <code>Polyphen2_HDIV_score</code>, <code>Polyphen2_HVAR_score</code> and <code>Polyphen2_HVAR_pred</code>.</li>

<li><p>There can be multiple records for a variant so output of <code>vtools output</code> might be surprising (e.g. output score 0.4 with criterion &lsquo;score &gt; 0.9&rsquo;). Use option <code>--all</code> if you would like to see scores for all records.</p>

<p>% vtools show annotation dbNSFP -v2</p>

<p>Annotation database dbNSFP (version hg18_hg19_2_1)
Description:            dbNSFP version 2.1, maintained by Xiaoming Liu from
  UTSPH. Please cite &ldquo;Liu X, Jian X, and Boerwinkle E. 2011. dbNSFP: a
  lightweight database of human non-synonymous SNPs and their functional
  predictions. Human Mutation. 32:894-899&rdquo; and &ldquo;Liu X, Jian X, and Boerwinkle
  E. 2013. dbNSFP v2.0: A Database of Human Nonsynonymous SNVs and Their
  Functional Predictions and Annotations. Human Mutation. 34:E2393-E2402.&rdquo; if
  you find this database useful.
Database type:          variant
Number of records:      89,617,785
Distinct variants:      84,484,850
Reference genome hg18:  chr, hg18_pos, ref, alt
Reference genome hg19:  chr, pos, ref, alt</p>

<p>Field:                  chr
Type:                   string
Comment:                Chromosome number
Missing entries:        0
Unique Entries:         24</p>

<p>Field:                  pos
Type:                   integer
Comment:                physical position on the chromosome as to hg19
                        (1-based coordinate)
Missing entries:        0
Unique Entries:         28,060,014
Range:                  6007 - 249212562</p>

<p>Field:                  ref
Type:                   string
Comment:                Reference nucleotide allele (as on the + strand)
Missing entries:        0
Unique Entries:         4</p>

<p>Field:                  alt
Type:                   string
Comment:                Alternative nucleotide allele (as on the + strand)
Missing entries:        0
Unique Entries:         4</p>

<p>Field:                  aaref
Type:                   string
Comment:                reference amino acid
Missing entries:        0
Unique Entries:         22</p>

<p>Field:                  aaalt
Type:                   string
Comment:                alternative amino acid
Missing entries:        0
Unique Entries:         22</p>

<p>Field:                  hg18_pos
Type:                   integer
Comment:                physical position on the chromosome as to hg19
                        (1-based coordinate)
Missing entries:        44,904 (0.1% of 89,617,785 records)
Unique Entries:         28,043,425
Range:                  4381 - 247179185</p>

<p>Field:                  genename
Type:                   string
Comment:                common gene name
Missing entries:        0
Unique Entries:         20,264</p>

<p>Field:                  Uniprot_acc
Type:                   string
Comment:                Uniprot accession number. Multiple entries separated
                        by &ldquo;;&rdquo;.
Missing entries:        17,068,597 (19.0% of 89,617,785 records)
Unique Entries:         55,816</p>

<p>Field:                  Uniprot_id
Type:                   string
Comment:                Uniprot ID number. Multiple entries separated by &ldquo;;&rdquo;.
Missing entries:        20,254,026 (22.6% of 89,617,785 records)
Unique Entries:         37,250</p>

<p>Field:                  Uniprot_aapos
Type:                   integer
Comment:                amino acid position as to Uniprot. Multiple entries
                        separated by &ldquo;;&rdquo;.
Missing entries:        17,068,597 (19.0% of 89,617,785 records)
Unique Entries:         2,687,476
Range:                  1 - 9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9</p>

<p>Field:                  Interpro_domain
Type:                   string
Comment:                Interpro_domain: domain or conserved site on which the
                        variant locates. Domain annotations come from Interpro
                        database. The number in the brackets following a
                        specific domain is the count of times Interpro assigns
                        the variant position to that domain, typically coming
                        from different predicting databases. Multiple entries
                        separated by &ldquo;;&rdquo;.
Missing entries:        60,454,832 (67.5% of 89,617,785 records)
Unique Entries:         9,922</p>

<p>Field:                  cds_strand
Type:                   string
Comment:                coding sequence (CDS) strand (+ or -)
Missing entries:        0
Unique Entries:         5</p>

<p>Field:                  refcodon
Type:                   string
Comment:                reference codon
Missing entries:        2,270,742 (2.5% of 89,617,785 records)
Unique Entries:         1,754</p>

<p>Field:                  SLR_test_statistic
Type:                   float
Comment:                SLR test statistic for testing natural selection on
                        codons. A negative value indicates negative selection,
                        and a positive value indicates positive selection.
                        Larger magnitude of the value suggests stronger
                        evidence.
Missing entries:        46,683,780 (52.1% of 89,617,785 records)
Unique Entries:         511,811
Range:                  -188.177 - 108.85</p>

<p>Field:                  codonpos
Type:                   integer
Comment:                position on the codon (1, 2 or 3)
Missing entries:        2,270,742 (2.5% of 89,617,785 records)
Unique Entries:         4
Range:                  1 - 3;2;3</p>

<p>Field:                  fold_degenerate
Type:                   integer
Comment:                degenerate type (0, 2 or 3)
Missing entries:        2,270,742 (2.5% of 89,617,785 records)
Unique Entries:         79
Range:                  0 - 2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;2;0</p>

<p>Field:                  Ancestral_allele
Type:                   string
Comment:                Ancestral allele (based on 1000 genomes reference
                        data). The following comes from its original README
                        file: ACTG - high-confidence call, ancestral state
                        supproted by the other two sequences actg - low-
                        confindence call, ancestral state supported by one
                        sequence only N    - failure, the ancestral state is
                        not supported by any other sequence -    - the extant
                        species contains an insertion at this postion .    -
                        no coverage in the alignment
Missing entries:        2,488,820 (2.8% of 89,617,785 records)
Unique Entries:         10</p>

<p>Field:                  Ensembl_geneid
Type:                   string
Comment:                Ensembl gene id
Missing entries:        0
Unique Entries:         20,839</p>

<p>Field:                  Ensembl_transcriptid
Type:                   string
Comment:                Ensembl transcript ids (separated by &ldquo;;&rdquo;)
Missing entries:        0
Unique Entries:         112,159</p>

<p>Field:                  aapos
Type:                   integer
Comment:                : amino acid position as to the protein &ldquo;-1&rdquo; if the
                        variant is a splicing site SNP (2bp on each end of an
                        intron)
Missing entries:        0
Unique Entries:         4,315,466
Range:                  -1 - 9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;9;27;9;9;9;9;35;9;9;9</p>

<p>Field:                  SIFT_score
Type:                   float
Comment:                SIFT score, If a score is smaller than 0.05 the
                        corresponding NS is predicted as &ldquo;D(amaging)&rdquo;;
                        otherwise it is predicted as &ldquo;T(olerated)&rdquo;.
Missing entries:        12,024,501 (13.4% of 89,617,785 records)
Unique Entries:         101
Range:                  0 - 1</p>

<p>Field:                  SIFT_score_converted
Type:                   float
Comment:                SIFTnew=1-SIFTori. The larger the more damaging.
Missing entries:        12,024,501 (13.4% of 89,617,785 records)
Unique Entries:         101
Range:                  0 - 1</p>

<p>Field:                  SIFT_pred
Type:                   string
Comment:                If SIFTori is smaller than 0.05 (SIFTnew&gt;0.95) the
                        corresponding NS is predicted as &ldquo;D(amaging)&rdquo;;
                        otherwise it is predicted as &ldquo;T(olerated)&rdquo;.
Missing entries:        12,024,501 (13.4% of 89,617,785 records)
Unique Entries:         2</p>

<p>Field:                  Polyphen2_HDIV_score_max
Type:                   float
Comment:                The maximum (most damaging) value of Polyphen2 score
                        based on HumDiv, i.e. hdiv_prob. Use
                        Polyphen2_HDIV_score to get a list of all scores.
Missing entries:        17,086,068 (19.1% of 89,617,785 records)
Unique Entries:         1,001
Range:                  0 - 1</p>

<p>Field:                  Polyphen2_HDIV_score
Type:                   string
Comment:                Polyphen2 score based on HumDiv, i.e. hdiv_prob. The
                        score ranges from 0 to 1, and the corresponding
                        prediction is &ldquo;probably damaging&rdquo; if it is in
                        [0.957,1]; &ldquo;possibly damaging&rdquo; if it is in
                        [0.453,0.956]; &ldquo;benign&rdquo; if it is in [0,0.452]. Score
                        cutoff for binary classification is 0.5, i.e. the
                        prediction is &ldquo;neutral&rdquo; if the score is smaller than
                        0.5 and &ldquo;deleterious&rdquo; if the score is larger than 0.5.
                        Multiple entries separated by &ldquo;;&rdquo;.
Missing entries:        17,084,053 (19.1% of 89,617,785 records)
Unique Entries:         8,590,602</p>

<p>Field:                  Polyphen2_HDIV_pred
Type:                   string
Comment:                Polyphen2 prediction based on HumDiv, &ldquo;D&rdquo; (&ldquo;probably
                        damaging&rdquo;), &ldquo;P&rdquo; (&ldquo;possibly damaging&rdquo;) and &ldquo;B&rdquo;
                        (&ldquo;benign&rdquo;). Multiple entries separated by &ldquo;;&rdquo;. Because
                        the availability of multiple values, use expression
                        such as &rsquo;D&rsquo; in Polyphen2_HDIV_pred instead of &rsquo;D&rsquo; =
                        Polyphen2_HDIV_pred to filter variants that are
                        probably damaging.
Missing entries:        17,084,053 (19.1% of 89,617,785 records)
Unique Entries:         83,942</p>

<p>Field:                  Polyphen2_HVAR_score_max
Type:                   float
Comment:                The maximum (most damaging) value of all Polyphen2
                        score based on HumVar, i.e. hvar_prob. Use
                        Polyphen2_HVAR_score_all to get a list of all scores.
Missing entries:        17,086,068 (19.1% of 89,617,785 records)
Unique Entries:         1,001
Range:                  0 - 1</p>

<p>Field:                  Polyphen2_HVAR_score
Type:                   string
Comment:                Polyphen2 score based on HumVar, i.e. hvar_prob. The
                        score ranges from 0 to 1, and the corresponding
                        prediction is &ldquo;probably damaging&rdquo; if it is in
                        [0.909,1]; &ldquo;possibly damaging&rdquo; if it is in
                        [0.447,0.908]; &ldquo;benign&rdquo; if it is in [0,0.446]. Score
                        cutoff for binary classification is 0.5, i.e. the
                        prediction is &ldquo;neutral&rdquo; if the score is smaller than
                        0.5 and &ldquo;deleterious&rdquo; if the score is larger than 0.5.
                        Multiple entries separated by &ldquo;;&rdquo;.
Missing entries:        17,084,053 (19.1% of 89,617,785 records)
Unique Entries:         10,999,020</p>

<p>Field:                  Polyphen2_HVAR_pred
Type:                   string
Comment:                Polyphen2 prediction based on HumVar, &ldquo;D&rdquo; (&ldquo;porobably
                        damaging&rdquo;), &ldquo;P&rdquo; (&ldquo;possibly damaging&rdquo;) and &ldquo;B&rdquo;
                        (&ldquo;benign&rdquo;). Multiple entries separated by &ldquo;;&rdquo;. Because
                        the availability of multiple values, use expression
                        such as &rsquo;D&rsquo; in Polyphen2_HVAR_pred instead of &rsquo;D&rsquo; =
                        Polyphen2_HVAR_pred to filter variants that are
                        probably damaging.
Missing entries:        17,084,053 (19.1% of 89,617,785 records)
Unique Entries:         83,681</p>

<p>Field:                  LRT_score
Type:                   float
Comment:                The original LRT two-sided p-value (LRTori).
Missing entries:        21,548,464 (24.0% of 89,617,785 records)
Unique Entries:         826,817
Range:                  0 - 1</p>

<p>Field:                  LRT_score_converted
Type:                   float
Comment:                Converted LRT original p-value (LRTnew). We converted
                        the LRTori to a score suggested by our Human Muation
                        (2011) paper: LRTnew=1-LRTori*0.5 if Omega<1, or
                        LRTnew=LRTori*0.5 if Omega>=1.
Missing entries:        21,548,464 (24.0% of 89,617,785 records)
Unique Entries:         1,168,826
Range:                  0 - 1</p>

<p>Field:                  LRT_pred
Type:                   string
Comment:                LRT prediction, D(eleterious), N(eutral) or U(nknown)
Missing entries:        21,548,464 (24.0% of 89,617,785 records)
Unique Entries:         3</p>

<p>Field:                  MutationTaster_score
Type:                   float
Comment:                MutationTaster score
Missing entries:        1,143,911 (1.3% of 89,617,785 records)
Unique Entries:         598,533
Range:                  0 - 1</p>

<p>Field:                  MutationTaster_score_converted
Type:                   float
Comment:                The converted score suggested by our Human Mutation
                        (2011) paper: if the prediction is &ldquo;A&rdquo; or &ldquo;D&rdquo;
                        MTnew=MTori; if the prediction is &ldquo;N&rdquo; or &ldquo;P&rdquo;,
                        MTnew=1-MTori.
Missing entries:        4,373,664 (4.9% of 89,617,785 records)
Unique Entries:         999,050
Range:                  0 - 1</p>

<p>Field:                  MutationTaster_pred
Type:                   string
Comment:                MutationTaster prediction, &ldquo;A&rdquo;
                        (&ldquo;disease_causing_automatic&rdquo;), &ldquo;D&rdquo;
                        (&ldquo;disease_causing&rdquo;), &ldquo;N&rdquo; (&ldquo;polymorphism&rdquo;) or &ldquo;P&rdquo;
                        (&ldquo;polymorphism_automatic&rdquo;)
Missing entries:        1,143,911 (1.3% of 89,617,785 records)
Unique Entries:         4</p>

<p>Field:                  MutationAssessor_score
Type:                   float
Comment:                MutationAssessor functional impact combined score
                        (MAori)
Missing entries:        14,986,410 (16.7% of 89,617,785 records)
Unique Entries:         2,145
Range:                  -5.545 - 5.975</p>

<p>Field:                  MutationAssessor_score_converted
Type:                   float
Comment:                Scaled to 0-1: MAnew=(MAori-(-5.545))/(5.975-(-5.545))
Missing entries:        14,986,410 (16.7% of 89,617,785 records)
Unique Entries:         2,139
Range:                  0 - 1</p>

<p>Field:                  MutationAssessor_pred
Type:                   string
Comment:                MutationAssessor&rsquo;s functional impact of a variant :
                        predicted functional (high, medium), predicted non-
                        functional (low, neutral)&rdquo; Please refer to Reva et al.
                        Nucl. Acids Res. (2011) 39(17):e118 for details
Missing entries:        14,986,410 (16.7% of 89,617,785 records)
Unique Entries:         4</p>

<p>Field:                  FATHMM_score
Type:                   float
Comment:                FATHMM default score (weighted for human inherited-
                        disease mutations with Disease Ontology); If a score
                        is smaller than -1.5 the corresponding NS is predicted
                        as &ldquo;D(AMAGING)&rdquo;; otherwise it is predicted as
                        &ldquo;T(OLERATED)&rdquo;. If there&rsquo;s more than one scores
                        associated with the same NS due to isoforms, the
                        smallest score (most damaging) was used. Please refer
                        to Shihab et al Hum. Mut. (2013) 34(1):57-65 for
                        details
Missing entries:        19,342,889 (21.6% of 89,617,785 records)
Unique Entries:         2,135
Range:                  -16.13 - 10.64</p>

<p>Field:                  FATHMM_score_converted
Type:                   float
Comment:                Scaled to 0-1 and reverse direction (the larger the
                        more damaging):
                        FATHMMnew=1-(FATHMMori-(-16.13))/(10.64-(-16.13))
Missing entries:        19,342,889 (21.6% of 89,617,785 records)
Unique Entries:         2,135
Range:                  0 - 1</p>

<p>Field:                  FATHMM_pred
Type:                   string
Comment:                If a FATHMM_score is &lt;=-1.5 the corresponding NS is
                        predicted as &ldquo;D(AMAGING)&rdquo;; otherwise it is predicted
                        as &ldquo;T(OLERATED)&rdquo;.
Missing entries:        19,342,889 (21.6% of 89,617,785 records)
Unique Entries:         2</p>

<p>Field:                  GERP_NR
Type:                   float
Comment:                GERP++ neutral rate
Missing entries:        541,067 (0.6% of 89,617,785 records)
Unique Entries:         1,258
Range:                  0.0465 - 6.17</p>

<p>Field:                  GERP_RS
Type:                   float
Comment:                GERP++ RS score, the larger the score, the more
                        conserved the site.
Missing entries:        541,067 (0.6% of 89,617,785 records)
Unique Entries:         8,412
Range:                  -12.3 - 6.17</p>

<p>Field:                  PhyloP_score
Type:                   float
Comment:                PhyloP score, the larger the score, the more conserved
                        the site.
Missing entries:        64,695 (0.1% of 89,617,785 records)
Unique Entries:         10,245
Range:                  -11.958 - 2.941</p>

<p>Field:                  mg29way_pi
Type:                   string
Comment:                The estimated stationary distribution of A, C, G and T
                        at the site, using SiPhy algorithm based on 29 mammals
                        genomes.
Missing entries:        0
Unique Entries:         7,239,991</p>

<p>Field:                  mg29way_logOdds
Type:                   float
Comment:                SiPhy score based on 29 mammals genomes. The larger
                        the score, the more conserved the site.
Missing entries:        1,348,155 (1.5% of 89,617,785 records)
Unique Entries:         223,955
Range:                  0.0003 - 37.9718</p>

<p>Field:                  LRT_Omega
Type:                   float
Comment:                estimated nonsynonymous-to-synonymous-rate ratio
                        (reported by LRT)
Missing entries:        21,548,464 (24.0% of 89,617,785 records)
Unique Entries:         842,708
Range:                  0 - 7780.54</p>

<p>Field:                  UniSNP_ids
Type:                   string
Comment:                &ldquo;rs numbers from UniSNP, which is a cleaned version of
                        dbSNP build 129, in format: rs number1;rs number2;&hellip;&rdquo;
Missing entries:        89,510,596 (99.9% of 89,617,785 records)
Unique Entries:         100,701</p>

<p>Field:                  KGp1_AC
Type:                   integer
Comment:                Alternative allele count in the whole 1000Gp1 data.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         2,172
Range:                  0 - 2184</p>

<p>Field:                  KGp1_AF
Type:                   float
Comment:                Alternative allele frequency in the whole 1000Gp1
                        data.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         2,571
Range:                  0 - 1</p>

<p>Field:                  KGp1_AFR_AC
Type:                   integer
Comment:                Alternative allele counts in the 1000Gp1 African
                        descendent samples.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         493
Range:                  0 - 492</p>

<p>Field:                  KGp1_AFR_AF
Type:                   float
Comment:                Alternative allele frequency in the 1000Gp1 African
                        descendent samples.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         1,062
Range:                  0 - 1</p>

<p>Field:                  KGp1_EUR_AC
Type:                   integer
Comment:                Alternative allele counts in the 1000Gp1 European
                        descendent samples.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         759
Range:                  0 - 758</p>

<p>Field:                  KGp1_EUR_AF
Type:                   float
Comment:                Alternative allele frequency in the 1000Gp1 European
                        descendent samples.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         1,185
Range:                  0 - 1</p>

<p>Field:                  KGp1_AMR_AC
Type:                   integer
Comment:                Alternative allele counts in the 1000Gp1 American
                        descendent samples.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         363
Range:                  0 - 362</p>

<p>Field:                  KGp1_AMR_AF
Type:                   float
Comment:                Alternative allele frequency in the 1000Gp1 American
                        descendent samples.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         735
Range:                  0 - 1</p>

<p>Field:                  KGp1_ASN_AC
Type:                   integer
Comment:                Alternative allele counts in the 1000Gp1 Asian
                        descendent samples.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         573
Range:                  0 - 572</p>

<p>Field:                  KGp1_ASN_AF
Type:                   float
Comment:                Alternative allele frequency in the 1000Gp1 Asian
                        descendent samples.
Missing entries:        89,278,976 (99.6% of 89,617,785 records)
Unique Entries:         939
Range:                  0 - 1</p>

<p>Field:                  ESP6500_AA_AF
Type:                   float
Comment:                Alternative allele frequency in the Afrian American
                        samples of the NHLBI GO Exome Sequencing Project
                        (ESP6500 data set).
Missing entries:        88,817,528 (99.1% of 89,617,785 records)
Unique Entries:         27,424
Range:                  0 - 1</p>

<p>Field:                  ESP6500_EA_AF
Type:                   float
Comment:                Alternative allele frequency in the European American
                        samples of the NHLBI GO Exome Sequencing Project
                        (ESP6500 data set).
Missing entries:        88,817,528 (99.1% of 89,617,785 records)
Unique Entries:         22,975
Range:                  0 - 1</p></li>
</ul>

<p>As a quick example, one can use dbNSFP to annotate all of the &ldquo;damaging&rdquo; non-synonymous variants from a list of variants. In this example, we find all of the variants predicted to be damaging by SIFT and PolyPhen2 from the master variant table, and we record these variants into a new table called &ldquo;damaging_ns_snps&rdquo;.</p>

<pre><code>vtools select variant &quot;SIFT_pred = 'D' OR PolyPhen2_HDIV_pred like '%D%'&quot; -t damaging_ns_snps
</code></pre>

<h2 id="dbnsfp-gene">dbNSFP_gene</h2>

<pre><code>% vtools use dbNSFP_gene --linked_by refGene.name2
% vtools show annotation dbNSFP_gene -v2


Annotation database dbNSFP_gene (version 2_1)
Description:            dbNSFP_gene version 2.1, maintained by Dr. Xiaoming
  Liu from UTSPH. Please cite &quot;Liu X, Jian X, and Boerwinkle E. 2011. dbNSFP:
  a lightweight database of human non-synonymous SNPs and their functional
  predictions. Human Mutation. 32:894-899&quot; and &quot;Liu X, Jian X, and Boerwinkle
  E. 2013. dbNSFP v2.0: A Database of Human Nonsynonymous SNVs and Their
  Functional Predictions and Annotations. Human Mutation. 34:E2393-E2402.&quot; if
  you find this database useful.
Database type:          field
Reference genome *:     Gene_name
  Gene_name             Gene symbol from HGNC
  Ensembl_gene          Ensembl gene id (from HGNC)
  chr                   Chromosome number (from HGNC)
  Gene_old_names        Old gene sybmol (from HGNC)
  Gene_other_names      Other gene names (from HGNC)
  Uniprot_acc           Uniprot acc number (from HGNC and Uniprot)
  Uniprot_id            Uniprot id (from HGNC and Uniprot)
  Entrez_gene_id        Entrez gene id (from HGNC)
  CCDS_id               CCDS id (from HGNC)
  Refseq_id             Refseq gene id (from HGNC)
  ucsc_id               UCSC gene id (from HGNC)
  MIM_id                MIM gene id (from HGNC)
  Gene_full_name        Gene full name (from HGNC)
  Pathway_Uniprot       Pathway(s) the gene belongs to (from Uniprot)
  Pathway_ConsensusPathDB Pathway(s) the gene belongs to (from
                        ConsensusPathDB)
  Function_description  Function description of the gene (from Uniprot)
  Disease_description   Disease(s) the gene caused or associated with (from
                        Uniprot)
  MIM_phenotype_id      MIM id(s) of the phenotype the gene caused or
                        associated with (from Uniprot)
  MIM_disease           MIM disease name(s) with MIM id(s) in &quot;[]&quot; (from
                        Uniprot)
  Trait_association_GWAS Trait(s) the gene associated with (from GWAS catalog)
  GO_Slim_biological_process GO Slim terms for biological process
  GO_Slim_cellular_component GO Slim terms for cellular component
  GO_Slim_molecular_function GO Slim terms for molecular function
  Expression_egenetics  Tissues/organs the gene expressed in (egenetics data
                        from BioMart)
  Expression_GNF_Atlas  Tissues/organs the gene expressed in (GNF/Atlas data
                        from BioMart)
  Interactions_IntAct   Other genes the gene interacted with (from IntAct)
                        gene name followed by Pubmed id in &quot;[]&quot;
  Interactions_BioGRID  Other genes the gene interacted with (from BioGRID)
                        gene name followed by Pubmed id in &quot;[]&quot;
  Interactions_ConsensusPathDB Other genes the gene interacted with (from
                        ConsensusPathDB) gene name followed by interaction
                        confidence in &quot;[]&quot;
  P_HI                  Estimated probability of haploinsufficiency of the
                        gene from doi:10.1371/journal.pgen.1001154)
  P_rec                 Estimated probability that gene is a recessive disease
                        gene from doi:10.1126/science.1215040)
  Known_rec_info        Known recessive status of the gene (from DOI]
                        10.1126/science.1215040) &quot;lof-tolerant = seen in
                        homozygous state in at least one 1000G individual&quot;
                        &quot;recessive = known OMIM recessive disease&quot; original
                        annotations from DOI:10.1126/science.1215040)
  Essential_gene        Essential (&quot;E&quot;) or Non-essential phenotype-changing
                        (&quot;N&quot;) based on Mouse Genome Informatics database. from
                        doi:10.1371/journal.pgen.1003484
</code></pre>

<h2 id="dbnsfp-light">dbNSFP_light</h2>

<p>This light version of dbNSFP is only available for dbNSFP 1.0.</p>

<pre><code>vtools show annotation dbNSFP_light -v2


Annotation database dbNSFP_light (version hg18_hg19_1.3)
Description: dbNSFP_light version 1.0, maintained by Xiaoming Liu from UTSPH.
    Please cite &quot;Liu X, Jian X, and Boerwinkle E. 2011. dbNSFP: a
    lightweight database of human non-synonymous SNPs and their
    functional predictions. Human Mutation. 32:894-899&quot; if you find
    this database useful.
Database type: variant
Number of records: 73,968,886
Number of distinct variants: 73,754,006
Reference genome hg18: ['chr', 'pos', 'ref', 'alt']
Reference genome hg19: ['chr', 'pos', 'ref', 'alt']

Field:           chr
Type:            string
Missing entries: 0 
Unique Entries:  24

Field:           pos
Type:            integer
Missing entries: 0 
Unique Entries:  24,918,243
Range:           4381 - 247179185

Field:           ref
Type:            string
Comment:         Reference nucleotide allele (as on the + strand)
Missing entries: 0 
Unique Entries:  4

Field:           alt
Type:            string
Comment:         Alternative nucleotide allele (as on the + strand)
Missing entries: 0 
Unique Entries:  4

Field:           aaref
Type:            string
Comment:         reference amino acid
Missing entries: 0 
Unique Entries:  21

Field:           aaalt
Type:            string
Comment:         alternative amino acid
Missing entries: 0 
Unique Entries:  21

Field:           hg19pos
Type:            integer
Comment:         physical position on the chromosome as to hg19 (1-based
                 coordinate)
Missing entries: 33 (0.0% of 73,968,886 records)
Unique Entries:  24,900,697
Range:           15925 - 249212562

Field:           PhyloP_score
Type:            float
Comment:         PhyloP score, phyloPnew=1-0.5x10^phyloPori if phyloPori&gt;0 or
                 phyloPnew=0.5x10^phyloPori if phyloPori&lt;0
Missing entries: 21,677 (0.0% of 73,968,886 records)
Unique Entries:  6,237
Range:           0 - 0.995645X

Field:           SIFT_score
Type:            float
Comment:         SIFT score, SIFTnew=1-SIFTori
Missing entries: 570,924 (0.8% of 73,968,886 records)
Unique Entries:  167
Range:           0 - 195561991

Field:           Polyphen2_score
Type:            float
Comment:         Polyphen2 score, i.e. pph2_prob
Missing entries: 10,400,231 (14.1% of 73,968,886 records)
Unique Entries:  1,005
Range:           0 - T

Field:           LRT_score
Type:            float
Comment:         LRT score, LRTnew=1-LRTorix0.5 if &lt;1, or LRTnew=LRTorix0.5 if
                 &gt;=1
Missing entries: 7,795,201 (10.5% of 73,968,886 records)
Unique Entries:  780,594
Range:           0 - T

Field:           LRT_pred
Type:            string
Comment:         LRT prediction, D(eleterious), N(eutral) or U(nknown)
Missing entries: 7,795,201 (10.5% of 73,968,886 records)
Unique Entries:  17

Field:           MutationTaster_score
Type:            float
Comment:         MutationTaster score
Missing entries: 5,514,812 (7.5% of 73,968,886 records)
Unique Entries:  999,920
Range:           0 - W

Field:           MutationTaster_pred
Type:            string
Comment:         MutationTaster prediction, &quot;A&quot; (&quot;disease_causing_automatic&quot;),
                 &quot;D&quot; (&quot;disease_causing&quot;), &quot;N&quot; (&quot;polymorphism&quot;) or &quot;P&quot;
                 (&quot;polymorphism_automatic&quot;)
Missing entries: 5,514,843 (7.5% of 73,968,886 records)
Unique Entries:  6

Field:           LRT_Omega
Type:            float
Comment:         estimated nonsynonymous-to-synonymous-rate ratio (reported by
                 LRT)
Missing entries: 7,795,201 (10.5% of 73,968,886 records)
Unique Entries:  837,566
Range:           0 - 0.995645X

Field:           GERP_NR
Type:            float
Comment:         GERP++ netral rate
Missing entries: 0 
Unique Entries:  1,218
Range:           0 - 195561992

Field:           GERP_RS
Type:            float
Comment:         GERP++ RS score
Missing entries: 0 
Unique Entries:  8,344
Range:           -11.6 - T

Field:           uniprot_acc
Type:            string
Comment:         Uniprot accession number
Missing entries: 0 
Unique Entries:  18,766

Field:           uniprot_id
Type:            string
Comment:         Uniprot ID number
Missing entries: 0 
Unique Entries:  17,552

Field:           uniprot_aapos
Type:            integer
Comment:         amino acid position as to Uniprot
Missing entries: 0 
Unique Entries:  8,815
Range:           0.53741 - Y
</code></pre>


    
    
        <div class="chevrons">
    <div id="navigation">
<a class="nav nav-prev" href="https://vatlab.github.io/vat-docs/applications/annotation/variants/cosmic/" title="COSMIC"> <i class="fa fa-chevron-left"></i><label>COSMIC</label></a>
    <a class="nav nav-next" href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbsnp/" title="dbSNP" style="margin-right: 0px;"><label>dbSNP</label><i class="fa fa-chevron-right"></i></a></div>
  </div>

  </section>
</article>

<footer>

<div class="footline">
    

    

  

    <div class="github-link">
      <a href="Applications/Annotation/Variants/dbnsfp.md" target="blank"><i class="fa fa-code-fork"></i>
        Improve this page</a>
    </div>
  </div>


	<div>


  
    Create a content/_footer.md file to customize the footer content
  



	</div>
</footer>

<script src="https://vatlab.github.io/vat-docs/js/clipboard.min.js"></script>

<link href="https://vatlab.github.io/vat-docs/css/featherlight.min.css" rel="stylesheet">
<script src="https://vatlab.github.io/vat-docs/js/featherlight.min.js"></script>



<script src="https://vatlab.github.io/vat-docs/theme-flex/script.js"></script>

    

    
    

    
  </body>
</html>