<!DOCTYPE html>
<html lang="en" class="js csstransforms3d">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
    <meta name="generator" content="Hugo 0.40.2" />
    <meta name="description" content="Documentation for Variant Tools">
<meta name="author" content="Bo Peng">

    <link rel="shortcut icon" href="https://vatlab.github.io/vat-docs/images/favicon.png" type="image/x-icon" />
<link rel="icon" href="https://vatlab.github.io/vat-docs/images/favicon.png" type="image/x-icon" />
    <title>Quality control :: Variant Tools</title>
    
    
    <link href="https://vatlab.github.io/vat-docs/css/nucleus.css?1529593313" rel="stylesheet">
    <link href="https://vatlab.github.io/vat-docs/css/font-awesome.min.css?1529593313" rel="stylesheet">
    <link href="https://vatlab.github.io/vat-docs/css/hybrid.css?1529593313" rel="stylesheet">
    <link href="https://vatlab.github.io/vat-docs/css/featherlight.min.css?1529593313" rel="stylesheet">
    <link href="https://vatlab.github.io/vat-docs/css/perfect-scrollbar.min.css?1529593313" rel="stylesheet">
    <link href="https://vatlab.github.io/vat-docs/css/auto-complete.css?1529593313" rel="stylesheet">
    <link href="https://vatlab.github.io/vat-docs/css/theme.css?1529593313" rel="stylesheet">
    <link href="https://vatlab.github.io/vat-docs/css/hugo-theme.css?1529593313" rel="stylesheet">
    
      <link href="https://vatlab.github.io/vat-docs/css/theme-mine.css?1529593313" rel="stylesheet">
    

    <script src="https://vatlab.github.io/vat-docs/js/jquery-2.x.min.js?1529593313"></script>
    
    <style type="text/css">
      :root #header + #content > #left > #rlblock_left{ 
          display:none !important;
      }
      
        :not(pre) > code + span.copy-to-clipboard {
            display: none;
        }
      
    </style>
    
  </head>
  <body class="" data-url="/documentation/tutorials/association/">
    <nav id="sidebar" class="">



  <div id="header-wrapper">
    <div id="header">
      <a id="logo"  href="https://vatlab.github.io/vat-docs/">
  <svg id="vtools-logo" width="150%" height="200%" viewBox="0 0 504 240" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:1.41421;">
<image id="image0" width="320" height="200" x="0" y="0"
xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAWYAAADeCAMAAAA5FlqIAAAMDWlDQ1BpY2MAAEiJlZcHVJPJFsfn
KymEhBaIgJTQmyCdANK7ICAdbIQkJKHEkBBU7MiigmsXC1Z0BUTBtQCyVuzKImCvDwsqK+tiwYbK
mySAPvfteedNznz55c69d/4zmfnODADqrmyxOAfVACBXlC+JCwtkpqSmMUmPAAI/VOAEdNgcqTgg
NjYK/GN5dwP6wnLVXp7rn/3+a9Hk8qQcAJBYyBlcKScX8iEAcH2OWJIPAKEd2s1m5Ivl/BaytgQK
BIBIljNfyQZyzlCyo8InIS4IcjAAZCqbLeEDoCbPzyzg8GEeNTFkRxFXKIK8HbIvR8DmQu6CPCY3
dzpkdSpk64zv8vD/I2fGSE42mz/CyrEoCjlYKBXnsGf9n9Pxv0tujmy4D1NYqQJJeJx8zHDeqrOn
R8oZakeOijKiYyBrQb4g5Cr85XxHIAtPHPLv5UiD4JwBBgAo4LKDIyHDuUQZsuzEgCF2ZksUsdAf
jRbmRyQMcYZketxQfrSAJw2JH2YBLyJqKOcSUU70MG/NFIZGQIYrDT1UKEhIVupEzxQIk6Ihq0Fu
l2bHRw75PygUBEUP+0hkcXLN5pDfZkpC45Q+mG6udHhcmAOHrdCgC9k/X5AQrozFUnjSlKhhbVxe
cIhSA8bliRKHNGNwdQXGDcWWiHNih/yxrbycsDjlPGP7pQXxw7Gd+XCBKecBe5TFHh+r1I+9E+fH
Jii14TiIAkEgGDCBDNYMMB1kAWFbb2Mv/KVsCQVsIAF8wAP2Q5bhiGRFiwg+40Eh+BMSD0hH4gIV
rTxQAO1fRqzKpz3IVLQWKCKywVPIubg+7ot741Hw6Q+rM87CPYfjmOrDvRJDiMHEcGIo0WZEBweq
zoFVAoR/t32LJDwldBAeEa4Tugi3QSRs5cExyxWKRkaWBJ4osgz9niYskvygnAkmgC4YFzo0ugwY
3TPsg1tC1W54IO4D9UPtOAPXB/a4KxxJAO4Hx+YGrd8rlI2o+DaXP/Yn1/f9GIfsarZqbkMqMkb0
B414/Zgl6Ls54sLvyB89sSXYQew8dgq7iB3FGgETO4E1Ya3YMTmPrIQnipUw3FucQls2zCMc9nGs
dexx/Py33tlDCiSK/xvk82bmyzdE0HTxLImQL8hnBsA3Mo8ZIeI4jGE6OzqxAJC/35WvjzcMxXsb
YVz6Zss7CYBnKTTyv9nYZgAceQoA/d03m9lruL1WAnCsnSOTFChtuPxBABSgDneGHjACZsAajskZ
uANv4A9CwHgQAxJAKpgKZ10AcqHqGWAOWAhKQBlYCdaBTWAb2AmqwT5wADSCo+AUOAcug3ZwHdyF
a6MbvAB94B0YQBCEhNAQOqKHGCMWiB3ijLAQXyQEiULikFQkHeEjIkSGzEEWIWXIamQTsgOpQX5F
jiCnkItIB3IbeYj0IK+RTyiGUlFt1BC1RMeiLDQAjUQT0CkoH81DC9FidDm6Aa1E96IN6Cn0Mnod
7UJfoP0YwFQxBmaC2WMsLAiLwdKwTEyCzcNKsXKsEqvDmuF/fRXrwnqxjzgRp+NM3B6uz3A8Eefg
efg8fBm+Ca/GG/Az+FX8Id6HfyXQCAYEO4IXIYKQQuATZhBKCOWE3YTDhLNwR3UT3hGJRAbRiugB
92YqMYs4m7iMuIVYTzxJ7CA+JvaTSCQ9kh3JhxRDYpPySSWkjaS9pBOkTlI36QNZlWxMdiaHktPI
InIRuZy8h3yc3El+Rh5Q0VCxUPFSiVHhqsxSWaGyS6VZ5YpKt8oARZNiRfGhJFCyKAspGyh1lLOU
e5Q3qqqqpqqeqhNVhaoLVDeo7le9oPpQ9SNVi2pLDaJOpsqoy6lV1JPU29Q3NBrNkuZPS6Pl05bT
aminaQ9oH9Toag5qEWpctflqFWoNap1qL9VV1C3UA9Snqheql6sfVL+i3quhomGpEaTB1pinUaFx
ROOmRr8mXdNJM0YzV3OZ5h7Ni5rPtUhallohWlytYq2dWqe1HtMxuhk9iM6hL6Lvop+ld2sTta20
I7SztMu092m3affpaOm46iTpzNSp0Dmm08XAGJaMCEYOYwXjAOMG49Mow1EBo3ijlo6qG9U56r3u
aF1/XZ5uqW697nXdT3pMvRC9bL1Veo169/VxfVv9ifoz9Lfqn9XvHa092ns0Z3Tp6AOj7xigBrYG
cQazDXYatBr0GxoZhhmKDTcanjbsNWIY+RtlGa01Om7UY0w39jUWGq81PmH8B1OHGcDMYW5gnmH2
mRiYhJvITHaYtJkMmFqZJpoWmdab3jejmLHMMs3WmrWY9Zkbm08wn2Nea37HQsWCZSGwWG9x3uK9
pZVlsuViy0bL51a6VhFWhVa1VvesadZ+1nnWldbXbIg2LJtsmy027baorZutwLbC9oodauduJ7Tb
YtcxhjDGc4xoTOWYm/ZU+wD7Avta+4cODIcohyKHRoeXY83Hpo1dNfb82K+Obo45jrsc7zppOY13
KnJqdnrtbOvMca5wvuZCcwl1me/S5PLK1c6V57rV9ZYb3W2C22K3Frcv7h7uEvc69x4Pc490j80e
N1narFjWMtYFT4JnoOd8z6OeH73cvfK9Dnj95W3vne29x/v5OKtxvHG7xj32MfVh++zw6fJl+qb7
bvft8jPxY/tV+j3yN/Pn+u/2fxZgE5AVsDfgZaBjoCTwcOD7IK+guUEng7HgsODS4LYQrZDEkE0h
D0JNQ/mhtaF9YW5hs8NOhhPCI8NXhd+MMIzgRNRE9I33GD93/JlIamR85KbIR1G2UZKo5gnohPET
1ky4F20RLYpujAExETFrYu7HWsXmxf42kTgxdmLFxKdxTnFz4s7H0+Onxe+Jf5cQmLAi4W6idaIs
sSVJPWlyUk3S++Tg5NXJXSljU+amXE7VTxWmNqWR0pLSdqf1TwqZtG5S92S3ySWTb0yxmjJzysWp
+lNzph6bpj6NPe1gOiE9OX1P+md2DLuS3Z8RkbE5o48TxFnPecH1567l9vB8eKt5zzJ9MldnPuf7
8NfwewR+gnJBrzBIuEn4Kis8a1vW++yY7KrswZzknPpccm567hGRlihbdGa60fSZ0zvEduIScVee
V966vD5JpGS3FJFOkTbla8OjTqvMWvaT7GGBb0FFwYcZSTMOztScKZrZOst21tJZzwpDC3+Zjc/m
zG6ZYzJn4ZyHcwPm7piHzMuY1zLfbH7x/O4FYQuqF1IWZi/8vcixaHXR20XJi5qLDYsXFD/+Keyn
2hK1EknJzcXei7ctwZcIl7QtdVm6cenXUm7ppTLHsvKyz8s4yy797PTzhp8Hl2cub1vhvmLrSuJK
0cobq/xWVa/WXF24+vGaCWsa1jLXlq59u27auovlruXb1lPWy9Z3bYja0LTRfOPKjZ83CTZdrwis
qN9ssHnp5vdbuFs6t/pvrdtmuK1s26ftwu23doTtaKi0rCzfSdxZsPPprqRd539h/VKzW3932e4v
VaKqruq46jM1HjU1ewz2rKhFa2W1PXsn723fF7yvqc6+bkc9o75sP9gv2//Hr+m/3jgQeaDlIOtg
3SGLQ5sP0w+XNiANsxr6GgWNXU2pTR1Hxh9pafZuPvybw29VR02OVhzTObbiOOV48fHBE4Un+k+K
T/ae4p963DKt5e7plNPXzkw803Y28uyFc6HnTp8POH/igs+Foxe9Lh65xLrUeNn9ckOrW+vh391+
P9zm3tZwxeNKU7tne3PHuI7jnX6dp64GXz13LeLa5evR1ztuJN64dXPyza5b3FvPb+fcfnWn4M7A
3QX3CPdK72vcL39g8KDyXzb/qu9y7zr2MPhh66P4R3cfcx6/eCJ98rm7+Cntafkz42c1z52fH+0J
7Wn/Y9If3S/ELwZ6S/7U/HPzS+uXh/7y/6u1L6Wv+5Xk1eDrZW/03lS9dX3b0h/b/+Bd7ruB96Uf
9D5Uf2R9PP8p+dOzgRmfSZ83fLH50vw18uu9wdzBQTFbwlYcBTBY0cxMAF5XAUBLhWcHeI+jqCnv
X4qCKO+MCgL/xMo7mqK4A1DlD0DiAgCi4BllK6wWkKnwW378TvAHqIvLSB0q0kwXZ2UuKrzFED4M
Dr4xBIDUDMAXyeDgwJbBwS+7oNjbAJzMU9775IUIz/jbbeV0ZZzeAvBD+TeQf2s6Nu1ZkgAAACBj
SFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAB71BMVEVVU1JcXFxdXV1e
Xl5hYWFra2tjY2NfX19gYGBiYmJsbGyJiYmLi4tzc3NlZWVwcHCAgICGhoaCgoJ3d3d0dHR9fX2Q
kJCvr6+pqalnZ2d4eHiVlZW8vLzIyMi6urqdnZ1xcXGKioqlpaW7u7vOzs7d3d3GxsZmZmZkZGRu
bm6BgYG9vb3j4+Pv7+/g4OB+fn52dnbNzc3t7e3r6+vJycmHh4d1dXWTk5OysrLS0tLp6en19fX5
+fnz8/PPz8/u7u7X19ewsLB6enqIiIi0tLTx8fH////+/v79/f2enp6NjY3Z2dnDw8Oampqbm5v6
+vrm5ua1tbWMjIxvb2+YmJjc3Nzk5ORqamrY2Nj4+Pj8/Pzo6OiSkpKmpqbh4eHFxcWxsbFtbW25
ubnn5+f39/etra3W1tarq6uFhYVpaWnBwcGPj4+3t7fs7Ox8fHzl5eX29vbR0dH09PTf39/MzMyf
n595eXmRkZHw8PDLy8vKysqUlJRycnLy8vKioqLU1NTExMTV1dWEhIS2traqqqri4uJ/f3+jo6O4
uLinp6ehoaH7+/vQ0NCOjo6kpKTAwMDb29ugoKCXl5d7e3va2tqurq7T09PCwsKoqKicnJy+vr6/
v7/q6uqZmZloaGisrKze3t6zs7PHx8eDg4OWlpavXb/fAAAAAWJLR0RGF7r57QAAAAlwSFlzAAAW
JQAAFiUBSVIk8AAAAAd0SU1FB+IFDBATK8QbO0MAAA4KelRYdFJhdyBwcm9maWxlIHR5cGUgaWNj
AABYha2Za5Lkqq6F/3sUZwgGhATD4Rlx5z+B88l2ZldWVe+4+8Z1Ne00NiD0WFqyj/8Z4/gPRyo1
HqcfO4kGO+3UccZwdenUZWIxRzGJ8cwl19ziedrivzM8bZxn4hzyoUGTJTsl5DOfMs7/w7FZ9Xhm
vo6Z4nxL9i+P4989HoKKZkua7sv89Gs8VLxb532jy3WW0NjwadHu6zyfATGZobnz1b/lWUAO1Hmp
8Z74LK8bal/7w3j3fzyfzq8TCZa5e1TuFco5MEKwqHZdr9cW4mmK/PZItOzpT+ehi11XXfeN9rqx
eLww2TNgvSYaGGjqfvr3+Ugay6F/k8j+Lqn+Iulx3bCfNz6s8+cogvzud8qUVv/c+Jfm//vx/z8R
KhyWv28lxufH0GJRSn6uw62z0Ip23dJzfE10az/0gRWjLNnPgHtgGAXLyp8FXv2zsUBmgXx8rrCG
h7Ckx+HOoLdgbjgNItKf/nvhiJ2y9pwlHJ83UnQ3/LLyvQAyYiuRmuWzXyNOu3GT/ZroHhzxW4wu
Oacf/UWz1BfMvCQtBVUMVDGOzxt1qWoVlcdDHoSJXQxt4KKvmHh2MIQlalZZL2XfMRDniRP2nN66
eBaYyyeSJOvHRNFyjm9lx2fPW1Qxs77NfE+UQtFpX1zj2XKKHpPBVfHa2g0fSUAa34KUj5VTLq67
XN46+qM7VsWab/Pfp+TduvJ3ZacbzCS+JX0WLupR51j/abWEsjNWsNeAx7NTR3em6C5/LjzjlWnk
7dmPX6QVtbHn9kKBxwhpq0tEnnn50a074e/KPOI6cqzxhS3fZ3U/JkQU79Y61hrd1rjv3YPuMaKO
wYzz542JNF92/3zoy2+fwP3nPWjdv12PxmJret8zkZv2x4rxy2B/Ru9J9vzlWVf2e2Wf+Yu418Bn
EcVBlz6TTk9XDFu0ztDhv497wqsz0gqtfl8xOsx/366ie8dyxefwmgPd1p86cYC7FD7u8yXJPz53
vB78ddXMdvIMnKEjE+svz7o+YX8Wybe1LT7KdiV+t1xGJ0v+SPN5P2+tJfq9Ww3H+VeTy7z90Vf8
8Jn8+Bo3V/05kUR9FJ4v/UqajxHuhwVvdmu5ZODJbRgJnLdb77gHpn9QpEv0qY/fHPT4zbm+TfjS
4V+s+T3WVmznSt/1EX8Jn18mi54g/0ma3wa5BT3H+fXlBp9+9GOLqRRwKJ1pGC3QKteDZtezqW8a
zwzvm8f7xg+IePuQ/eJj8Y/utL8k+hFb6WyIsmo692849N0Y4aWj9GX/T7SfngSyFXvne0m6QclK
Zq3wKe7gag+V9qRzCLRPSStxRLILKgnpAvY30/9y7HRB7IuXw5VHez33wY/soQCplvZAs/02oYZ4
Sd3rurA7gOnHbw/GtO+OcWfuPieuaJmsqh8P/iE6V17z1fedSUJ1buNTOUO4rZXc1nrlpGukpyRh
jHMozyq5Ot4cj9uXC0JviMXcxh4LuihMUJjAJatMUJmgMkFjTGNMY0xnTPcsgpOdOBzsjsYgLxcA
sXP2G0ac6q5+CU0dROuo2/+MNlF5otWD/zaJNtN4gG241QLwATWgDQoX+hDfDYbyadwDBQJSh8Ik
SBy88AuIGxorNB7unDtnJAxoOgCzAWgNizMgFjBC2A3/SzQvEoU2yeyAf4wb3RaYTaD1k0xO4yar
R6OPJBgLfWVBxHgOE8fGJF3hR9xHHXFitehGQgdx59O9I539dMdL1KQUgmdiEegpjSCFpCZyX8Io
yRaBzXXlurG11Ojo7Y7yyYDlZIKI3hVrO7ymK+nKBbGeEIHYDKpibUFiYUIpBK2gI8GUgqgywGIm
E3TibiJ7UejOM+PyGWPk1HCdeuZcSDIG4tJwkVzRUW786OQs+NKdu2iY+fpjizBgd+s7o2I9haWo
dVxvABy4fNuOe2RaLKMrAQD59GA2vNuiMyuYKWIZMOGFnrGyVe6haAPUbFIFsqjtQvlFFin4UcF/
CmBWUCbM+SyFa5RYGtedawYVysWCM1Z8t+J3lW1WFvLSohoOWQsXjQsUXge/WaUyoIVwNrbVkKzh
Eo0tNcKm4QINAzVU0eY423Y7I1En7XfyWs/h7KAlMAw+IGz36OGM4jsWHPDMQaYZuMLI46SWOge+
NZqdw3U08NzBrNP/yLZTPNTaOdHL5MGJx8/B/clv9LFwh4XzLaxHPUyBvEFlJFqYfaGDRQXMxs7N
/jdevqEeG6/e6Gp3h+997uXRjZaofXAqrMT+DZGZCLm5Mc1HBiL5+hdkg6MzhDJCaGhr1BBWYXMW
iKsQJYeoEgidEFsKcaQjxAVMO1RHGqtQEAMAtIZPjwwQZEITzkzgUWgF0RakdDj0BBwWawIOIR4B
Fh5y1kDOAC16yH0GduXR5nU5pX8NqiNo2UE7pd6kb7dgQINJCPD4YKUewdoKOFewbaGAIwQ6JUwO
pbRQ2g5l8puBFQSpbKlqD5Wd1o7rzum0NLTYjtBQTTMDjlbAN0JDoQgeeuqhs83OhB1J+mzBEQgX
CCNzth4G+hljhLHzgYArTMJ4mmNYDZg6TCZbQQNZOyx0sqqEhUOtVcPGGjutAPULm/7dd9irHfHC
UpRP5EXcHK8DNMA6TAUAFvbM8i0gwoqMihHiSKEXY+Fn2zHOSc5hIoArAlYxVYlpRHKag+WOIjOK
jSitR5nNoytmns1KI3PnQdt218fZjuhvBrTTFiQ10KRGM6KMCWwOstuMhYxZcKHSEGQKzBx2zqQo
PtY6I2hyED1ImHKkZo2tduB4AyIx9qTAco2dB/GI6NA2Uo1D4QlMOKaBmSNO1DJNjzhZeS78Eq+h
7I4IBoRbJGQiGwTKW9w1xM1W9ga4EmWREnyN6J2dOMCTpR3Uvfzo2HHH5LJFTSnWDu5L4smUAOdk
O8HOUsJkgqzixWRdSWZJ+H7KMg8qXqreEVPeM/n7DKrspG0lhWhRcFOBkjDqSEaZXQLsOjsrHKmM
Avjky0bV9pFqZ+cbvGOaZpLYKYllJzacuuKWraW+Cga1BKVKgy2NlShnYyJaEn6aQJiDMtbfqNHI
SCAJ9nG6Q2t4wopkxiRXPdUu8EG1eKES0W3gpQt5yFQUNWw5o1FEIdyRn+UuTkstT31eGnXdctAg
lVFMg8l5LtEY/U0DQNAF1igo5xCzjt+R48hYRboUkKIAjwQLtBCdtiowGWkU/42FEITEPuR68cHE
HamJv0OwqozeqIVwYOFcMcYcsiABS5fgCneuFHIG9/Yk+UUA20IG2EF6oCKfB7jAxYKYJoLOiGtQ
HVjLmJHE2jIKy/4OQsrGyp0oKzkbGx1EC7CqZFZt9ciw3ow2shFKpPFccMFiM5fRcwVkqlbwzjOe
5caw1mg7547WeqUt5kRHeYBiY2qeiDut5DlqBpmoQkdefWYgnwweMglAr7Iy41nMRjYAf7nTgFHw
SCMqjXUoNAnV4kzVNJFargLa30mtivewUiXIl7+qnUoIKkGupHW1ig+veiiRraRnLWtoZaJam9bt
nlLY+oYjmHZW7o3z3shp8PQNdS3AN9ShQ/kBf3h/1zVwBzS+YQobTIeOkejhLQvb4O+hZoMiYF+i
uSEIDDKhCOIPDxV/KXSYDMwE2sDcDeX4vkxBdt3FDBJkPZFfSBS6rPBsjc1qgTYsUEvUMCwkRw4D
4K0TSACrjULzpEK5QgyZ89cFBC7uAw4wGLG9YFliMInilKKQo0oY8wBETqAOBN4CJlghSsmfs+Dq
RRbBzqDcOuzItRv9ZVu59lQ3paGUQv1fxjpKBbRrRdi9CnFZ2miFVF/QYel7lKEgBn0T5JqMmXuX
ddlmlZ1Im9WLfz2AX35McgvsM7RSSR+Adqukcg+WCpYAOXCqQtARJznvmke9whDzVhhURZlHRRG1
yKilIxhOBburdQ+coNQ2cWqwvrddSYN18MjYViceRhaq5L262sJb5sFOe90bjqfQ20laFSO1KvBO
q0r5QjMamVKIKqKcACwtN9aCqqCAhs8cuNlqSNUKiaQMaQAsj1TIzGiNpE9GaZ0w66uS0UcbMMEp
cBFy+SJ1rBZgCHo0JPL3qdDY3M/VOjSkh0nCEjL+AOkTum+LHRKnlUgOqefSoGfIbbUDbDhdOTrS
ENvE96ScIa1WBoPSvfXMTIsf7mahj9qd8PQJtWUDwDiYtsn8ZNJN0EKaKXMWlEIN5gDG503WIyMA
JGlSC0D7gduRpbMzvEEoqwcFhzghwT/SHKWPY5A/WKzCUMpAGQO0GL3rYKbBRsbkDJ6ROHQACvBW
Gxv5qWbm2aFtabCXdUAl9qRcmsD9JN+gx4ocAJGsmSfaynnqBMWZDzXMoqTFhW8qfBi7Y63Zdj9Q
PuyXKmSUBv+CBVclg4+5WpwberAblQ3pklobwki5M+A3mSw2O0KmBXYt0sixZA/CEJw/5/KPSODF
wgFIidCAPuBYuP7EnoBqW4Am4NLRN8i64L1r+lvYWI6FYdcm9e0Bkc7UEyvvAF8Ie8NlUEmoG0ja
QgIh74DWYO9cW3VudaNBNAvF8S54HVyHGLANtOwGle9mG3MSKAUF1j2JkSV9rznASMJ2b+ou/2z2
/ux4XMXn8wJCVa83DRvsunpmrvcri+eLXzvXx/e1P0f4+Jj56vx2/l/1f36oex/pkqy9PjXm+808
qHyd63o+Qjz9d3XkDz7D8/09Dn6a0i/njwMiBQFL77cz99bG85kn7+KKxFK3Uqxfr3/2dNbHAZ+8
zrM9r/bnpUPsEo6Pb4qv4zEGPPX1UdJLuhFqOf92vF9EHf8FGWg5TAwZlgMAACYUSURBVHja7V35
I2LtF/9OoUJptWQnS1kiWVJkrSyjbFEMItkrIqFQlCVjGdmbLM0f+r13vF5LRTcpM2+fH2d6ju6n
555znrM9//vflxA+Hv8L0RwIhGgOCEI0BwQhmgOCEM0BQYjmgMADzTA4HO6TvLDw8AgELNhP9eng
nmYYEhWB8oVnWGRUNBqDCPZTfTq4oxmBicHi8AQMVFlh4URSbFw8NiFE80u4oxlDTkxKTklNgyor
PD0jk5KVHZUTovklXGiG51JpqXn5BYUpdKg6tii+OJ9RwiyN8E2t/814STM8t6y8glXJ5lRVR0Pc
lXB8MpfNramtC7Hsgpc0h0XWN9Q08nh8QVZTWS4USQhMcwuPV9L6NSbYz/QJ8ZJmDKGN1SgEaOaL
2jsioUhCdXZ183jiOFxPsJ/pE+IlzbnEtl6JFKS5L7OfCkVSAunbAF/anQrtHfiPwFU34wZLJCDN
4qF6SPtSFjs8IpWzokNesxu4OnSEUQWgm6WSrLHxCSiSiBkUrrxykh46ArqBK82yqekRHk8ywxqD
tpuVSSp1wWx8UbCf6FPClWYMIbVGLp3TzGcuaKF4wFhmo7AvZVEW7Cf6lHClGYHBtVbKdaylrOUV
fYT3krBMg7BwVQv5iP6fgLvDdtHYmq5kfb1EXFvqtdqAhZVWz6mNsRPIj/qmyAl9XVHOH3r0cUez
bHF4RNTeLmIsraK9lRMWXt6r5iabkB9mASfGFzY2lX/oQd4dzTl1q1ma+E0z25xi8Za2nLqtwrnC
7bqP+p6IBMvOYM3keO7fQzMioXlW0bSrsc7tTfV4qQQiv0+PcOf3iR/1PRMsq3vi7kyaT1Hw4MNt
WB9GSmrtb8vi8zVbdTneySFuzXNHtqdkH/Mtw8JpqweH3a27hI9TSh8K99kT9NG38R/dwIE7xdsD
tzJTrC7cP/byR4GK8OjRWVvySQca82ey7IHmifHouhWbgF9Q/c0rNQDPjSrWGU4Xw8M+4jvCUdr9
M9byeX14MJl6F9zTHBaepl+pEQrmNBteGbVc4oXNIL6khX3IZkNR09svM77X9XzIjxgQeMpsoyaa
ixuF8r4rkjdSwqOGVLq1sQ86aKOo4xn7+I95U9wBkRtZRKD6NdXmqU4DHobNNLPlfQde0YzeZTLs
Zz8/KKAPR00QiY6PeVPcAVPWsXm96K3x9wqey2HqNm7mONz5RSrqbSnk2hYDlDPj5wUMkVBWetF+
Wn3u1/SEZ5pJeSq1gGPe9iYYVDddOKdZKPoLAvoIDOk8u1eku6lN96fB9Uyz9ra7isdrrIkjv/22
1k1rdIoPjGcEDqiY8uUZCY+nqggQzTE/z+w8Hod753y7kIA8ZFQtf2A8I3Bw0M67wSSdn3Oanml2
YL+1sK0Ca++K7A0bj3SUJhlPG7TBpsgfcGDPB/h8Hi9/DO/wo1jPNCMd9Zf5bL7APPpWsgr4QXpv
tsshJWg/K8JkicCxjMcb2KEn+FHsa4W3RfEsHY83k3z0Rji088dVZaHX0Y/PDVhY/alQwOP5OUHx
Gs332lmtyrO8HhYrGlsraOn/GwwgAJhzfs7K42V9TYOQOXoTr9HcgzsxS/lSNSv29XAoKcnMrcb9
LVXNljudEHjmKb9um9doRlFXFBKAZ8059jVzAHcul4jOLH8Jy1/o6wyOWnRF86vf9BrNMCR+HSyN
sR/86PT8KaRjysi1VZCDTY+/QD+zSwwttf59nld7T2Dk6zXACL7+RzHko5uZ5V2vs4ZeAhast4M0
WSkvmD0v86vQ11t8qP3bGraVnZ9J8mwEw6MqVO/Om8BzqfqYewuAwITHlJFJTpPluEgfcx+Xg4X1
EMgyX/OA8NxIAiEcTNYiciZyetBaJYlExwLiYybcxKPoAM2G3jjt4/8gcxyYdybHXqc5t+hXq5k9
dzON9fxnYr62lrw3bwKPKFs5+npf5YQhp3/dqbii7A0Obx59jfp95A2LxI3WTvkaM8klNp+MpoOe
Q4LSqcSttl9OXp0t32WO/TRFuhhuOBZQGpyCO+e/uhnuqMOSvQmg+UwzGEgp5qrF2W2eaSRcn5Yo
yiM9nhRhSExkGpGgxDpNJrzlmIDufLmH4BGdztSku7xdNAIjKypdqJik2EbUEu7A6d0VmMSF56I7
Ro2zu64Hhrdlg/Udxz82aqoXSD0JxPqt64y4pZuRPrO4oCB/bfK6nNTzaOoi9IAg3NFsAUdgvflW
iq0rKiMf0/G42NWjdOidOBBohiE6v2+IpOoRSrwnXQUjXQ6I7kyeo8FIB+F76n7cJXN2zbiXnXmy
+mNc9nwPRaRNJdlGRmztdTnkqbHWXrHIrlNbpRy2rlKTRIcDu7H8QFNwuOr6U78t+8uXCWfGvKgx
a7MUTdpvbckfEDPm5AaRSsee09kH1trHHf86bvqjpOS9tRuAZb5gbmB2cHssvnZ4vfi0tzv5Qvm+
w9db7Zc5yv1CPp/DTa53X1ALC8PNq2av3edNYMgcWRHt+9H05MHg3mFfQWNBn0axdHCOlzmelrUA
f6NbKOAwin/Wfz1ZN2oKa9YUtkJVSYGhci2uDv7Fgd+8EQpunncPeCkbsC8reX1gtXZT+UJrb5a5
j6HTMQqXmDY74ENJ5S1bpH/ekTBZx+WhaqZRDVZ3SyWNIsXg7dDkqUjXWJJ1sPLOqu23aAZr8Pl8
Aaelyf1rA7hzisJrk9uoBwzZc7x4fXaqEd3cDQ9tJK/ZNCo7g2HfW8XRn1bnTYyPmgV8qbxv9m6J
2ZrUsJPY0VG+ddu618JMBUmQTW2LBPzD8qdpKm9lf/mS1rZewucX7LUmF7PuRuNTmKenzIrY76PF
XJDOyuwm/f3nZFNx8y2K3qw+OVh2PNM72L7aMTV2J2YMMPdL9e8sw3mLZkRO9B74+3ryJTDaI5ux
312eDiCCFLuVYhzhzhQWD6XG9u9eDw2fVRdyOSWD11vfOx9fbgd+LB98ZPZIL3M4/muzhYrJDa8D
DOHQLhk0vWCBOo/X2/G4S72XDezm8qQ+Hk8tPmzZuwQUbv/R5uZRtCyyftQGHqrnbh6SyrKpnfbh
iqGNZQaPZ63K2l7od6aVTZ1P3qXsKt+dF3yLZhjCkqdS83iNilGCu/+nrkzXZLvVzEiHqYGV1dfI
HWDufFemyajoIsCgTJ1QuJKSwppt56MSikBf3ID9AYbDpER6UUxkAgoBD8sB3To0BmR2wnkN/Aw1
pY/keS/7vlkD2J9q8V18FLHHQdUXFenDw8LCo5IK58A67nX6/efCZGXaY7L2OBX4LpyC+UU9dSIi
V1ZGshyjc94dSHi7Nb5sp7oA+DolgyY3Th1cezLPmnZbZIB01B8MMHQltqsjy2NMpAd3ThHbGQWz
T2o6kI7YGimfL+07+Ep090A5x/uFUsme0xfZYD6k6xCwadbCk+e5Sn1T9gxIfzHt2bZCRtWANGdb
3kksVJojO9pFoLZS/Op08R3hqPE7TfGmWwMIQxYt5M2etm5G6RMenSbgbf8Zd7A20/LzMQIGR9UD
ikkqV3Tpc91tm9yiBY1kppXui2zQ5y5XAB5aVfWLekAH9tzM4wmsLOfTf71vu7NWLdG++BNv04wh
XChA7Zw/6mroUNTE3r7WCw9txxO0H/GbX18mBXLLojMG7YUZpEcvGG4qZluFumSa+3cT1bnIVInP
LD7JBshfNHIAP2ad/vyzYbJf3QIhmzH4TC4ip2NWLZXYz158+sNpRiTgLsWAdmYsu1Y7O2jxAyqP
/YNIR2cZcKB9EVCE55LbJu2i6e9PVjmr5VLPpTcIzPHC5ECxCe6TbOTEYouAL698WXGCyJlqERry
53fKnv+t8SuVWrd27d/SHi/GlsCL9ucB54dtzrS81JygT1q4D9V1R68uMRjPgk3OWaGAnZ9y7P7z
MASxbXLk1KuItqtsREKsDfA0RHnKF1LD6k/ZBYcpv54V8cBR2GHN3Mzdgj7QNH8Jr68VA/qKvdfx
coQDYB4Zvf3e1kA/4DhTrJ4rfGo4TWsCPtvsiWbgpJmnYhtx3riurrLhER29wCbJz3wpHYZncnVZ
0y9qXuH0S5G6ZL0N8vyFV+ENzRHotm7QE6hZfPGSwpTDN/bicQgBcBg8LNcRXVxlfa4rTQoez2Ab
8hhupS3NWRU4mE+yQZoFnL4l1xJ3yx2jSpR99GLfWga5nIK9gAZC74FI+H4qATwBVvPz3QxDOJcr
B/JIEFhG5EaWYXdbJFJr1TL2Oc0MpueYtSWbyzGO+yYbVBocBuXCNSVMm5/jsCsnXxg7fLVEyuEO
BtqhAx+A1mqXSNWsxedxOMC4KBheGAuAgPA0YhFZW0cydaw0xU+OWAXCRuYTKsBHm0n2rA+BHSZZ
M/kmGzjHUhgjzP4YF3cUT1FLhYblF66b8xR8c0+dX/wJryZ3wQibLJ1U0pv6PK6Ro9zX2L3QYoic
sui2jOuh9o3LYoXtxmyXv7T89NYSiadzJgjsskHi/sHflg1DKNtbKm2jroWUtKUqq6vhBc3xS286
MDR/kS2m9AmE4pTEsqcnQTBv8lYDPQITWaS0lCYufPtNxfzpmlFhy2dw5COXT6jQ3h5WqVVJSk9S
sMk6teuDeycblG7jVrrRzceZZoPBnPLir4Jvllrs+bt8IM0Y8q6GL2CPDC4+ffVksdui7jb0a/UM
sBxC81jD5kU9iUwAXmwlzWkyjfdfLzGAAwD28RfTHyXPvDz2PgXpSmSYdfom+3ebM0fOcNW2+qZW
cyOXiX9uW2jAUWlkOFYWBJpBIyi3SgXmMee/5y4Yomj1TjX7ymAH5IReiYttqmjfKdc6nvgo1P6U
PuDM96QEEpxewOd7NnLkipaZarxvsu/dRR5PMf7SVE84v7EY7NPS5/44NrmRM5Ky6N9SNe9oBo1g
ZRVH+tSfROTgrrLMrTTPh4YJZ1PKwdBCNKmIikE+C01mAiqI9eS4QV+fkfD5Co+eMXqVKaLgfZMN
0iwFaca5pq/SW0fYNdHP0wD0sxKJvO/A4tcGRG+HV6K3Bm0lkqrDW+XDA2C0C4rKlluPdaCoTtzO
wezy9VTaS7+a2p/ZB5Ba+vggoNnh8WrqPf1kxP1l1VOVAkX2/W4GqHfZzajOxWS75DDxuf8E1mmA
Ydeg0ByBbq6dLRDqltIfQtwxXQcj9uUtT74uLObX9lrNWYZTFvHy8cAwPUDFk8fGU+RSHq833VNG
sW47q2Se5pvsL7BxBZ8POm4vP0stTzFLpC+KLGHY7AJO0GgGDcbdDF+adfRQKQn2m3iuz0D21A/b
7Da3E2Z66odUfH5N1ONOBGdxADRHeypTwN4xDNUmmE+yv8BMp1YB4OS5xNyKxk51fH7+tfOJHBgS
VwxmVR5/csCfSXtn+QAUmon784CZqjzo+ifUAuowz+WpE6b4Xq6d2eGu4A9so+fzbbGPpot0UCnn
8VqaPWTc4OMsjtAY9fDg0GT/Pu1Z3XU4ahuyqvj8561JgLFXWAEFZlt8kIDRdlxAG671LprTUsFs
Q9VhAwncKbCw0j22es1jv4n+iMngcLNp7nhLoO8M8KU1U49UKIfNbHAclXua4aj0XuDBYx+O+tBk
30dE2PmZLp4w8dtyoY6jun3aZRIWvnLIA6A5uj+KISdIF7Up3vX6+oXm8OgGEY8nNFC+g9SGyRJb
1CXreE9hI2WeSC2V79W7UwPUlbxKqZqCe6zrOW44NKgrW03ulQYYGebxun88vLqQZIPvwixHYGVT
TC8/7sC2TZ/qRBvNT6LTKOqvbpDmkX9i1hPjOxTb8rsDSd7TjCEvtEikfIFmFTQZOcqMAcDv8Ohn
gFXCAqui2dWrRlGjbo2NVnb1vzTBUbRMDaPE1k53/6Mhe8qBHfZYpwFFNuiM1htBv9lV8ydYfk6v
6Z4nAVDUH79p5hbH02SyGEL62LIo/+zivWFR72lGYEqzSwDvti8J7DKRTW2PvBbsAXecQKh4piPv
Qe3fsDUCNPUmPrQOIRKa97iqFma81j3NYDqJx3tMH0CRDaq39Bqwacfm0k+A3h3U6ITPd3NYZOJv
pcE2Z6/GTv0cbe0WmZd2se9t94Ey9J0Q12vg8XRro8dIGHr3TpUVF+Wxd474jTIjlHRnOAmynH/I
gyETImPKlFMnxTNSPp9//1b8fjRAH8orD6tr6e6VRoS+LUsgPDwiY6DLfkii8nhZP1+2OShTzGy+
QPysA/D3eRdwLoWNhdnb21fFim7j5Ddn5Hv7xaHQ3Jl4WQlWNiRHO5Da27Xu5xGO53Bgvx2y2X3z
G1tTdffZFRiih9788zxz/YwyopY+ndoI6kOp3MCd73C4NagY7WqhsJES+/CwUGSDRqzfBtJcuPoi
4gzHM8Gd/7xmCjjvnoFej0A4NzMiKjS2jv1yEife3S8OhebfVpwv4NxsmiJpreasPNprBYq4bUVh
vkaRvLH7nU6oq9OSCfSOrtW4y6uhzBtdI0NVvP/QeZdbdtFSZc/Puip1P8fcgY/vtp+e0B8OxVBk
g2pgpVc9oyoc/PW8Dg6GMM1z7fktB7/Iz5xSYsayii2UNDIYfTfVmavjMe/1maHSjOzpbxEAPOtq
KmjRs42VrVjPPzIMKTOlbhcfikX5tqX2k43p29GMi8WO5mgctu4XqzJr6bbr+GHvOrD7p92T103p
VPeH7cjmuFPKAmkC7oNsULOXn4rvhlabic/dRXDftmTXdjnTnpvSnOO2bLNuxri8nLTbbEFP+GW6
EqQLOeC0VrAXRajb60rNktsnXx0CgUgg95+n3FWfGpc24gAqrvd/jB9r03IQcG1ca/tq+uPeyjlO
vG3ochZ5St1OjKdW7Bw/fVzvZYNKY/w2s6le66Jf4YSMitTSmJc0wnPrjobvWkd3d1e0frslANq9
J+iFZOCIYp3rrmgXvTU5BobAUMuOsc76qQ6LFnixi4idEzkY8LYODIGuRYc/7i1ETqdWGzOR68kH
R/agtWXPBv15Lxvc+xPaY30PxlXBYYjktB7XzQrP0R9b6AQ0morxWxMeNJodtPhCKV8qqayeLfCy
DQYW5nj3e+ep3ccfsgMDaDQjJ6YAd4cvlRcUcOyTXV5Ng4EhkR/GxEfK9isgXpYEs/zWziBEDdF/
7iStQAPqnVTk30cUEOZNmj9HIfzdgEoz2J52T7NmlxwaIuwtoNJ8XzkPorur0x+O+38DUGlGxfyo
4QjATpGaqdDdU14DKs1wFC6ZywGr92f/mtEOAQD0axnrNjRzPJ638xZD+A3oNKO3wKJy7vyWv2cO
/M2ATrMsdto8V3IK8bqO/zig05xA2mXdDJ7j3p1U/y8BOs0oan3cRqpJ9jcM9wsYoNMMR/UQ6tLe
n1D4TyF0AXRAEKI5IAjRHBCEaA4IQjQHBCGaA4IQzQFBiOaAIERzQBCiOSAI0RwQhGgOCEI0BwQh
mgOCEM0BQYjmgCBEc0AQojkgCNEcEIRoDghCNAcEIZoDghDNAUGI5oAgRHNAEKI5IAjRHBCEaA4I
QjS7Bzwiwp+NnSGa3SNCrw/UXa7/VcBRPURnV3+MH1uYQjS7AhWD2780nuEw/mvI+1ia4WEoxJ/R
u/4IpENbHlesUnf/kr13wtEjPpRmeK6MmvBnTGJ4hIO2uiTmqqWaBV9v23SDj6MZOYG2RCf+iMUd
ozvDHTm5qDA3G/u1i3FhcHABDInBIJA9nZ1Qr/3wFfc3svF4hVvvuzj1GT6MZliP8+iKZevuXWpY
TUzHHxM7w3MeNva/5MKQns0MDIHC5MJhYT1kAqan/sePQDXI3V8JBtJc9+lphju0U2PJI+AURkP3
0uXt2FZqV/lUOs5Jo+GdJhzOhKdhLRaaqbQUryQUaUlYrPL4uI5MAEEmE8rSiPTx0vp6k1I5vnjU
RsPF3+57upLJ3+hMzBsBpzBqdrX+m7DwMTTDI+hx1ZoZOTh6T1jFqFSZNTfdhy29a9XF8xSWUWE8
3ZtnDg4uV6+xli9HN9sPlpmXw8MbtaMA4oaGTnbaMlrXjLPJV+3by8bu2Ybb4vWuQI2VyKnbPQSb
/zUL77wm9+NpRvZE5XWrSrgFBdyZmZkSu6q75cYsqiyxj6jE4hH7DIMB/BeDWyW3SgoGipdrzPbK
rJbDtdYDZvXsqaLGSDlbGmHP6foGDjVcubBAsUSZxk0ESDcjHfVX+Ww+/8XdsZ+R5og03GpD0vr8
XvV8cnb24FnD1/7U+PbL9Tsm827wrPVuafbQXqXmWKUCYaOoV3FjHrixHQ6cDp0sdQ+ogJ+hZERU
2ScW981wdcBPoe6rHuuICVS7JwypX8kE1MbNBfGz04zqPMZNrbRt7XzbP2pquvgandZDxDb/alvY
2tpt+vq1aSt+ej6rZI4jlRqyig+SJieThoeTztoTy2sns5eXimdZLMrdZFLe4NIeq3pWsTZ/mx7I
ESkYQmoNW9L7y4/t0h+km1E5PTJqGrqMSNTr9Wkx4RHIXEdkZ5oejUbrY2L06DJa4ihTw1XLB25X
nEoSCTCASrq2k0omWbA0mtPpxFtISqUFsJd4U1SUUxseSO8bORGdLZqhpP8xp0DPiNCnXy+LZ/Jf
G+d6D1hYYPuWYchwetO8qpKJ8+NPGyya4RFl5e22m4Mm8psPAwtsdzh467GipIqxnP7ua58fESya
EZjjprwayq4l4f2y/Iuw8FgWePPa/XB7PyFYNGO0F/MDtiTc5xtLdX8dtVRi7H/3cOxHBItm8KZ4
L6eMBhj30/Wtc3vfHX/+bnZg482VeSv+vWDLLwCHgYNX7F5h/eipB4vmCedYfsng1niMP1NB/vlm
46P5v28yhnB56psIFs3h6bcqid043KX/bOFocAOA98bPm/6CJNWE6dosEOiMJ3WfjWaMdvdGwBdY
10r/AqWBiumySQWGw9u698vyLxAJ39c4AM+K+r9gN8Mj0lkS6Wt3awcNv2dU8/l+vcouaJltuHPZ
ICzY2/mELl1RPEsH3jnoR5HBKyDA3hVwDL212s83aA29usTgC9zf6+0jgkYzjJbN5QgNS7jPN9CO
cFLTKBD69abt4NFcN9TLEAo8XywaPJDyVGoBZw//fkn/InhKI7x0TKHjKPx7Z6o/ADdR1NK/huac
46bZAo7r7dfBBgxRX8Pj/TU0R3ZUZLGtrrdfBxt/Fc0wBPnbvF0oeHaH8KfAPc1W9tN7vd+NoIX1
c9KZXImA39Lsx4ybX3BPM6cg2+JHocGi+f5+VndXfwYb9zS/dbEZRASb5m5/pun9gnuaZ7JT33tN
7lMEi+aw8Huas776tfnAD4AhStek/MrLX51+FBpsmv1aQuUf/PabGcu7/rwJI9g0+7Xu0j+A0Scr
5TN3ni+39gHBprlw34/F2n4C4drYOLI9JfOjyCCaQBufLzSwvr5DacDCHFQiue6YZKHRLORIhywS
84YX7tWKsniW7v7SdrcSJmIISizNclyGRhPL0ry7fjN4NIO3HBts7U7fKzVgyEhs+behjczJ7KWl
u4oO2uJ38uvuoXcr6rYH2I3G6yL3EmSmn6N5yUvZKfG7Wxk7F06vLogO3vGkgyWX6libZB9DR7CI
TjJpPDb1umEaJI1COTvP2J7ebVbKUO9dYRnkcuSVB27u3AqbINJWrs8Uosa5gnzW8jylOumXV7Yl
eIdt8DJpg61C6+N6JPrXdcNJVzQOTwJVQP2v+DNjvuhmvjaW+s4VMNMsRyCVU1xjGrBw09GBccDe
qBZaOWwdg8uYYe2SvClPC17oSJknUs/dtB/7EtJAJKTR+6+Hp887tNT7w3qEvqOCUsiV6Gpuce56
riCsgI0bwd4TxYvsCRwVqZ0aG+yuLJkpqVT1VQmlUklB4VmiV21tQaT5ckRe1d1O9yWsn0D6sXFV
sYAri8Sg7vPP8IhI8vcTClfYWLOJ63nfCifLKnClGRWTPlR8M2AsZmYPXg5lZjVK1CWssQ507mc2
gV++aNu7Z8StCwToNEegS3c3sq92TRPP/72n9No2Z+UWj06RE+DvWIHf47jQjHRo+4fWxJq9jZ3d
povE+vIGZnXx2aYp3LtyxuDRjN5ar8ku12Ogb2b90QFr/qS/7qWPgOypz7yZ43AL92otz/cYtBV4
ilz6kmYHfpWpEa/FlVvK0Pq0zh6qFoun0Yu8vZoreDQTM1qL232IgqFkzWc2Y9KiuwacmK48TZXV
Orf28+n/QlsBQ+BmJS9oRiTQM5glut6NdB8DXUHUzUk11XHQi2Fg1NjbFuM5Ls2dTozQx2Z2Vwms
+RvfI31dgcj5XiN9YQIT6Lt7DIm4NirGx0aJINZpMEcOL6fScqEpjbDw0tvZwuR0D4caeNkqk8Hj
MZZW0b6ueAgDPKU5pmvSzuNpjnxuyAwezbRiQ2PhwUURNBMIGK0aleJW6am+rbM/s4/HM7TUkn1d
ERaZ6EJz3XThHI9nPsc6fHzYINK8NGdlmzNpkF5DeFHGPKPyoCvG0wcctE0zn68WZx77ugI5McVS
S/l84xOascmNQh7PPrlbX4ZBJvSEy6gxaTHUyJ4cbzM/waOZvs7gWNkUHJSUKzxivLVv7nDBc/82
qvPHoYDPYazTfV0BQ9IuVWq+4CnNtPk5K48nsdvudsg9pProqZWutp/9Hbg6h5d7JHg0k6765Dwe
tKqjCP3PXnbBUpTnNC08IgrwEyQlZ3RfVwDGOcXMtrLnn2S2jzPzq0CzyGFULyxmDDVs5022nmW2
n+w2K2O8amsLHs3HmWI1j9fyHUpmOzxqKN9wM016pfIYThtkcB4TptBXgKEjBketSlI+fqQzMXNA
LQWrNxg2hUYlGqm0l9j7ROJuysYvpTfx8uDGNHi8wxUvz1G/oV9InmEwj17Na9DPSiS6tbEiX1fA
kOMUtbTqsOKJs4nRJg7PdotHZhobZ/ryVTMFukZDlYFtFRpurhboXuTmg6uboeYCtbfdVeI43Ktz
YrDLVVYG8yGTB30F4DcrpPzGmhPC4ycQmE5S89HJcLbiUJFcMZQ9yzLaum1mtkA4V1n97eUR3g2C
6TeD1ltzBGUcCbjvzOevd8Y69yRSxtIW2tcViJxmo1BgaHnRRwBPIONiU0dv43bTo5rO46+HbocO
CkHDWDJ48XapQXAdOmA3p0KZWgGuEbW/Oo4HjluzCh6bLaCvgCGcwAZgm4ePn38MgemRpRG0BHR4
uL6srIisJU8l3QDetFeNHUGkeZ4N0gxpOAgY1Hl998AQOBZH8HjYgL7iCxwP0CyZWae91eIT8/XM
Drp5jz6KRwSbZmgHWFCLurzOzwHHLxuEj0YP+go4KupUKJBKWOlvRZIx5F0NjydVU0xves/Bphla
AQGYCpDY18c9T6NFUcv3DBxVw4OagL4CjjINlgDHwGfVfXCEmyF6iIRYG4/HlxpL32xtCzbN0OIE
RZunOqlE8SPNY0GYbHE7nyM8/IqO8HXFFzixaV0kkXZ3Peas4ShMjusP9Q/NAiPu0+9m8cn42+7Q
v0BvzXN5PNVttAeTBkMqa2saOQXM0gdSoK8A24l3jDrh4/BKOEqGLcW75vzAWQV8AadqCf/mQTaY
ETqQZlFDdLj3a8rOZwsA2+6xzQ3Z831eJzT0Dv3bogx9BZjSKl9XqQe+PeSsUdTo6cFb19JFcFaB
QKi7mT7+8haCRzN4pOXzxXGlEIZSpqUOVkr5wN4bR7k5ooOlKmMDVo44KbHT9xXAv+G+zY/I86+d
/7xnPfUn3SPV8S6Nop0reZXyyrWUnx6jf5+AZm17VhVfcAOpVNFB2zmUSAUc42K4G5uGdOBPWAWc
xtmfjydL6CuQjtKkG4Zc+rgByLW9BQU1cYQXi2F1FTYDg7lT6kXdURBpbsiqEuoo/VCya2Gy5kmx
GngHavujLGXPiIPnUpUd8bN93BFWHP3xP6CvQMX8aAELCFQVUf+oM7DYSy0+W9RjnqyG5aaVz5fo
bPFOb2xLEGMarTMSQ+8tpBo6WFhR25kdDML3zh7sRz+pkYBHlK0MFQ8UcG9aL+gT8HeswBAWbsBy
mEergd5dZkjV+etHT6aMwcKIP5LydeaDdK8GHgePZjxFLTX0tkdBm/mdQEpNFuskUql8pLjhomOc
RrOQjrV1JHx6atKayD6wN9xGfq4doK5IsJybATeNb37UzaXxrBL5nCp5wUSgTuRgcoBTt9b0NYVl
zjpbQHsVLg/eBALcmpQvNBi36JBGpCESCIu1LLuEL5VzRTc1s8Xzd1fD7dMHzNOsEVXv+k6sJQ0D
f9eK8KgKFXjoOGx78KSRPditwRH5XElLa/yK6ZisxC2mxt0ZW4zMoaky7yxL8EoVwUYawG9+I0jp
ClRn6WbraWEft0ouFKqrDDPmltm9mhuVOKu4vY3uLnoNbUVkx22+vLEy6+nwNgc2ddI2wuXmF6ds
Nv34uX+y0WqsYW7vd3R6OdokaDQjo+5phnQ8AQFH9RThp1YzKd19jWr2nIEr6qYMth5c3TZ1kNIS
3KW8oK3oqR89rKy52sXJHilEOoi4piSWRmzWtFSfXQ2uZ96OLkQdEyNRXmbYgjYdBhXd69tuvn9s
bX/G7RWTUk0pXh7Mq93ZveiKLnptbpz3K3KUXSlncb+0z00zAkNY2dw+Y84vDQ43TFdk/IjSQmnm
CBrNuc0tIM2PbhMUwJAYKlFLwuLx4PB4JbkMnRYTnvtajtz7FcgJogVL7sQ8VwcwBIZaVEfHAgYU
sJ9kYmc4pO7c4M06ijaC3qmoIVAz870FDIH0/10twTOBpmIweyLaaP5kNH+BfcCI3eD1nkRXgzSr
btN9UBp/HII28ZZ81A0qjfxrL/LCfz6CNlVx/EQMmsDHcONfjWDRHNnRLvrdTOzPm58+L4JF88PN
TyGaPxQxP8Hke0hpfDD0R2CNPGACx/C+lmb/SQgWzWD7+W+aR0OexgeCXNtikMoLxMkXn2/Qwwcg
WDTXbWiq1CPVJ4sEP94u8nkRtN1cwWo5Pfg2Hsi7poKIYNFM3G/YX8ETvRv68ecjeMeTcvpnG432
gQjarKPIP+/O4ncgdJ15QBCiOSAI0RwQhGgOCEI0BwQhmgOCEM0BQYjmgCBEc0DwfxzTzllVslpK
AAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE4LTA1LTEzVDAwOjA0OjE2LTA1OjAwFMgNNQAAACV0RVh0
ZGF0ZTptb2RpZnkAMjAxOC0wNS0xMlQyMToxOTo0My0wNTowMNmtTIMAAAAASUVORK5CYII=" />
</svg>

</svg>
</a>
    </div>
    
        <div class="searchbox">
    <label for="search-by"><i class="fa fa-search"></i></label>
    <input data-search-input id="search-by" type="text" placeholder="Search...">
    <span data-search-clear=""><i class="fa fa-close"></i></span>
</div>

<script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/lunr.min.js?1529593313"></script>
<script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/auto-complete.js?1529593313"></script>
<script type="text/javascript">
    
        var baseurl = '\/';
    
</script>
<script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/search.js?1529593313"></script>

    
  </div>

    <div class="highlightable">
    <ul class="topics">

        
          
          


 
    
    
    <li data-nav-id="/installation/" title="Installation" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/installation/">Installation</a>
      </div>
      
      
              
    </li>
  
 

          
          


 
    
    
    <li data-nav-id="/documentation/" title="Documentation" class="dd-item 
        parent
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/">Documentation</a>
            <i  class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
    
    <li data-nav-id="/documentation/keyconcepts/" title="Key concepts" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/">Key concepts</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/documentation/keyconcepts/supportedtypes/" title="Variants" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/supportedtypes/">
        Variants
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/tutorials/" title="Tutorials" class="dd-item 
        parent
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/tutorials/">Tutorials</a>
            <i  class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
             
            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/help/" title="Help" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/help/">
        Help
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/vtools_commands/" title="Vtools commands" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/">Vtools commands</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/init/" title="init" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/init/">
        init
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/import/" title="Import" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/import/">
        Import
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/update/" title="update" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/update/">
        update
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/phenotype/" title="phenotype" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/phenotype/">
        phenotype
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/liftover/" title="lifttover" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/liftover/">
        lifttover
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/show/" title="show" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/show/">
        show
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/use/" title="use" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/use/">
        use
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/select/" title="select" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/select/">
        select
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/exclude/" title="exclude" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/exclude/">
        exclude
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/compare/" title="compare" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/compare/">
        compare
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/associate/" title="associate" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/associate/">
        associate
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/output/" title="output" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/output/">
        output
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/export/" title="export" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/export/">
        export
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/remove/" title="remove" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/remove/">
        remove
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/execute/" title="execute" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/execute/">
        execute
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_commands/admin/" title="admin" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/admin/">
        admin
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/functions/" title="Functions" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/functions/">Functions</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/documentation/functions/ref_sequence/" title="ref_sequence" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/functions/ref_sequence/">
        ref_sequence
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/functions/mut_sequence/" title="mut_sequence" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/functions/mut_sequence/">
        mut_sequence
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/functions/track/" title="track" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/functions/track/">
        track
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/functions/genotype/" title="genotype" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/functions/genotype/">
        genotype
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/functions/samples/" title="samples" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/functions/samples/">
        samples
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/functions/in_table/" title="in_table" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/functions/in_table/">
        in_table
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/functions/otherfunctions/" title="other functions" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/functions/otherfunctions/">
        other functions
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/vtools_report/" title="Vtools report" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/vtools_report/">Vtools report</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/avg_depth/" title="avg_depth" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/avg_depth/">
        avg_depth
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/discordance_rate/" title="discordance_rate" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/discordance_rate/">
        discordance_rate
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/sam_info/" title="sam_info" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sam_info/">
        sam_info
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/sequence/" title="sequence" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sequence/">
        sequence
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/trans_ratio/" title="trans_ratio" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/trans_ratio/">
        trans_ratio
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/variant_stat/" title="variant_stat" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/variant_stat/">
        variant_stat
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/inbreeding_coefficient/" title="inbreeding_coefficient" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/inbreeding_coefficient/">
        inbreeding_coefficient
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/transmission/" title="transmission" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/transmission/">
        transmission
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/meta_analysis/" title="meta_analysis" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/meta_analysis/">
        meta_analysis
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/plot_fields/" title="plot_fields" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_fields/">
        plot_fields
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/vtools_report/plot_geno_fields/" title="plot_geno_fields" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_geno_fields/">
        plot_geno_fields
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/vtools_report/plot_pheno_fields/" title="plot_pheno_fields" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_pheno_fields/">plot_pheno_fields</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/vtools_report/plot_association/" title="plot_association" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_association/">plot_association</a>
      </div>
      
      
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/customization/" title="Customization" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/customization/">Customization</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
    
    <li data-nav-id="/documentation/customization/simulation/" title="Simulation" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/">Simulation</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/documentation/customization/simulation/customized_simu/" title="Customized" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/customized_simu/">
        Customized
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/customization/simulation/models/" title="Models" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/">Models</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
    
    <li data-nav-id="/documentation/customization/simulation/models/peng1/" title="Peng2014_ex1" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/peng1/">Peng2014_ex1</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/customization/simulation/models/peng2/" title="Peng2014_ex2" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/peng2/">Peng2014_ex2</a>
      </div>
      
      
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/customization/format/" title="Format" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/customization/format/">Format</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
    
    <li data-nav-id="/documentation/customization/format/supportedformats/" title="Formats" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/">Formats</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/txt/" title="basic" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/txt/">
        basic
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/csv/" title="csv" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/csv/">
        csv
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/plink/" title="plink" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/plink/">
        plink
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/rsname/" title="rsname" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/rsname/">
        rsname
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/new/" title="cumtomized" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/new/">
        cumtomized
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/tped/" title="tped" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/tped/">
        tped
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/map/" title="map" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/map/">
        map
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/annovar/" title="ANNNOVAR" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/annovar/">
        ANNNOVAR
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/cga/" title="CGA" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/cga/">
        CGA
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/casava18snps/" title="CASAVA18Snps" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/casava18snps/">
        CASAVA18Snps
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/casava18indels/" title="CASAVA18Indels" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/casava18indels/">
        CASAVA18Indels
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/functor/" title="functor" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/functor/">
        functor
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/annovav/" title="ANNNOVAR variants" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/annovav/">
        ANNNOVAR variants
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/documentation/customization/format/supportedformats/pileup/" title="ANNNOVAR variants" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/supportedformats/pileup/">
        ANNNOVAR variants
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
             
            
          
            
            


 
    
    
    <li data-nav-id="/documentation/utilities/" title="Utilities" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/utilities/">Utilities</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/documentation/utilities/report_progress/" title="report progress" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/documentation/utilities/report_progress/">
        report progress
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/documentation/utilities/md5sumd/" title="md5sumd" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/documentation/utilities/md5sumd/">md5sumd</a>
      </div>
      
      
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

          
          


 
    
    
    <li data-nav-id="/applications/" title="Application" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/">Application</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
    
    <li data-nav-id="/applications/annotation/" title="Annotation" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/annotation/">Annotation</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/applications/annotation/tutorial/" title="Tutorial" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/tutorial/">
        Tutorial
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/customized/" title="Customized" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">
        Customized
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/annotation/genes/" title="Genes" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/annotation/genes/">Genes</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/applications/annotation/genes/ccdsgene/" title="ccdsGene" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/ccdsgene/">
        ccdsGene
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/genes/hugogene/" title="HugoGene" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/hugogene/">
        HugoGene
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/genes/entrezgene/" title="EntrezGene" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/entrezgene/">
        EntrezGene
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/genes/knowngene/" title="knownGene" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/knowngene/">
        knownGene
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/genes/refgene/" title="refGene" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/refgene/">
        refGene
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/genes/cancergenecensus/" title="CancerGeneCensus" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/cancergenecensus/">
        CancerGeneCensus
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/annotation/variants/" title="Variants" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/annotation/variants/">Variants</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/cosmic/" title="COSMIC" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/cosmic/">
        COSMIC
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/dbnsfp/" title="dbNSFP" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbnsfp/">
        dbNSFP
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/dbsnp/" title="dbSNP" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbsnp/">
        dbSNP
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/linkagephysicalmap/" title="Rutgers map" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/linkagephysicalmap/">
        Rutgers map
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/databaseof/" title="database" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/databaseof/">
        database
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/esp/" title="ESP" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/esp/">
        ESP
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/gwas/" title="gwasCatalog" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/gwas/">
        gwasCatalog
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/hapmap/" title="hapmap" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/hapmap/">
        hapmap
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/thousand/" title="thousandGenomes" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/thousand/">
        thousandGenomes
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/variants/ebi/" title="EBI" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/ebi/">
        EBI
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/annotation/regions/" title="Regions" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/annotation/regions/">Regions</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/applications/annotation/regions/phast_cons/" title="PhastCons" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/phast_cons/">
        PhastCons
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/regions/genomic_super_dups/" title="genomicSuperDups" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/genomic_super_dups/">
        genomicSuperDups
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/regions/cytoband/" title="cytoBand" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/cytoband/">
        cytoBand
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/annotation/misc/" title="Misc" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/annotation/misc/">Misc</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/applications/annotation/misc/illumina/" title="Illumina" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/illumina/">
        Illumina
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/misc/keggpathway/" title="keggPathway" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/keggpathway/">
        keggPathway
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/annotation/misc/encode/" title="ENCODE" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/encode/">
        ENCODE
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/discovery/" title="Discovery" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/discovery/">Discovery</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/" title="Association" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/">Association</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
    
    <li data-nav-id="/applications/association/qc/" title="QC" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/qc/">QC</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
             
            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/association_analysis/" title="Association analysis" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/association_analysis/">
        Association analysis
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/post_assoc/" title="Post-assoc analysis" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/post_assoc/">
        Post-assoc analysis
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/basic_data_statistics/" title="Basic statistics" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/basic_data_statistics/">Basic statistics</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/applications/association/basic_data_statistics/group_stat_write/" title="GroupStat&amp;Write" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/basic_data_statistics/group_stat_write/">
        GroupStat&amp;Write
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/single_variant/" title="Single variant" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/single_variant/">Single variant</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/applications/association/single_variant/fisher-exact-test/" title="Fisher exact test" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/single_variant/fisher-exact-test/">
        Fisher exact test
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/single_gene/" title="Single gene" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/single_gene/">Single gene</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
    
    <li data-nav-id="/applications/association/single_gene/cmctest/" title="CMC test" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/single_gene/cmctest/">CMC test</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/single_gene/c_alpha-test/" title="C-alpha test" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/single_gene/c_alpha-test/">
        C-alpha test
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/single_gene/kbac-test/" title="KBAC test" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/single_gene/kbac-test/">
        KBAC test
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/single_gene/rbt-test/" title="RBT test" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/single_gene/rbt-test/">
        RBT test
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/single_gene/rarecover/" title="RareCover test" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/single_gene/rarecover/">RareCover test</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/single_gene/vt-test/" title="VT test" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/single_gene/vt-test/">
        VT test
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/single_gene/wss-test/" title="WSS test" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/single_gene/wss-test/">
        WSS test
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/single_gene/asum/" title="aSum test" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/single_gene/asum/">aSum test</a>
      </div>
      
      
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/joint_conditional/" title="Multivariate" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/joint_conditional/">Multivariate</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
            
          
          
          
        
          
            
            


 
    
    
    <li data-nav-id="/applications/association/joint_conditional/collapisng/" title="collapsing" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/joint_conditional/collapisng/">collapsing</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/joint_conditional/aggre/" title="aggregation" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/joint_conditional/aggre/">aggregation</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/joint_conditional/weighted-methods/" title="weighted" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/joint_conditional/weighted-methods/">weighted</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/joint_conditional/variable-thresholds/" title="variable thresholds" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/joint_conditional/variable-thresholds/">
        variable thresholds
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/joint_conditional/skat-test/" title="SKAT test" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/joint_conditional/skat-test/">
        SKAT test
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/joint_conditional/metaskat-analysis/" title="MetaSKAT" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/joint_conditional/metaskat-analysis/">
        MetaSKAT
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/vat_stacking/" title="VAT stacking" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/vat_stacking/">VAT stacking</a>
      </div>
      
      
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/create_your_test/" title="Create your test!" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/create_your_test/">Create your test!</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/applications/association/create_your_test/running-r-programs/" title="running R programs" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/create_your_test/running-r-programs/">
        running R programs
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/applications/association/create_your_test/programming-in-vat/" title="programming in VAT" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/applications/association/create_your_test/programming-in-vat/">
        programming in VAT
        
        </a>
    </li>
     
  
 

            
          
             
            
          
        
        </ul>
              
    </li>
  
 

            
          
            
            


 
    
    
    <li data-nav-id="/applications/association/view_association_result/" title="View association result" class="dd-item 
        
        
       
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/applications/association/view_association_result/">View association result</a>
      </div>
      
      
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

            
          
        
        </ul>
              
    </li>
  
 

          
          


 
    
    
    <li data-nav-id="/development/" title="Development" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/development/">Development</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/development/changelog/" title="ChangeLog" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/development/changelog/">
        ChangeLog
        
        </a>
    </li>
     
  
 

            
          
            
            


 
    
      <li data-nav-id="/development/getinvolved/" title="Get involved" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/development/getinvolved/">
        Get involved
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

          
          


 
    
    
    <li data-nav-id="/administration/" title="Administration" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/administration/">Administration</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/administration/autolink/" title="Autolink" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/administration/autolink/">
        Autolink
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

          
          


 
    
    
    <li data-nav-id="/under-development/" title="Under development" class="dd-item 
        
        
        haschildren
        ">
     
      <div>
      <a  href="https://vatlab.github.io/vat-docs/under-development/">Under development</a><i  class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
      
      
      
        <ul>
        
          
          
          
          
        
          
            
            


 
    
      <li data-nav-id="/under-development/faq/" title="FAQ" class="dd-item ">
        <a href="https://vatlab.github.io/vat-docs/under-development/faq/">
        FAQ
        
        </a>
    </li>
     
  
 

            
          
        
        </ul>
              
    </li>
  
 

          
         
    </ul>

    
    
      <section id="shortcuts">
        <h3>More</h3>
        <ul>
          
              <li> 
                  <a class="padding" href="https://github.com/vatlab/vat-docs"><i class='fa fa-fw fa-github'></i> Github repo</a>
              </li>
          
              <li> 
                  <a class="padding" href="https://twitter.com/ScriptOfScripts"><i class='fa fa-twitter'></i> ScriptOfScripts</a>
              </li>
          
              <li> 
                  <a class="padding" href="https://vatlab.github.io/vat-docs/"><i class='fa fa-envelope'></i> bpeng@mdanderson.org</a>
              </li>
          
        </ul>
      </section>
    

    
    <section id="footer">
      <p>Built with <a href="https://github.com/matcornic/hugo-theme-learn">Hugo</a></p>

    </section>
  </div>
</nav>




<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>


        <section id="body">
        <div id="overlay"></div>
        <div class="padding highlightable">
              
              <div>
                <div id="top-bar">
                
                
                <div id="breadcrumbs" itemscope="" itemtype="http://data-vocabulary.org/Breadcrumb">
                    <span id="sidebar-toggle-span">
                        <a href="#" id="sidebar-toggle" data-sidebar-toggle="">
                          <i class="fa fa-bars"></i>
                        </a>
                    </span>
                  
                  <span id="toc-menu"><i class="fa fa-list-alt"></i></span>
                  
                  <span class="links">
                    
          
          
            
            
          
          
            
            
          
          
            
            
          
          
            <a href='https://vatlab.github.io/vat-docs/'></a> > <a href='https://vatlab.github.io/vat-docs/documentation/'>Documentation</a> > <a href='https://vatlab.github.io/vat-docs/documentation/tutorials/'>Tutorials</a> > Quality control
          
         
          
         
          
         
          
           
                  </span>
                </div>
                
                    <div class="progress">
    <div class="wrapper">
<nav id="TableOfContents">
<ul>
<li>
<ul>
<li><a href="#demonstration-of-quality-control-and-association-analysis">Demonstration of Quality Control and Association Analysis</a>
<ul>
<li><a href="#1-getting-started">1. Getting Started</a>
<ul>
<li><a href="#1-1-data-source">1.1 Data Source</a>
<ul>
<li><a href="#genotypes">Genotypes</a></li>
<li><a href="#genotype-attributes">Genotype attributes</a></li>
<li><a href="#phenotypes">Phenotypes</a></li>
</ul></li>
<li><a href="#1-2-software">1.2 Software</a>
<ul>
<li><a href="#annovar">ANNOVAR</a></li>
<li><a href="#king">KING</a></li>
<li><a href="#r-with-ggplot2">R with ggplot2</a></li>
</ul></li>
<li><a href="#computer-environment">Computer Environment</a>
<ul>
<li><a href="#cpu">CPU</a></li>
<li><a href="#ram-4x4-16gb">RAM (4X4=16GB)</a></li>
<li><a href="#cache">Cache</a></li>
<li><a href="#motherboard">Motherboard</a></li>
<li><a href="#harddrive-2x2tb">Harddrive (2X2TB)</a></li>
</ul></li>
</ul></li>
<li><a href="#2-initializing-a-project">2. Initializing a Project</a>
<ul>
<li><a href="#2-1-preparing-input-format-configuration-file">2.1 Preparing input format configuration file</a>
<ul>
<li><a href="#optimize-the-performance">Optimize the performance</a></li>
<li><a href="#import-from-multiple-sources">Import from multiple sources</a></li>
</ul></li>
<li><a href="#2-2-creating-annotations">2.2 Creating Annotations</a>
<ul>
<li><a href="#annotations-from-the-source-vcf-file">Annotations from the source VCF file</a></li>
<li><a href="#annovar-annotations"><code>ANNOVAR</code> Annotations</a></li>
</ul></li>
</ul></li>
<li><a href="#3-basic-quality-control-data-preprocessing">3. Basic Quality Control &amp; Data Preprocessing</a>
<ul>
<li><a href="#3-1-variant-genotype-level-qc">3.1 Variant &amp; Genotype Level QC</a>
<ul>
<li><a href="#low-quality-variants-filter">Low quality variants filter</a></li>
<li><a href="#low-quality-genotypes-filter">Low quality genotypes filter</a></li>
<li><a href="#qc-statistics-conditioned-on-genotype-information">QC statistics conditioned on genotype information</a></li>
</ul></li>
<li><a href="#3-2-processing-phenotype-sample-level-information">3.2 Processing Phenotype / Sample Level Information</a>
<ul>
<li><a href="#creating-sample-information-columns-based-on-sample-genotypes">Creating sample information columns based on sample genotypes</a></li>
<li><a href="#creating-phenotypes-based-on-existing-sample-information">Creating <em>phenotypes</em> based on existing sample information</a></li>
<li><a href="#updating-sample-information-for-association-testing">Updating sample information for association testing</a></li>
</ul></li>
<li><a href="#3-3-variant-and-sample-selection-for-association-analysis">3.3 Variant and Sample Selection for Association Analysis</a>
<ul>
<li><a href="#alternative-allele-frequency-calculations">Alternative allele frequency calculations</a></li>
<li><a href="#hwe-filter">HWE filter</a></li>
<li><a href="#creating-common-and-rare-variant-subsets">Creating common and rare variant subsets</a></li>
<li><a href="#selecting-functional-rare-variants">Selecting functional rare variants</a></li>
<li><a href="#summary">Summary</a></li>
</ul></li>
<li><a href="#3-4-additional-qc-for-missing-data">3.4 Additional QC for Missing Data</a>
<ul>
<li><a href="#variant-level-missingness">Variant level missingness</a></li>
<li><a href="#individual-level-missingness">Individual level missingness</a></li>
<li><a href="#group-specific-filtering-of-missing-data-for-rare-variant-association-analysis">Group specific filtering of missing data for rare variant association analysis</a></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
</nav>
    </div>
</div>

                
              </div>
            </div>
            

        
        <div id="body-inner">
          
            <h1>Quality control</h1>
          

        




<h2 id="demonstration-of-quality-control-and-association-analysis">Demonstration of Quality Control and Association Analysis</h2>

<h3 id="1-getting-started">1. Getting Started</h3>

<p>This section describes the data-set, required software and computational environment for the exome association analysis to be demonstrated in this tutorial.</p>

<h4 id="1-1-data-source">1.1 Data Source</h4>

<p>We use the exome data from the <strong>1000 genomes project</strong>. The entire data-set can be found at the <a href="ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20110521/">NCBI ftp site</a>. This release (version 3.0, April 30th, 2012) contains phased genotype calls on 1092 samples in VCF format, with 38.2M SNVs, 3.9M Short Indels and 14K Large Deletions. Data from different sources are pooled together. Targets for exome sequencing can be found at the <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/exome_pull_down_targets/">1000 genomes ftp site</a>.</p>

<p>We created a snapshot dataset for this demonstration with the original exome data plus simulated genotype quality scores and phenotype values. The dataset is 2.1GB and can be automatically downloaded and loaded for use with the following commands:</p>

<pre><code>vtools init qc
vtools admin --load_snapshot vt_qc
</code></pre>

<p><details><summary> Details about this snapshot data </summary></p>

<h5 id="genotypes">Genotypes</h5>

<p>Exome variant sites with higher coverage are extracted from the original VCF files, resulting the <code>g1000exomesnv.vcf.gz</code> (411MB) input exome data file.</p>

<pre><code>#!bin/bash 
chrs='1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 X' 
data\_dir=&quot;/home/guest/Data1000g&quot; 
zcat$data\_dir/ALL.chr1.phase1\_release\_v3.20101123.snps\_indels\_svs.genotypes.vcf.gz | head -500 |grep -P        &quot;^#&quot; | bgzip -c &gt; g1000exomesnv.header.vcf.gz 
for chr in \\(chrs 
    do echo \\(chr
    zcat \\(data_dir/ALL.chr$chr.phase1_release_v3.20101123.snps_indels_svs.genotypes.vcf.gz      | grep -v -P              &quot;^#&quot;     | grep -P &quot;SNPSOURCE=EXOME&quot; | grep -P &quot;VT=SNP&quot; | bgzip -c &gt; g1000exomesnv_chr$chr.vcf.gz
done 
fn=\`for i in \\(chr; do echo g1000exomesnv_chr$i.vcf.gz; done\` 
zcat g1000exomesnv.header.vcf.gz \\(fn | bgzip -c &gt; g1000exomesnv.vcf.gz 
</code></pre>

<h5 id="genotype-attributes">Genotype attributes</h5>

<p>The VCF file from 1000 genomes project has two genotype attributes, <em>Genotype dosage from MaCH/Thunder</em> and <em>Genotype Likelihoods</em>. To demonstrate the data QC procedures we need to have additional attributes such as <em>Genotype Depth</em> (GD) and <em>Genotype Quality</em> (GQ). We randomly simulate GD and GQ scores and append them to each genotype call, which is neither based on genotype dosage nor on genotype likelihoods. These scores will be used in QC for demonstration purposes.</p>

<h5 id="phenotypes">Phenotypes</h5>

<p>Two simulated phenotypes <em>BMI</em> and <em>smoking</em> combined with the original <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20110521/supporting/phase1_samples_integrated_20101123.ped">PED file</a> will be used as phenotype data for this demonstration.</p>

<p></details></p>

<h4 id="1-2-software">1.2 Software</h4>

<p>Quality control and association analysis will be performed with <em>variant association tools</em> (<code>vtools update</code>, <code>vtools select</code> and <code>vtools associate</code>) with simple annotations from <a href="https://vatlab.github.io/vat-docs/applications/association/"><em>variant association tools</em></a>. A few other software are occasionally needed for certain tasks along the pipeline.</p>

<p><details><summary>Other software</summary></p>

<h5 id="annovar">ANNOVAR</h5>

<p>We will use the <a href="http://www.openbioinformatics.org/annovar/"><code>ANNOVAR</code></a> program to determine which variants should be analyzed in association tests based on their functionality (nonsynonymous, splicing sites, etc).</p>

<h5 id="king">KING</h5>

<p>We will use <a href="http://people.virginia.edu/~wc9c/KING/"><code>KING</code></a> program for phenotype level quality control, i.e., to infer kinship and population structure. <a href="http://pngu.mgh.harvard.edu/~purcell/plink"><code>PLINK</code></a> program is needed to convert output to KING compatible format.</p>

<h5 id="r-with-ggplot2">R with ggplot2</h5>

<p><code>vtools report</code> requires R with ggplot2 in order to produce QQ plot and Manhattan plot. To install it in <code>R</code>:</p>

<pre><code>install.packages(&quot;ggplot2&quot;) 
</code></pre>

<p></details></p>

<h4 id="computer-environment">Computer Environment</h4>

<p>Demonstrative analysis for this tutorial are not computationally intensive and can be efficiently carried out in a reasonably powered computer. We have applied similar analysis on ~6,500 whole exome sequencing samples with a total of ~2 million variants and experienced satisfactory performance on a regular Debian GNU/Linux based workstation.</p>

<p><details><summary> Read the workstation configurations </summary></p>

<h5 id="cpu">CPU</h5>

<ul>
<li>product: AMD FX&trade;-8150 Eight-Core Processor</li>
<li>vendor: Advanced Micro Devices [AMD]</li>
<li>size: 3600MHz</li>
<li>capacity: 3600MHz</li>
<li>width: 64 bits</li>
<li>clock: 200MHz</li>
</ul>

<h5 id="ram-4x4-16gb">RAM (4X4=16GB)</h5>

<ul>
<li>description: DIMM Synchronous 1066 MHz (0.9 ns)</li>
<li>size: 4GiB (each, 16GB total)</li>
<li>width: 64 bits</li>
<li>clock: 1066MHz (0.9ns)</li>
</ul>

<h5 id="cache">Cache</h5>

<ul>
<li>description: L1 cache</li>
<li>size: 384KiB</li>
<li>capacity: 384KiB</li>
<li>clock: 1GHz (1.0ns)</li>
<li>description: L2 cache</li>
<li>size: 8MiB</li>
<li>capacity: 8MiB</li>
<li>clock: 1GHz (1.0ns)</li>
<li>description: L3 cache</li>
<li>size: 8MiB</li>
<li>capacity: 8MiB</li>
<li>clock: 1GHz (1.0ns)</li>
</ul>

<h5 id="motherboard">Motherboard</h5>

<ul>
<li>product: Crosshair V Formula</li>
<li>vendor: ASUSTeK COMPUTER INC.</li>
</ul>

<h5 id="harddrive-2x2tb">Harddrive (2X2TB)</h5>

<ul>
<li>product: ST32000641AS</li>
<li>vendor: Seagate</li>
<li>version: CC13</li>
<li>serial: 9WM6ZQJG</li>
<li>size: 1863GiB</li>
</ul>

<p></details></p>

<h3 id="2-initializing-a-project">2. Initializing a Project</h3>

<p>for convenience and portability of commands, we set <code>bash</code> variables:</p>

<pre><code>script_dir=&quot;./cache&quot;
data_dir=&quot;./data&quot;
</code></pre>

<h4 id="2-1-preparing-input-format-configuration-file">2.1 Preparing input format configuration file</h4>

<p>A &ldquo;data format configuration file&rdquo; *.fmt is required to build a project. This mechanism allows importing variants/samples from a number of arbitrary input format. It is always good practice to review and adapt a template fmt file for a specific data-set. We provide a few <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/import/">FMT templates</a>, among which a vcf format template is available. A modified version of the template <code>vcf.fmt</code> results in <code>g1000vcf.fmt</code>. We created this file by extracting the meta information of the source vcf file and make changes accordingly on the <code>fmt</code> template. You should find this file in your project directory if you have loaded our snapshot <code>vt_qc</code>. Using this snapshot, you do not need to import any data for this tutorial. However you can modify and use the configuration files for your other projects.</p>

<div class="notices info" ><p><code>g1000vcf.fmt</code> is configured such that it will not import variant INFO fields (we will create separated annotation database for it) and genotype likelihood GL field (we will not use it in the analysis)</p>
</div>


<h5 id="optimize-the-performance">Optimize the performance</h5>

<p>If you have a large volume of data to import, it is important to set &ldquo;runtime optimization&rdquo; parameters to speed up the process. Please read <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/import/">this documentation</a> for details.</p>

<h5 id="import-from-multiple-sources">Import from multiple sources</h5>

<p>If your dataset is stored in multiple VCF file, please read <a href="https://vatlab.github.io/vat-docs/documentation/vtools_commands/import/">this documentation</a> for properly importing such dataset.</p>

<h4 id="2-2-creating-annotations">2.2 Creating Annotations</h4>

<p>Annotations of variants can be incorporated to the project via</p>

<ul>
<li>building/using annotation databases (<code>vtools use</code>)</li>
</ul>

<p>or,</p>

<ul>
<li>adding annotation fields to variant database (<code>vtools update</code>).</li>
</ul>

<p>We will demonstrate both approaches.</p>

<h5 id="annotations-from-the-source-vcf-file">Annotations from the source VCF file</h5>

<p>Two files are required to create annotation database: the source data and the <code>.ann</code> configuration file. A number of <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">ANN templates</a> are provided. To <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">create customized annotation configuration</a> please refer to the online documentation.</p>

<p>The snapshot we provided has intentionally left out information in the <code>INFO</code> field of the VCF file, which you will find in the file <code>variants.tsv.gz</code> in this snapshot, as well as the <code>g1000.ann</code> file to build annotation database from the <code>INFO</code> field of VCF file.</p>

<pre><code>vtools use \\(script_dir/g1000.ann --file \\(data_dir/g1000exomesnv.vcf.gz -j8




INFO: Importing annotation data from variants.tsv.gz
variants.tsv.gz: 100% [================] 142,403 7.8K/s in 00:00:18
INFO: 291088 records are handled, 0 ignored.
INFO: Using annotation DB g1000 in project g1000.
INFO: 1000genome VCF file
</code></pre>

<p>Available annotation fields can be viewed by either</p>

<pre><code>vtools show annotation g1000
</code></pre>

<p>or</p>

<pre><code>vtools show fields
</code></pre>

<p><details><summary> which will show all the variant fields in current project.</summary></p>

<pre><code>variant.variant_id
variant.bin
variant.chr
variant.pos
variant.ref
variant.alt
g1000.chr                    Chromosome
g1000.pos                    1-based position
g1000.dbsnp_id               variant id (rs number or other IDs)
g1000.ref                    Reference allele, '-' for insertion.
g1000.alt                    Alternative allele, '-' for deletion.
g1000.qual                   phred-scaled quality score for the assertion made in
                             ALT. High QUAL scores indicate high
                             confidence calls.
g1000.filter                 PASS if this position has passed all filters, i.e. a
                             call is made at this position. Otherwise,
                             if the site has not passed all filters, a
                             semicolon-separated list of codes for
                             filters that fail.
g1000.LDAF                   MLE Allele Frequency Accounting for LD
g1000.AVGPOST                Average posterior probability from MaCH/Thunder
g1000.RSQ                    Genotype imputation quality from MaCH/Thunder
g1000.ERATE                  Per-marker Mutation rate from MaCH/Thunder
g1000.THETA                  Per-marker Transition rate from MaCH/Thunder
g1000.CIEND                  Confidence interval around END for imprecise variants
g1000.CIPOS                  Confidence interval around POS for imprecise variants
g1000.END                    End position of the variant described in this
                             record
g1000.HOMLEN                 Length of base pair identical micro-homology at event
                             breakpoints
g1000.HOMSEQ                 Sequence of base pair identical micro-homology at
                             event breakpoints
g1000.SVLEN                  Difference in length between REF and ALT alleles
g1000.SVTYPE                 Type of structural variant
g1000.AN                     Total Allele Count
g1000.AC                     Alternate Allele Count
g1000.AA                     Ancestral Allele, ftp://ftp.1000genomes.ebi.ac.uk/
                             vol1/ftp/pilot_data/technical/reference/a
                             ncestral_alignments/README
g1000.AF                     Global Allele Frequency based on AC/AN
g1000.AMR_AF                 Allele Frequency for samples from AMR based on AC/AN
g1000.ASN_AF                 Allele Frequency for samples from ASN based on AC/AN
g1000.AFR_AF                 Allele Frequency for samples from AFR based on AC/AN
g1000.EUR_AF                 Allele Frequency for samples from EUR based on AC/AN
g1000.VT                     indicates what type of variant the line represents
                             (eg: INDEL, SNP)
g1000.SNPSOURCE              indicates if a snp was called when analysing the low
                             coverage or exome alignment data (eg:
                             EXOME, LOWCOV)
</code></pre>

<p></details></p>

<h5 id="annovar-annotations"><code>ANNOVAR</code> Annotations</h5>

<p>We want to annotate variants with <code>ANNOVAR</code> for variants functionality so that we can focus on functional variants in rare variants association analysis. Although all variants in this dataset should be found in the large collection of readily available annotation databases in <code>variant annotation tools</code>, we want to demonstrate here the use of customized text based annotation sources. For readily usable annotation please refer to documentation of [<code>variant annotation tools</code>][11].</p>

<p>Firstly we output the variants for <code>ANNOVAR</code> to annotate:</p>

<pre><code>vtools output variant chr pos pos ref alt &gt; \\(data_dir/exomesnv.txt
/path/to/annotate_variation.pl \\(data_dir/exomesnv.txt \\(data_dir/humandb/ -buildver hg19




NOTICE: The --geneanno operation is set to ON by default
NOTICE: Reading gene annotation from humandb/hg19_refGene.txt ... Done with 40272 transcripts (including 6670 without coding sequence annotation) for 23416 unique genes
NOTICE: Reading FASTA sequences from /home/min/HD1s/humandb/hg19_refGeneMrna.fa ... Done with 29603 sequences
NOTICE: Finished gene-based annotation on 291088 genetic variants in exomesnv.txt
NOTICE: Output files were written to exomesnv.txt.variant_function, exomesnv.txt.exonic_variant_function
</code></pre>

<div class="notices tip" ><p>We use <em>refGene</em> hg19 database in <code>ANNOVAR</code>, although there are other options (see <code>ANNOVAR</code> website for details).</p>
</div>


<p>Then we import the annotated result into the project database, using two of our template format configurations:</p>

<pre><code>vtools update variant --format ANNOVAR_exonic_variant_function \
           --from_file \\(data_dir/exomesnv.txt.exonic_variant_function \
           --var_info mut_type function genename
vtools update variant --format ANNOVAR_variant_function \
           --from_file \\(data_dir/exomesnv.txt.variant_function \
           --var_info region_type region_name




INFO: Using primary reference genome hg19 of the project.
Getting existing variants: 100% [========================] 291,088 385.1K/s in 00:00:00
INFO: Updating variants from cache/exomesnv.txt.exonic_variant_function (1/1)
exomesnv.txt.exonic_variant_function: 100% [========================] 184,931 3.5K/s in 00:00:53
INFO: Fields mut_type, function, genename of 184,931 variants are updated
INFO: Using primary reference genome hg19 of the project.
Getting existing variants: 100% [==============================] 291,088 395.1K/s in 00:00:00
INFO: Updating variants from cache/exomesnv.txt.variant_function (1/1)
exomesnv.txt.variant_function: 100% [============================] 291,088 5.8K/s in 00:00:50
INFO: Fields region_type, region_name of 291,088 variants are updated
</code></pre>

<div class="notices tip" ><p><code>vtools show format ANNOVAR_exonic_variant_function</code> and <code>vtools show format ANNOVAR_variant_function</code> will display details of these configuration formats</p>
</div>


<pre><code>vtools show format ANNOVAR_exonic_variant_function




Format:      ANNOVAR_exonic_variant_function
Description: Output from ANNOVAR, generated from command
  &quot;path/to/annovar/annotate_variation.pl annovar.txt
  path/to/annovar/humandb/&quot;. This format imports chr, pos, ref, alt
  and ANNOVAR annotations. For details please refer to
  http://www.openbioinformatics.org/annovar/annovar_gene.html

Columns:
  None defined, cannot export to this format

variant:
  chr          Chromosome
  pos          1-based position, hg18
  ref          Reference allele, '-' for insertion.
  alt          Alternative allele, '-' for deletion.

Variant info:
  mut_type     the functional consequences of the variant.

Other fields (usable through parameters):
  genename     Gene name (for the first exon if the variant is in more than one
               exons, but usually the names for all exons are the
               same).
  function     the gene name, the transcript identifier and the sequence change in
               the corresponding transcript

Format parameters:
  var_info     Fields to be outputted, can be one or both of mut_type and function.
               (default: mut_type)




vtools show format ANNOVAR_variant_function




Format:      ANNOVAR_variant_function
Description: Output from ANNOVAR for files of type &quot;*.variant_function&quot;, generated
  from command &quot;path/to/annovar/annotate_variation.pl annovar.txt
  path/to/annovar/humandb/&quot;. This format imports chr, pos, ref, alt
  and ANNOVAR annotations. For details please refer to
  http://www.openbioinformatics.org/annovar/annovar_gene.html

Columns:
  None defined, cannot export to this format

variant:
  chr          Chromosome
  pos          1-based position, hg18
  ref          Reference allele, '-' for insertion.
  alt          Alternative allele, '-' for deletion.

Variant info:
  region_type The genomic region type (i.e., intergenic, ncRNA_intronic, etc) where
               this variant lies.

Other fields (usable through parameters):
  region_name Genomic region name that corresponds to the region_type. If the
               variant lies in an intergenic region, this field will
               specify the closest known regions upstream and
               downstream of this variant.

Format parameters:
  var_info     Fields to be outputted, can be one or both of region_type and
               region_name. (default: region_type)
</code></pre>

<p>A few additional fields regarding variant functionality are added to the variant table. They will be useful for selecting, filtering and grouping variants for association analysis.</p>

<pre><code>vtools show table variant -l3




variant_id, bin, chr, pos, ref, alt, region_type, region_name
1, 585, 1, 69536, C, T, exonic, OR4F5
2, 591, 1, 861275, C, T, intronic, SAMD11
3, 591, 1, 861315, G, A, UTR5, SAMD11
(291085 records omitted, use parameter --limit to display more or all records)
</code></pre>

<h3 id="3-basic-quality-control-data-preprocessing">3. Basic Quality Control &amp; Data Preprocessing</h3>

<p>Quality control is a crucial yet tedious step in association analysis. To facilitate this step, we provide a variety of summary statistic along with a number of sample/variant selection and filtering operations for quality control purpose. In this section we will demonstrate some basic quality control procedures. Please refer to our <a href="https://vatlab.github.io/vat-docs/applications/association/qc/">data exploration tutorial</a> for advanced summary statistics and QC measures.</p>

<h4 id="3-1-variant-genotype-level-qc">3.1 Variant &amp; Genotype Level QC</h4>

<p><code>vtools select</code> and <code>vtools exclude</code> commands implement variant level data selection and filtering. Variants can be subsetted on the basis of criteria defined by variant properties (variant information, annotations, summary statistics, etc) displayed by vtools show fields. We could either focus on subsets of variants of interest or remove non-informative subsets of variants after variants are subsetted.</p>

<h5 id="low-quality-variants-filter">Low quality variants filter</h5>

<p>We have made the <code>FILTER</code> column from the original vcf file available as annotation field <code>g1000.filter</code>. It can be used directly to remove low quality variants from the database. We select variants of good quality into a separate table, and use this table (instead of using table <code>variant</code>) for next steps of analysis.</p>

<pre><code>vtools select variant &quot;g1000.filter='PASS'&quot; -t variant_pass
</code></pre>

<p>An alternative approach would be applying <code>vtools remove</code> so that variants of low quality are removed from the variant table. <strong>Note that the remove command is irreversible.</strong></p>

<pre><code>vtools exclude variant &quot;g1000.filter='PASS'&quot; -t variant_to_rm
vtools remove variants variant_to_rm
</code></pre>

<h5 id="low-quality-genotypes-filter">Low quality genotypes filter</h5>

<p>Information on genotype calls (displayed in FORMAT column for each variant in VCF files) can be used for quality control of individual genotypes. To list available genotype information fields,</p>

<pre><code>vtools show genotypes -l 4




sample_name filename    num_genotypes   sample_genotype_fields
HG00096 simulatedQc1000g.tar.gz 291088  GT,DS,GD,GQ
HG00097 simulatedQc1000g.tar.gz 291088  GT,DS,GD,GQ
HG00099 simulatedQc1000g.tar.gz 291088  GT,DS,GD,GQ
HG00100 simulatedQc1000g.tar.gz 291088  GT,DS,GD,GQ
(1088 records omitted, use parameter --limit to display more or all records)
</code></pre>

<p>The <code>GT</code> and <code>DS</code> genotype fields are genotypes and genotype dosage from MaCH/Thunder. Both of them are from the original 1000 genomes project data. <code>GQ</code> and <code>GD</code> are simulated <em>genotype quality</em> and <em>genotype depth of coverage</em>. Genotype level filtering can be performed by removing genotype calls of given quality measure, which will <strong>irreversibly</strong> remove genotypes of low coverage and quality:</p>

<pre><code>vtools remove genotypes &quot;GD&lt;10 or GQ&lt;20&quot;
</code></pre>

<div class="notices tip" ><p>In many cases, however, we do not want to remove genotypes at the early stage of analysis, since we may later decide to change the QC criteria. It is possible to always calculate various statistics conditioning on genotypes defined by expressions such as <code>GD&gt;10</code> and <code>GQ&gt;20</code>, for example:
   <code>vtools update variant --from_stat ... --genotypes &lt;em&gt;GD&gt;10 and GQ&gt;20&lt;/em&gt;</code>
In this way we obtain statistic over genotype calls <em>excluding</em> the low quality genotypes. This approach is more flexible and allows evaluation of the same statistic conditional on different criteria.</p>
</div>


<h5 id="qc-statistics-conditioned-on-genotype-information">QC statistics conditioned on genotype information</h5>

<p>We will demonstrate here the calculation of summary statistics conditioning on genotype information. We set up a genotype condition expression from genotype information fields <code>GD</code> and <code>GQ</code>. The calculation of various statistic will skip genotypes failing to satisfy this expression, as you can see by comparing the counts of total genotypes and minor allele counts after executing the following commands.</p>

<p>Statistic without genotype condition constraints:</p>

<pre><code>vtools update variant --from_stat 'total=#(GT)' 'num=#(alt)' \
       'het=#(het)' 'hom=#(hom)' 'other=#(other)' \
       'wildtype=#(wtGT)' 'mutants=#(mutGT)' 'missing=#(missing)'




Counting variants: 100% [===================] 1,092 1.5/s in 00:12:08
INFO: Adding field num
INFO: Adding field hom
INFO: Adding field het
INFO: Adding field other
INFO: Adding field total
Updating variant: 100% [=================] 291,088 79.2K/s in 00:00:03
INFO: 291087 records are updated
</code></pre>

<p>Statistic with genotype condition constraints:</p>

<pre><code>vtools update variant --from_stat 'total_GD10GQ20=#(GT)' 'num_GD10GQ20=#(alt)' \
       'het_GD10GQ20=#(het)' 'hom_GD10GQ20=#(hom)' 'other_GD10GQ20=#(other)' \
       'wildtype_GD10GQ20=#(wtGT)' 'mutants_GD10GQ20=#(mutGT)' 'missing_GD10GQ20=#(missing)\
'\
       --genotypes &quot;GD&gt;10 and GQ&gt;20&quot;




Counting variants: 100% [================] 1,092 15.2/s in 00:01:11
INFO: Adding field num_GD10GQ20
INFO: Adding field hom_GD10GQ20
INFO: Adding field het_GD10GQ20
INFO: Adding field other_GD10GQ20
INFO: Adding field total_GD10GQ20
Updating variant: 100% [======================] 290,999 79.1K/s in 00:00:03
Updating missing variants in variant: 100% [=======================] 89 60.6K/s in 00:00:00
</code></pre>

<p>Compare the statistic from original data vs. statistic conditional on genotype info scores:</p>

<pre><code>vtools output variant chr pos ref alt total total_GD10GQ20 num num_GD10GQ20 missing missing\
_GD10GQ20 -l4 --header




chr     pos     ref     alt     total   total_GD10GQ20  num     num_GD10GQ20    missing missing_GD10GQ20
1       69536   C       T       1092    1088    0       0       0       4
1       861275  C       T       1092    1091    1       1       0       1
1       861315  G       A       1092    1091    2       2       0       1
1       865488  A       G       1092    1092    1       1       0       0
</code></pre>

<p>Looking at the first variant &ldquo;1:69536&rdquo; we see the original total count is 1092 genotype calls; after conditioning on <em>GD&gt;10 and GQ&gt;20</em>, 4 genotype calls are skipped, resulting in 1088 counts. If all genotype calls in a variant site fail the genotype quality constraint, the total count will be zero. These variants are:</p>

<pre><code>vtools select variant &quot;total_GD10GQ20 == 0&quot; -o chr pos ref alt




1       10719967        C       T
1       11199698        T       C
1       11210331        C       T
1       40432550        C       T
1       44057597        C       T
1       44063508        C       T
1       203452763       G       A
1       212506861       C       T
1       212506964       G       A
1       212530649       G       A
...
</code></pre>

<p>A more informative use of the statistic would be for example a ratio of the two total fields:</p>

<pre><code>vtools update variant --set &quot;missing_ratio = missing_GD10GQ20/(total * 1.0)&quot;




chr pos ref alt missing_ratio
1       69536   C       T       0.003663003663
1       861275  C       T       0.000915750915751
1       861315  G       A       0.000915750915751
1       865488  A       G       0.0
1       865545  G       A       0.0
...
</code></pre>

<p>This calculates the proportion of genotypes having <strong>low quality scores per variant</strong>. Such variant level statistics can be useful in creating customized variant level quality control criteria.</p>

<p>The statistics calculated above are for demonstration only and will not be used in our next analysis steps. We will not pursue any QC related summary statistic calculations for now, since groups of samples have to be defined before statistics of subset of samples can be calculated (e.g., population specific minor allele frequency). In the next section we will demonstrate manipulation of sample information.</p>

<h4 id="3-2-processing-phenotype-sample-level-information">3.2 Processing Phenotype / Sample Level Information</h4>

<p>We discuss in this section the organization and use of sample phenotype information. This snapshot provides a phenotype file that we can load phenotype data from:</p>

<pre><code>vtools phenotype --from_file \\(data_dir/g1000_simulated_phenotypes.txt




INFO: Adding field Family_ID
INFO: Adding field Paternal_ID
INFO: Adding field Maternal_ID
INFO: Adding field Gender
INFO: Adding field Population
INFO: Adding field BMI
INFO: Adding field smoking
INFO: 7 field (7 new, 0 existing) phenotypes of 1092 samples are updated.
</code></pre>

<p>To view the phenotypes available,</p>

<pre><code>vtools show samples -l10




sample_name filename    Family_ID   Paternal_ID Maternal_ID Gender  Population  BMI smoking
HG00096 simulatedQc1000g.vcf.gz HG00096 0 0 1 GBR 23.04 0
HG00097 simulatedQc1000g.vcf.gz HG00097 0 0 2 GBR 37.09 0
HG00099 simulatedQc1000g.vcf.gz HG00099 0 0 2 GBR 28.87 0
HG00100 simulatedQc1000g.vcf.gz HG00100 0 0 2 GBR 24.12 1
HG00101 simulatedQc1000g.vcf.gz HG00101 0 0 1 GBR 27.72 0
</code></pre>

<p>To view specific phenotypes,</p>

<pre><code>vtools phenotype --output sample_name Population BMI smoking




HG00096 GBR 23.04   0
HG00097 GBR 37.09   0
HG00099 GBR 28.87   0
HG00100 GBR 24.12   1
HG00101 GBR 27.72   0
HG00102 GBR 27.41   0
</code></pre>

<h5 id="creating-sample-information-columns-based-on-sample-genotypes">Creating sample information columns based on sample genotypes</h5>

<p>Individual level genotype summary can be generated and appended to the sample information table. For example we want to calculate the number of alternative alleles per individual:</p>

<pre><code>vtools phenotype --from_stat &quot;allele_counts=#(alt)&quot; -j8




Calculating phenotype: 100% [==================] 1,092 90.9/s in 00:00:12
INFO: 1092 values of 1 phenotypes (1 new, 0 existing) of 1092 samples are updated.
</code></pre>

<p>Similar to variant level genotype summary, genotype information condition can be applied, for example:</p>

<pre><code>vtools phenotype --from_stat &quot;allele_counts_GD10GQ20=#(alt)&quot; --genotypes 'GD&gt;10 and GQ&gt;20' \
-j8
</code></pre>

<p>Two additional sample information fields <code>allele_counts</code> and <code>allele_counts_ds</code> are added to the database.</p>

<pre><code>vtools phenotype --output sample_name allele_counts allele_counts_GD10GQ20 -l 3 --header




sample_name allele_counts   allele_counts_GD10GQ20
HG00096 1232    1182
HG00097 1317    1263
HG00099 1431    1363
</code></pre>

<div class="notices tip" ><p>Sample information thus created can be useful in individual genotype level quality control. For example, we may want to remove sample having too many low quality or missing genotype calls from the analysis (<code>vtools remove samples</code>). Advanced use of phenotype <code>--from_stat</code> command include calculating sample level <strong>&lsquo;transition-transversion ratio</strong>&rsquo;, or <strong>&lsquo;synonymous-nonsynonymous SNV ratio</strong>&rdquo;, see this [tutorial].</p>
</div>


<h5 id="creating-phenotypes-based-on-existing-sample-information">Creating <em>phenotypes</em> based on existing sample information</h5>

<p><code>vtools phenotype --set</code> accepts arithmetic operations on existing columns, producing new useful sample information, or <em>phenotypes</em>. For example,</p>

<pre><code>vtools phenotype --set &quot;ds_proportion=allele_counts_ds / (allele_counts * 1.0)&quot;
</code></pre>

<p>Or, to create a new phenotype <code>BMI_bin</code> on the basis of existing phenotypes (note the use of <code>--samples</code> condition):</p>

<pre><code>vtools phenotype --set &quot;BMI_bin=1&quot; --samples &quot;BMI&gt;25&quot;
vtools phenotype --set &quot;BMI_bin=0&quot; --samples &quot;BMI&lt;=25&quot;




INFO: 590 values of 1 phenotypes (1 new, 0 existing) of 590 samples are updated.
INFO: 502 values of 1 phenotypes (0 new, 1 existing) of 502 samples are updated.
</code></pre>

<p>In later sections we will demonstrate the association analysis using both the quantitative trait <code>BMI</code> and the binary trait <code>BMI_bin</code>.</p>

<h5 id="updating-sample-information-for-association-testing">Updating sample information for association testing</h5>

<p>There are 14 different population groups in our data-set</p>

<pre><code>vtools phenotype --output Population | sort -u | wc -l # 14
</code></pre>

<p>also there are a few related individuals</p>

<pre><code>vtools phenotype --output sample_name Paternal_ID Maternal_ID \
       --samples &quot;Maternal_ID != 0 or Paternal_ID != 0&quot;




HG00155 0   HG00144
NA07048 NA07034 NA07055
NA10847 NA12146 NA12239
NA10851 NA12056 NA12057
NA19129 NA19128 NA19127
NA19434 0   NA19432
NA19444 0   NA19432
NA19469 0   NA19470
NA19675 NA19679 NA19678
NA19685 NA19661 NA19660
</code></pre>

<p>We choose to group and analyze some unrelated European populations (CEU, BGR and TSI) in association tests. Firstly we define a new phenotype <code>Unrelated_Europeans</code>:</p>

<pre><code>vtools phenotype --set &quot;Unrelated_Europeans='yes'&quot; \
       --samples &quot;Population in ('CEU', 'GBR', 'TSI') and Maternal_ID == 0 and Paternal_ID \
== 0&quot;
vtools phenotype --set &quot;Unrelated_Europeans='no'&quot; \
       --samples &quot;Population not in ('CEU', 'GBR', 'TSI') or Maternal_ID != 0 or Paternal_I\
D != 0&quot;




INFO: Adding field Unrelated_Europeans
INFO: 268 values of 1 phenotypes (1 new, 0 existing) of 268 samples are updated.
INFO: 824 values of 1 phenotypes (0 new, 1 existing) of 824 samples are updated.
</code></pre>

<p>There are 268 samples having a <em>yes</em> tag for being unrelated Europeans. Within these European samples, there are population substructures which we would want to control for by incorporating <strong>principle components</strong> (PC) from MDS analysis as covariates. We calculated the PC for the 268 samples using the <code>KING</code> program, and import the output into the database:</p>

<pre><code>vtools phenotype --from_file \\(data_dir/European_PC.txt




INFO: Adding field PC1
INFO: Adding field PC2
INFO: 3 field (2 new, 1 existing) phenotypes of 268 samples are updated.
</code></pre>

<div class="notices info" ><p>Population structure determination and selection of unrelated samples is very important in exome association analysis. Even though the 1000 genomes data comes with self-reported kinship and population information, it is still necessary to conduct population structure and kinship analysis. One can use the same technique as has been applied in GWAS studies.</p>
</div>


<h4 id="3-3-variant-and-sample-selection-for-association-analysis">3.3 Variant and Sample Selection for Association Analysis</h4>

<h5 id="alternative-allele-frequency-calculations">Alternative allele frequency calculations</h5>

<p>We calculate AF within the 268 European samples, by first calculate total genotypes and number of alternative alleles, then calculate AF from them.</p>

<pre><code>vtools update variant --from_stat 'total_ie=#(GT)' 'num_ie=#(alt)' 'het_ie=#(het)' \
           'hom_ie=#(hom)' 'other_ie=#(other)' --samples &quot;Unrelated_Europeans='yes'&quot; \\

               --genotypes &quot;GD&gt;10 and GQ&gt;20&quot;
vtools update variant --set 'af_ie=num_ie/(total_ie * 2.0)'
</code></pre>

<p>Notice that</p>

<ul>
<li>As was previously discussed, genotype conditions <code>--genotypes</code> can be applied with specified criteria.</li>
<li>Several other fields <code>num_ie, het_ie, hom_ie, other_ie</code> are calculated and will be used in HWE analysis.</li>
</ul>

<div class="notices warning" ><p>Allele frequency thus calculated af=num/(total * 2.0) will be wrong for variants on X chromosome if males present in samples. Correct allele frequency can be calculated using more involved vtools update command, which we will not demonstrate here.</p>
</div>


<p>A new allele frequency field is added to the database. We can look at variants having non-trivial <code>af_ie</code> field and compare it with the allele frequency estimate based on the entire data-set</p>

<pre><code>vtools select variant &quot;af_ie&gt;0&quot; -o chr pos ref alt g1000.AF af_ie -l5




1   874551  G   A   0.0005  0.0018656716417910447
1   874706  A   G   0.0005  0.0018656716417910447
1   878709  C   T   0.0014  0.0018656716417910447
1   878712  C   T   0.0015  0.0018656716417910447
1   878744  G   C   0.0057  0.0018656716417910447
</code></pre>

<p>Note that all the 5 variants displayed above are singletons ($MAF=\frac{1}{268\times2}$). We can explore the variants distribution in the 268 samples using <code>select</code> commands</p>

<p><details><summary> variant counts by type</summary></p>

<pre><code> arr=(\`vtools select variant -c -v0\`) 
 arr+=(\`vtools select variant &quot;af\_ie&gt;0&quot; -c -v0\`) 
 for i in 1 2 3 
 do arr+=(\`vtools select variant &quot;num\_ie=$i&quot; -c -v0\`) 
 done 
 echo \\({arr[1]} &quot;out of a total of&quot; \\({arr[0]} &quot;variants are selected in our sample, among which&quot; \\({arr[2]}          &quot;are singletons,&quot; \\({arr[3]} &quot;are doubletons and&quot; \\({arr[4]} &quot;are tripletons&quot;.
</code></pre>

<p></details></p>

<p>The output says: <strong>60896 out of a total of 291088 variants are selected in our sample, among which 49813 are singletons, 5663 are doubletons and 1546 are tripletons.</strong></p>

<h5 id="hwe-filter">HWE filter</h5>

<p>Exact Test of Hardy-Weinberg Equilibrium is implemented as in Wigginton et al (2005)[^J WIGGINTON, D CUTLER and G ABECASIS (2005) <strong>A Note on Exact Tests of Hardy-Weinberg Equilibrium</strong>. <em>The American Journal of Human Genetics</em> doi:<code>10.1086/429864</code>. <a href="http://linkinghub.elsevier.com/retrieve/pii/S0002929707607356">http://linkinghub.elsevier.com/retrieve/pii/S0002929707607356</a>^] to detect and filter out variants deviating from HWE.</p>

<pre><code>vtools update variant --set &quot;hwe=HWE_exact(num_ie, het_ie, hom_ie, other_ie)&quot;
</code></pre>

<p>An <code>hwe</code> field is created with <em>p-values</em> from the HWE test. We want to filter out variant sites with \(HWE&lt;5\times10^{-8}$, which is 1132 variants. It turns out most of these variants have MAF greater than 0.01.</p>

<pre><code>vtools select variant 'hwe&lt;5e-8' -c
</code></pre>

<h5 id="creating-common-and-rare-variant-subsets">Creating common and rare variant subsets</h5>

<p>Unlike traditional GWAS analysis, exome association analysis will study both common variants (variants having relatively high minor allele frequency) and rare variants (variants having relatively low minor allele frequency). Statistical methods for analyzing common and rare variants are different. It is thus necessary to subset variants based on minor allele frequency. We define common variants as having MAF greater than 1%, and rare variants MAF smaller than 1%.</p>

<pre><code>vtools select variant &quot;af_ie&gt;=0.01 and af_ie&lt;=0.99 and hwe&gt;=5e-8&quot; \
           -t common_var &quot;common variants for the 268 unrelated European samples, hwe \
and quality pass&quot;
vtools select variant &quot;((af_ie&lt;0.01 and af_ie&gt;0.0) or (af_ie&gt;0.99 and af_ie&lt;1.0)) and hwe&gt;=\
5e-8&quot; \
           -t rare_var &quot;rare variants for the 268 unrelated European samples, hwe and \
quality pass&quot;
vtools show tables




table                 #variants     date  message
variant                 291,088
common_var                1,711    Jul10  common variants for the 268 unrelated European samples,
                        hwe and quality pass
rare_var                 58,053    Jul10  rare variants for the 268 unrelated European samples,
                        hwe and quality pass
</code></pre>

<div class="notices info" ><p>The 1% cutoff for rare/common variants is arbitrary. For this small demonstration data-set (60k variants, 268 individuals) using 1% cut off we have 58062 rare variants and 2825 common variants. Alternatively, for single variant association tests (common variants analysis) one could analyze all variants except for singletons and doubletons regardless of their MAF classification; for aggregated variants analysis (rare variants analysis) one may want to use other MAF cutoffs depending on the available sample size, and particularly, for <em>variable thresholds</em> or <em>RareCover</em> methods, the cutoff should be higher (say, 5%).</p>
</div>


<h5 id="selecting-functional-rare-variants">Selecting functional rare variants</h5>

<p>Rare variants analysis typically focus only on variants that effect protein function. We have already annotated the variants with <code>ANNOVAR</code>. The annotation fields can be used to select functional rare variants for association studies.</p>

<pre><code>vtools select rare_var &quot;mut_type like 'nonsynonymous%'&quot; \
           -t rare_nonsyn &quot;nonsynonymous variants selected from table rare_var&quot;
vtools select rare_var &quot;mut_type like 'nonsynonymous%' OR mut_type like 'stoploss%' OR mut_\
type like 'stopgain%' OR mut_type like 'splicing%'&quot; \
           -t rare_fvar &quot;nonsynonymous, stoploss, stopgain and splicing variants selec\
ted from table rare_var&quot;
vtools show tables




rare_nonsyn              23,207    Jul10  nonsynonymous variants selected from table rare_var
rare_fvar                23,757    Jul10  nonsynonymous, stoploss, stopgain and splicing variants
                        selected from table rare_var
</code></pre>

<h5 id="summary">Summary</h5>

<p>We originally have a total of 291,088 variants. After</p>

<ul>
<li>Restricting the variants to only 268 selected samples</li>
<li>Filtering for low variants/genotypes quality</li>
<li>Filtering for HWE</li>
<li>Sub-setting into common/rare variants</li>
<li>Filtering rare variants by functionality</li>
</ul>

<p>we end up having 1,711 common variants and 23,757 rare variants for association analysis.</p>

<h4 id="3-4-additional-qc-for-missing-data">3.4 Additional QC for Missing Data</h4>

<p>Although there is no missing genotype calls in our test data (or, missing genotype calls have already being imputed), in practice it is not always the case. In addition to missing calls, low quality genotypes will be regarded missing in effect. As has already been discussed, we can use <code>--genotypes</code> conditions to skip low quality genotypes. The overall missingness of data per variant can be evaluated on both variant level and individual level.</p>

<h5 id="variant-level-missingness">Variant level missingness</h5>

<pre><code>vtools update variant --from_stat &quot;mnum=#(missing)&quot; --genotypes CONDITION
vtools update variant --set &quot;missing_ratio = mnum/(N * 1.0)&quot;
</code></pre>

<p>where N is the total sample size.</p>

<h5 id="individual-level-missingness">Individual level missingness</h5>

<pre><code>vtools phenotype --from_stat &quot;num=#(missing)&quot; --genotypes CONDITION
vtools phenotype --set &quot;missing_ratio = mnum/(M * 1.0)&quot;
</code></pre>

<p>where M is the total number of variant sites.</p>

<h5 id="group-specific-filtering-of-missing-data-for-rare-variant-association-analysis">Group specific filtering of missing data for rare variant association analysis</h5>

<p>After the variant/sample information field <code>missing_ratio</code> is calculated, an overall variant/sample level filtering can be performed. However we do not recommend the overall filtering for rare variant association analysis. In practice the distribution of missing genotype calls are not uniform. Some samples may not have data on a particular genetic region while other samples have (due to different exome capture arrays). To fully exploit the data we will filter variants and samples for missing data specific to each small genomic unit while carrying out association tests, as will be introduced in the next section of this tutorial.</p>

<p>[11]: 1313  /applications/annotation/introduction/</p>


<footer class=" footline" >
	
</footer>


        
        </div> 
        

      </div>

    <div id="navigation">
        
        
        
        
            
            
                
                    
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                        
                        
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                        
                        
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
                        
            
            
        
                    
            
        
                    
                        
            
            
                
                    
                        
                        
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                        
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
                        
            
            
                
                    
                
                

                    
                    
                    

                    
            
        
                    
            
        
                    
            
        
        
        


        
            <a class="nav nav-prev" href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/supportedtypes/" title="Variants"> <i class="fa fa-chevron-left"></i></a>
        
        
            <a class="nav nav-next" href="https://vatlab.github.io/vat-docs/documentation/help/" title="Help" style="margin-right: 0px;"><i class="fa fa-chevron-right"></i></a>
        
    </div>

    </section>
    
    <div style="left: -1000px; overflow: scroll; position: absolute; top: -1000px; border: none; box-sizing: content-box; height: 200px; margin: 0px; padding: 0px; width: 200px;">
      <div style="border: none; box-sizing: content-box; height: 200px; margin: 0px; padding: 0px; width: 200px;"></div>
    </div>
    <script src="https://vatlab.github.io/vat-docs/js/clipboard.min.js?1529593313"></script>
    <script src="https://vatlab.github.io/vat-docs/js/perfect-scrollbar.min.js?1529593313"></script>
    <script src="https://vatlab.github.io/vat-docs/js/perfect-scrollbar.jquery.min.js?1529593313"></script>
    <script src="https://vatlab.github.io/vat-docs/js/jquery.sticky.js?1529593313"></script>
    <script src="https://vatlab.github.io/vat-docs/js/featherlight.min.js?1529593313"></script>
    <script src="https://vatlab.github.io/vat-docs/js/html5shiv-printshiv.min.js?1529593313"></script>
    <script src="https://vatlab.github.io/vat-docs/js/highlight.pack.js?1529593313"></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script src="https://vatlab.github.io/vat-docs/js/modernizr.custom.71422.js?1529593313"></script>
    <script src="https://vatlab.github.io/vat-docs/js/learn.js?1529593313"></script>
    <script src="https://vatlab.github.io/vat-docs/js/hugo-learn.js?1529593313"></script>

    <link href="https://vatlab.github.io/vat-docs/mermaid/mermaid.css?1529593313" type="text/css" rel="stylesheet" />
    <script src="https://vatlab.github.io/vat-docs/mermaid/mermaid.js?1529593313"></script>
    <script>
        mermaid.initialize({ startOnLoad: true });
    </script>
    

  </body>
</html>

