<!DOCTYPE html>
<html>
  <head>
    <title>vatlab</title>
    
      <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
<meta name="generator" content="Hugo 0.36.1" />
<title>Update :: vatlab</title>
<link rel="shortcut icon" href="https://vatlab.github.io/vat-docs/images/favicon.png" type="image/x-icon" />
<link href="https://vatlab.github.io/vat-docs/css/font-awesome.min.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/css/nucleus.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/theme-flex/style.css" rel="stylesheet">

<link rel="stylesheet" href="https://vatlab.github.io/vat-docs/css/bootstrap.min.css">
<script src="https://vatlab.github.io/vat-docs/js/jquery-2.x.min.js"></script>
<script type="text/javascript">
      var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
</script>
<meta name="description" content="">


    
  </head>
  <body data-url="/documentation/vtools-commands/update/">
    
      <header>
  <div class="logo">
    
	
  
    <a class="baselink" href="https://vatlab.github.io/vat-docs/">vatlab</a>
  

  </div>
  <div class="burger"><a href="javascript:void(0);" style="font-size:15px;">&#9776;</a></div>
</header>
<article>
  <aside>
    <ul class="menu">
          <li data-nav-id="/" class="dd-item">
          <a href="https://vatlab.github.io/vat-docs/">
            <i class="fa fa-fw fa-home"></i>
          </a>
          </li>
    <li data-nav-id="/install/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/install/">Installation</a>
      </div>
    </li>
    <li data-nav-id="/documentation/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/">Documentation</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/keyconcepts/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/">
            Key concepts
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/tutorials/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/tutorials/">
            Tutorials
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/help/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/help/">
            Help
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/vtools-commands/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/">Vtools commands</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools-commands/init/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/init/">
            Init
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/import/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/import/">
            Import
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/update/" class="dd-item active">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/update/">
            Update
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/phenotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/phenotype/">
            Phenotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/liftover/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/liftover/">
            lifttover
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/show/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/show/">
            show
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/use/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/use/">
            use
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/select/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/select/">
            select
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/exclude/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/exclude/">
            exclude
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/compare/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/compare/">
            compare
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/associate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/associate/">
            associate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/output/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/output/">
            output
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/export/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/export/">
            export
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/remove/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/remove/">
            remove
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/execute/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/execute/">
            execute
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/admin/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/admin/">
            admin
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/functions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/functions/">Functions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/functions/ref_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/ref_sequence/">
            ref_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/mut_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/mut_sequence/">
            mut_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/track/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/track/">
            track
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/genotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/genotype/">
            Genotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/samples/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/samples/">
            samples
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/in_table/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/in_table/">
            in_table
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/otherfunctions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/otherfunctions/">
            Other functions
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/vtools_report/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/">Vtools_report</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools_report/avg_depth/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/avg_depth/">
            avg_depth
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/discordance_rate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/discordance_rate/">
            discordance_rate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sam_info/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sam_info/">
            sam_info
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sequence/">
            sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/trans_ratio/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/trans_ratio/">
            trans_ratio
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/variant_stat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/variant_stat/">
            variant_stat
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/inbreeding_coefficient/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/inbreeding_coefficient/">
            inbreeding_coefficient
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/meta_analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/meta_analysis/">
            meta_analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_fields/">
            plot_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_geno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_geno_fields/">
            plot_geno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_pheno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_pheno_fields/">
            plot_pheno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_association/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_association/">
            plot_association
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/customization/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/">Customization</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/format/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/">
            Format
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/">Simulation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/custimzedsimu/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/custimzedsimu/">
            Customized simulations
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/models/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/">Models</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/models/peng1/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/peng1/">
            Peng2014_ex1
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/models/pen2/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/pen2/">
            Peng2014_ex2
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/">Pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/customizedpipeline/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/customizedpipeline/">
            Customized pipeline
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/pipelines/variant-calling/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/">Variant calling</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_hg19/">
            bwa_gatk33_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_b37/">
            bat_gatk33_b37
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_hg19/">
            bwa_gatk28_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_b37/">
            bwa_gatk28_b37
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/other-pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/">Other pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/other-pipelines/import_vcf/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/import_vcf/">
            import_vcf
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/anno_utils/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/anno_utils/">
            anno_utils
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/illumina/">
            Illumnina
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/annovar/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/annovar/">
            ANNOVAR
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/snpeff/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/snpeff/">
            snpEff
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/king/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/king/">
            KING
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/utilities/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/utilities/">Utilities</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/utilities/report_progress/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/report_progress/">
            report_progress
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/utilities/_md5sumd/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/_md5sumd/">
            md5sumd
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/">Application</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
    <li data-nav-id="/applications/annotation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/">Annotation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/tutorial/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/tutorial/">
            Tutorial
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/customized/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">
            Customized annotation databases
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/annotation/genes/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/">Genes</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/genes/ccdsgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/ccdsgene/">
            ccdsGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/hugogene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/hugogene/">
            Hugo gene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/entrezgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/entrezgene/">
            EntrezGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/knowngene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/knowngene/">
            knownGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/refgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/refgene/">
            refGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/cancergenecensus/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/cancergenecensus/">
            CancerGeneCensus
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/variants/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/">Variants</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/variants/cosmic/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/cosmic/">
            COSMIC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbnsfp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbnsfp/">
            sbNSFP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbsnp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbsnp/">
            dbSNP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/linkagephysicalmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/linkagephysicalmap/">
            linkage-physical map
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/databaseof/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/databaseof/">
            Database of Genomic Variants
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/exomesequencingpro/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/exomesequencingpro/">
            Exome Sequencing Project
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/gwas/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/gwas/">
            gwasCatalog
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/hapmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/hapmap/">
            HapMap
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/thousand/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/thousand/">
            Thousand Genomes
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/regions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/">Regions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/regions/low-complexity-regions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/low-complexity-regions/">
            Low complexity regions
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/phast-cons/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/phast-cons/">
            Phast cons
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/genomic-super-dups/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/genomic-super-dups/">
            Genomic super dups
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/cytoband/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/cytoband/">
            cytoBand
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/misc/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/">Misc</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/misc/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/illumina/">
            Illumina
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/keggpathway/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/keggpathway/">
            keggPathway
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/encode/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/encode/">
            ENCODE
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
      <li data-nav-id="/applications/discovery/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/discovery/">
            Discovery
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/">Association</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/data-explorationqc/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/data-explorationqc/">
            Data exploration/QC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/association-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/association-analysis/">
            Association analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/post-assoc-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/post-assoc-analysis/">
            Post-assoc analysis
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/basic-data-statistics/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/">Basic data statistics</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/basic-data-statistics/group-statwrite/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/group-statwrite/">
            Group stat&amp;Write
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-variant-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/">Single variant analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-variant-analysis/fisher-exact-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/fisher-exact-test/">
            Fisher exact test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-gene-association-methods/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/">Single gene association methods</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-gene-association-methods/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/cmc-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/cmc-test/">
            CMC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/c_alpha-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/c_alpha-test/">
            C-alpha test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/kbac-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/kbac-test/">
            KBAC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rbt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rbt-test/">
            RBT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rarecover-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rarecover-test/">
            RareCover test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/vt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/vt-test/">
            VT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/wss-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/wss-test/">
            WSS test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/asum-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/asum-test/">
            aSum test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/jointconditional-association-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/">Joint/Conditional association analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/collapsing-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/collapsing-methods/">
            Collapsing methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/aggregation-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/aggregation-methods/">
            Aggregation methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/weighted-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/weighted-methods/">
            Weighted methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/variable-thresholds/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/variable-thresholds/">
            Variable thresholds
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/skat-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/skat-test/">
            SKAT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/metaskat-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/metaskat-analysis/">
            MetaSKAT analysis
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/vat-stacking/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/vat-stacking/">VAT stacking</a>
      </div>
    </li>
    <li data-nav-id="/applications/association/create-your-test/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/">Create your test!</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/create-your-test/running-r-programs/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/running-r-programs/">
            Running R programs
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/create-your-test/programming-in-vat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/programming-in-vat/">
            Programming in VAT
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/view-association-result/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/">View association result</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/view-association-result/qq-plot-and-manhattan-plot/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/qq-plot-and-manhattan-plot/">
            QQ plot and Manhattan plot
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/development/">Development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/development/chagelog/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/chagelog/">
            ChangeLog
          </a>
        </div>
    </li>
      <li data-nav-id="/development/getinvolved/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/getinvolved/">
            Get Involved
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/administration/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/administration/">Administration</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/administration/allrecent/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/allrecent/">
            All recent changes
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/autolink/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/autolink/">
            Autolink
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/stylesheet/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/stylesheet/">
            Style sheet
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/trafficstat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/trafficstat/">
            Traffic statistics
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/under-development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/under-development/">Under development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/under-development/faq/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/under-development/faq/">
            FAQ
          </a>
        </div>
    </li>
        </ul>
    </li>




    </ul>
    <section>
    </section>
  </aside>
  <section class="page">
    
    <div class="nav-select">
      <center>Navigation : 
        <select onchange="javascript:location.href = this.value;">
          
    <option value="/install/" >
   Installation</option>
    <option value="/documentation/" >
   Documentation</option> 
      <option value="/documentation/keyconcepts/" >- Key concepts</option>
      <option value="/documentation/tutorials/" >- Tutorials</option>
      <option value="/documentation/help/" >- Help</option>
    <option value="/documentation/vtools-commands/" >
  - 
   Vtools commands</option> 
      <option value="/documentation/vtools-commands/init/" >-- Init</option>
      <option value="/documentation/vtools-commands/import/" >-- Import</option>
      <option value="/documentation/vtools-commands/update/"  selected>-- Update</option>
      <option value="/documentation/vtools-commands/phenotype/" >-- Phenotype</option>
      <option value="/documentation/vtools-commands/liftover/" >-- lifttover</option>
      <option value="/documentation/vtools-commands/show/" >-- show</option>
      <option value="/documentation/vtools-commands/use/" >-- use</option>
      <option value="/documentation/vtools-commands/select/" >-- select</option>
      <option value="/documentation/vtools-commands/exclude/" >-- exclude</option>
      <option value="/documentation/vtools-commands/compare/" >-- compare</option>
      <option value="/documentation/vtools-commands/associate/" >-- associate</option>
      <option value="/documentation/vtools-commands/output/" >-- output</option>
      <option value="/documentation/vtools-commands/export/" >-- export</option>
      <option value="/documentation/vtools-commands/remove/" >-- remove</option>
      <option value="/documentation/vtools-commands/execute/" >-- execute</option>
      <option value="/documentation/vtools-commands/admin/" >-- admin</option>
  
    <option value="/documentation/functions/" >
  - 
   Functions</option>
    <option value="/documentation/vtools_report/" >
  - 
   Vtools_report</option>
    <option value="/documentation/customization/" >
  - 
   Customization</option>
    <option value="/documentation/pipelines/" >
  - 
   Pipelines</option>
    <option value="/documentation/utilities/" >
  - 
   Utilities</option>
  
    <option value="/applications/" >
   Application</option>
    <option value="/development/" >
   Development</option>
    <option value="/administration/" >
   Administration</option>
    <option value="/under-development/" >
   Under development</option>



        </select>
      </center>
    </div>
      <div>
        <div class="searchbox">
          <input data-search-input id="search-by" type="text" placeholder="Search...">
        </div>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/lunr.min.js"></script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/auto-complete.js"></script>
        <link href="https://vatlab.github.io/vat-docs/css/auto-complete.css" rel="stylesheet">
        <script type="text/javascript">
          
              var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
          
        </script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/search.js"></script>
      </div>
    

    <h1>Update</h1>
    
    
    
    

<h1 id="add-new-or-update-existing-variant-and-genotype-info-fields">Add new or update existing variant and genotype info fields</h1>

<h2 id="usage">Usage</h2>

<pre><code>% vtools update -h


usage: vtools update [-h] [--from_file FROM_FILE [FROM_FILE ...]]
                     [--build BUILD] [--format FORMAT] [-j N]
                     [--sample_name [SAMPLE_NAME [SAMPLE_NAME ...]]]
                     [--set [EXPR [EXPR ...]]] [--from_stat [EXPR [EXPR ...]]]
                     [-s [COND [COND ...]]] [--genotypes [COND [COND ...]]]
                     [-v {0,1,2}]
                     table

Add or update fields of existing variants and genotype from other fields,
statistics of genotypes and genotype info, or files that annotate variants or
their locations (e.g. Read annotation from ANNOVAR output files, import
additional variant or genotype fields from .vcf files).

positional arguments:
  table                 variants to be updated.

optional arguments:
  -h, --help            show this help message and exit
  -v {0,1,2}, --verbosity {0,1,2}
                        Output error and warning (0), info (1) and debug (2)
                        information to standard output (default to 1).

Update from files:
  --from_file FROM_FILE [FROM_FILE ...]
                        A list of files that will be used to add or update
                        existing fields of variants. The file should be
                        delimiter separated with format described by parameter
                        --format. Gzipped files are acceptable. If input files
                        contains genotype information, have been inputted
                        before, and can be linked to the samples they created
                        without ambiguity (e.g. single sample, or samples with
                        detectable sample names), genotypes and their
                        information will also be updated.
  --build BUILD         Build version of the reference genome (e.g. hg18) of
                        the input files, which should be the primary (used by
                        default) or alternative (if available) reference
                        genome of the project. An alternative reference genome
                        will be added to the project if needed.
  --format FORMAT       Format of the input text file. It can be one of the
                        variant tools supported file types such as
                        ANNOVAR_output (cf. 'vtools show formats'), or a local
                        format specification file (with extension .fmt). Some
                        formats accept parameters (cf. 'vtools show format
                        FMT') and allow you to update additional or
                        alternative fields from the input file.
  -j N, --jobs N        Number of processes to import input file. Variant
                        tools by default uses a single process for reading and
                        writing, and can use one or more dedicated reader
                        processes (jobs=2 or more) to process input files. Due
                        to the overhead of inter-process communication, more
                        jobs do not automatically lead to better performance.
  --sample_name [SAMPLE_NAME [SAMPLE_NAME ...]]
                        Name of the samples to be updated by the input files.
                        If unspecified, headers of the genotype columns of the
                        last comment line (line starts with #) of the input
                        files will be used (and thus allow different sample
                        names for input files). Sample names will be used to
                        identify samples to be updated. Filename will be used
                        to uniquely identify a sample if mutliple samples with
                        the same name exist in the project. No genotype info
                        will be updated if samples cannot be unquely
                        determined.

Set value from existing fields:
  --set [EXPR [EXPR ...]]
                        Add a new field or updating an existing field using a
                        constant (e.g. mark=1) or an expression using other
                        fields (e.g. freq=num/120, refgene=refGene.name). If
                        multiple values are returned for a variant, only one
                        of them will be used. Parameter samples could be used
                        to limit the affected variants. In addition, special
                        function are provided, including 'HWE_exact' (exact
                        test of Hardy-Weinberg Equilibrium) and 'Fisher_exact'
                        (Fisher's exact test for case/ctrl association).

Set fields from sample statistics:
  --from_stat [EXPR [EXPR ...]]
                        One or more expressions such as meanQT=avg(QT) that
                        aggregate genotype info (e.g. QT) of variants in all
                        or selected samples to specified fields (e.g. meanQT).
                        Functions sum, avg, max, and min are currently
                        supported. In addition, special functions #(GT),
                        #(hom), #(het), #(alt), #(other), #(missing), #(wtGT),
                        #(mutGT), and maf(), are provided to count the number
                        of valid genotypes (not missing), homozygote
                        genotypes, heterozygote genotypes, alternative alleles
                        (#(het) + 2*#(hom) + #(other)), genotypes with two
                        different alternative alleles, missing genotypes
                        (number of samples - #(GT)), number of non-missing
                        wildtype genotypes (#(GT) - #(hom) - #(het) -
                        #(other)), number of non-wildtype genotypes (#(hom) +
                        #(het) + #(other)), and minor allele frequency. The
                        maf() function treats chromosomes 1 to 22 as
                        autosomes, X and Y as sex chromosomes, and other
                        chromosomes as single-copy manifolds. It requires a
                        phenotype named sex or gender that codes male/female
                        by 1/2, M/F or Male/Female if maf of variants on sex
                        chromosomes are calculated. This function by default
                        calculates allele frequency among existing-alleles,
                        but will treat all missing values as wild type alleles
                        if runtime option treat_missing_as_wildtype is set to
                        true.
  -s [COND [COND ...]], --samples [COND [COND ...]]
                        Limiting variants from samples that match conditions
                        that use columns shown in command 'vtools show sample'
                        (e.g. 'aff=1', 'filename like &quot;MG%&quot;').
  --genotypes [COND [COND ...]]
                        Limiting variants from samples that match conditions
                        that use columns shown in command 'vtools show
                        genotypes' (e.g. 'GQ&gt;15').
</code></pre>

<h2 id="details">Details</h2>

<p>Command <code>`vtools update</code> updates <strong>variant info fields</strong> (and to a lesser extend <strong>genotype info fields</strong>) by adding more fields or updating values at existing fields. It does not add any new variant or genotype, and does not change existing variant, sample, or genotype. Using three parameters <code>--from_file</code>, <code>--from_stat</code>, and <code>--set</code>, variant information fields could be updated from external file, sample genotypes, and existing fields.</p>

<p>An illustration about <code>`vtools update</code> ▸</p>

<p>Attach:update.png</p>

<h3 id="import-variant-info-from-external-files-in-standard-formats-from-file">Import variant info from external files in standard formats (<code>--from_file</code>)</h3>

<p>Option <code>--from_file</code> allows command <code>vtools update</code> to update variant and/or genotype info fields by reading info from an external file. This process is similar to <code>vtools import</code> but has two major differences:</p>

<ul>
<li><code>`vtools update</code> does not import any new variant even if the external file has more variants.</li>
<li><code>`vtools update</code> can add or update fields from <strong>position, range or field based input files</strong>. That is to say, you could update fields from input files that describes variants at particular locations or chromosomsal regions. Variants belonging to the same location or region will share the updated info.</li>
</ul>

<p>The variant or genotype fields that will be added or updated depends on the file format (cf. <code>vtools show formats</code>). File formats accept parameters to specify what fields to import. For example, parameters <code>--var_info</code> and <code>--geno_info</code> can be used with <a href="http://www.1000genomes.org/node/101" target="_blank">format vcf</a> to specify which variant and genotype info fields to be updated from a <code>vcf</code> file.</p>

<p>(:toggleexample Examples: create a project:)</p>

<p>Let us create a directory update and import an empty project with a few test vcf files <code>V1.vcf</code>, <code>V2.vcf</code> and <code>V3.vcf</code>,</p>

<pre><code>% mkdir update
% cd update
% vtools init update
% vtools admin --load_snapshot vt_testData
</code></pre>

<p>The project does not have any variant so we import some from these VCF files:</p>

<pre><code>% vtools import V*.vcf --build hg18


INFO: Importing variants from V1.vcf (1/3)
V1.vcf: 100% [========================================================] 1,000 10.9K/s in 00:00:00
INFO: 989 new variants (989 SNVs) from 1,000 lines are imported.
INFO: Importing variants from V2.vcf (2/3)
V2.vcf: 100% [========================================================] 1,000 11.7K/s in 00:00:00
INFO: 352 new variants (352 SNVs) from 995 lines are imported.
INFO: Importing variants from V3.vcf (3/3)
V3.vcf: 100% [========================================================] 1,000 13.2K/s in 00:00:00
INFO: 270 new variants (270 SNVs) from 1,000 lines are imported.
Importing genotypes: 100% [============================================] 2,995 1.5K/s in 00:00:02
Copying genotype: 100% [==================================================] 3 11.2K/s in 00:00:00
INFO: 1,611 new variants (1,611 SNVs) from 2,995 lines (3 samples) are imported.
</code></pre>

<p>(:exampleend:)</p>

<p>(:toggleexample Examples: import additional fields from source files:) As we can see from the output of <code>vtools show fields</code>, this project does not have any variant info field,</p>

<pre><code>% vtools show fields


variant.chr
variant.pos
variant.ref
variant.alt
</code></pre>

<p>Because the input file has two variant info fields <code>DP</code> (depth of coverage) and <code>NS</code> (number in sample), we can add these two fields to our project using commands</p>

<pre><code>% vtools update variant --from_file V*.vcf --var_info DP NS


INFO: Using primary reference genome hg18 of the project.
Getting existing variants: 100% [====================================] 1,611 170.4K/s in 00:00:00
INFO: Updating variants from V1.vcf (1/3)
V1.vcf: 100% [=========================================================] 1,000 7.0K/s in 00:00:00
INFO: Fields DP, NS of 989 variants are updated
INFO: Updating variants from V2.vcf (2/3)
V2.vcf: 100% [=========================================================] 1,000 7.4K/s in 00:00:00
INFO: Fields DP, NS of 990 variants are updated
INFO: Updating variants from V3.vcf (3/3)
V3.vcf: 100% [=========================================================] 1,000 6.8K/s in 00:00:00
INFO: Fields DP, NS of 988 variants are updated
INFO: Fields DP, NS of 2,967 variants are updated
</code></pre>

<p>The project now has two variant info fields <code>DP</code> and <code>NS</code>,</p>

<pre><code>% vtools show fields 


variant.chr
variant.pos
variant.ref
variant.alt
variant.DP
variant.NS
</code></pre>

<p>and we can output them with variants</p>

<pre><code>% vtools output variant chr pos ref alt DP NS -l 5


1   4540    G   A   9   1
1   5683    G   T   7   1
1   5966    T   G   24  1
1   6241    T   C   23  1
1   9992    C   T   13  1
</code></pre>

<p>(:exampleend:)</p>

<p>(:toggleexample Examples: add genotype info field:) We have just added two variant info fields <code>DP</code> and <code>NS</code> from input vcf files, but what we did probably does not make any sense. This is because <code>DP</code> and <code>NS</code> describes variants in each sample so the same variants from different samples will have different depth, and sample count should probaby be added together to reflect the total sample count. That is to say, <code>DP</code> here should better be considered as genotype field instead of variant field.</p>

<p>Variant tools is flexible enough to handle such a situation (check <code>vtools show format vcf</code> for details). By specifying field <code>DP</code> in parameter <code>--geno_info</code>, this field will be added as genotype info field and be processed for each sample:</p>

<pre><code>% vtools update variant --from_file V*.vcf --geno_info DP


INFO: Using primary reference genome hg18 of the project.
Getting existing variants: 100% [====================================] 1,611 176.1K/s in 00:00:00
INFO: Updating variants from V1.vcf (1/3)
V1.vcf: 100% [=========================================================] 1,000 7.6K/s in 00:00:00
INFO: Fields  of 0 variants and geno fields of 989 genotypes are updated
INFO: Updating variants from V2.vcf (2/3)
V2.vcf: 100% [=========================================================] 1,000 7.7K/s in 00:00:00
INFO: Fields  of 0 variants and geno fields of 990 genotypes are updated
INFO: Updating variants from V3.vcf (3/3)
V3.vcf: 100% [=========================================================] 1,000 7.5K/s in 00:00:00
INFO: Fields  of 0 variants and geno fields of 988 genotypes are updated
INFO: Fields  of 0 variants and geno fields of 2,967 genotypes are updated
</code></pre>

<p>The project now has a genotype info field <code>DP</code>, with coverage depth for each genotype.</p>

<pre><code>% vtools show genotypes


sample_name filename    num_genotypes   sample_genotype_fields
SAMP1   V1.vcf  989 GT,DP
SAMP1   V2.vcf  990 GT,DP
SAMP1   V3.vcf  988 GT,DP
</code></pre>

<p>Variant tools depends on filename to determine which sample to update from each input file so <strong>the original input files must be used to update genotype info fields</strong>.</p>

<p>(:exampleend:)</p>

<p>(:toggleexample Examples: adding annotation info from ANNOVAR output:)</p>

<p>Suppose that we would like to use ANNOVAR to annotate our variants. The basic step is to export the variants in ANNOVAR input format and call ANNOVAR.</p>

<pre><code>% vtools export variant --format ANNOVAR &gt; ann.in


vtools export variant --format ANNOVAR &gt; ann.in
Writing: 100% [=======================================================] 1,611 45.1K/s in 00:00:00
INFO: 1611 lines are exported from variant table variant 




% annotate_variation.pl ann.in humandb/


NOTICE: The --geneanno operation is set to ON by default
NOTICE: The --buildver is set as 'hg18' by default
NOTICE: Reading gene annotation from humandb/hg18_refGene.txt ... Done with 41481 transcripts (including 7217 without coding sequence annotation) for 23669 unique genes
NOTICE: Reading FASTA sequences from humandb/hg18_refGeneMrna.fa ... Done with 14 sequences
WARNING: A total of 318 sequences will be ignored due to lack of correct ORF annotation
NOTICE: Finished gene-based annotation on 1611 genetic variants in ann.in
NOTICE: Output files were written to ann.in.variant_function, ann.in.exonic_variant_function
</code></pre>

<p>We then want to update our project with ANNOVAR annotations. Because gene annotation of ANNOVAR generates two output files, two output file formats are available. One of the formats is <code>ANNOVAR_exonic_variant_function</code>,</p>

<pre><code>% vtools show format ANNOVAR_exonic_variant_function


Format:      ANNOVAR_exonic_variant_function
Description: Output from ANNOVAR, generated from command
  &quot;path/to/annovar/annotate_variation.pl annovar.txt
  path/to/annovar/humandb/&quot;. This format imports chr, pos, ref, alt
  and ANNOVAR annotations. For details please refer to
  http://www.openbioinformatics.org/annovar/annovar_gene.html

Columns:
  None defined, cannot export to this format

variant:
  chr          Chromosome
  pos          1-based position, hg18
  ref          Reference allele, '-' for insertion.
  alt          Alternative allele, '-' for deletion.

Variant info:
  mut_type     the functional consequences of the variant.

Other fields (usable through parameters):
  genename     Gene name (for the first exon if the variant is in more than one
               exons, but usually the names for all exons are the
               same).
  function     the gene name, the transcript identifier and the sequence change in
               the corresponding transcript

Format parameters:
  var_info     Fields to be outputted, can be one or both of mut_type and function.
               (default: mut_type)
</code></pre>

<p>As you can see, this is a variant based format. It has a default variant info field <code>mut_type</code>, but you can add more fields using parameter <code>var_info</code>. To update variants,</p>

<pre><code>% vtools update variant --format ANNOVAR_exonic_variant_function --from_file ann.in.exonic_variant_function


INFO: Using primary reference genome hg18 of the project.
Getting existing variants: 100% [===============================================================] 1,611 154.0K/s in 00:00:00
INFO: Updating variants from ann.in.exonic_variant_function (1/1)
ann.in.exonic_variant_function: 100% [===============================================================] 23 2.5K/s in 00:00:00
INFO: Field mut_type of 23 variants are updated
</code></pre>

<p>By default, field <code>mut_type</code> is added.</p>

<pre><code>% vtools show fields


variant.chr
variant.pos
variant.ref
variant.alt
variant.DP
variant.NS
variant.mut_type
</code></pre>

<p>To update additional variant/genotype fields, use <code>--var_info</code> of the <code>ANNOVAR_exonic_variant_function</code> format:</p>

<pre><code>% vtools update variant --format ANNOVAR_exonic_variant_function \
                       --from_file ann.in.exonic_variant_function \
                       --var_info mut_type function 


INFO: Using primary reference genome hg18 of the project.
Getting existing variants: 100% [===============================================================] 1,611 180.6K/s in 00:00:00
INFO: Updating variants from ann.in.exonic_variant_function (1/1)
ann.in.exonic_variant_function: 100% [===============================================================] 23 1.8K/s in 00:00:00
INFO: Fields mut_type, function of 23 variants are updated
</code></pre>

<p>Fields specified by option <code>--var_info</code> are added. Now we have one more field <code>function</code></p>

<pre><code>% vtools show fields


variant.chr
variant.pos
variant.ref
variant.alt
variant.DP
variant.NS
variant.mut_type
variant.function
</code></pre>

<p>(:exampleend:)</p>

<h3 id="import-variant-info-from-external-files-in-customized-formats-from-file">Import variant info from external files in customized formats (<code>--from_file</code>)</h3>

<p>During the analysis of variants, it is common to have annotations in different formats from various sources (e.g. from annotation servers such as <a href="http://regulome.stanford.edu/" target="_blank">regulome DB</a>). Although variant tools has a growing number of <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.HomePage?action=edit" target="_blank">formats</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.HomePage?action=edit" target="_blank">?</a>, it is sometimes needed to create your own format files to import annotations.</p>

<p>(:toggleexample Examples: Update annotation for variants presented by rsnames:) Let us first create some data with rsname. Because this sample project uses hg18, we cannot use the default <code>dbSNP</code> database and has to use</p>

<pre><code>% vtools use dbSNP-hg18_130


INFO: Downloading annotation database from annoDB/dbSNP-hg18_130.ann
INFO: Downloading annotation database from http://vtools.houstonbioinformatics.org/annoDB/dbSNP-hg18_130.DB.gz
dbSNP-hg18_130.DB.gz: 100% [================================================] 617,049,496.0 1.3M/s in 00:08:07
INFO: Decompressing /Users/bpeng/.variant_tools/annoDB/dbSNP-hg18_130.DB.gz
INFO: Using annotation DB dbSNP in project update.
INFO: dbSNP version 130
</code></pre>

<p>Let us find some variants with rsname and output it</p>

<pre><code>% vtools select variant 'dbSNP.func = &quot;intron&quot;' --output dbSNP.name dbSNP.func &gt; dbSNP.info
</code></pre>

<p>The file looks like:</p>

<pre><code>% head -10 dbSNP.info




rs3131969   intron
rs3131968   intron
rs3131967   intron
rs3115859   intron
rs3131966   intron
rs3131964   intron
rs3115858   intron
rs12567639  intron
rs3131963   intron
rs3115853   intron
</code></pre>

<p>To update your variants with this information, you will need a format file that knows how to get <code>chr</code>, <code>pos</code>, <code>ref</code> and <code>alt</code> from <code>rsname</code>. Fortunately, the <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.Map?action=edit" target="_blank">map</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.Map?action=edit" target="_blank">?</a> format provides such an example so we can adapt it to the following format:</p>

<pre><code>[format description]
description=a format that determins chr, pos, ref, alt from rsname using a dbSNP database
variant=chr,pos,ref,alt
variant_info=func

[DEFAULT]
db_file=dbSNP.DB

#
# NOTE: Functor FieldFromDB(db, res, fields) matches input(s) from
# columns 'index' (column 1 in this example), compare it with field
# 'fields' (name), and return results 'res' (chr, start, refNCBI,
# alt). You can get a list of fields by 'vtools use dbSNP-hg18_130',
# and run 'vtools show fields'
#
[chr]
index=1
type=VARCHAR(20)
adj=FieldFromDB(&quot;%(db_file)s&quot;, &quot;chr&quot;, &quot;name&quot;)

[pos]
index=1
adj=FieldFromDB(&quot;%(db_file)s&quot;, &quot;start&quot;, &quot;name&quot;)
type=INTEGER NOT NULL

[ref]
index=1
adj=FieldFromDB(&quot;%(db_file)s&quot;, &quot;refNCBI&quot;, &quot;name&quot;)
type=VARCHAR(255)

[alt]
index=1
type=VARCHAR(255)
adj=FieldFromDB(&quot;%(db_file)s&quot;, &quot;alt&quot;, &quot;name&quot;)

[func]
index=2
type=VARCHAR(255)
</code></pre>

<p>We can then use this format (named <code>rsname.fmt</code>) to import variant info <code>func</code> to the project:</p>

<pre><code>% vtools update variant --format rsname --from_file dbSNP.info --db_file dbSNP-hg18_130.DB


INFO: Using primary reference genome hg18 of the project.
Getting existing variants: 100% [====================================] 1,611 229.7K/s in 00:00:00
INFO: Updating variants from dbSNP.info (1/1)
dbSNP.info: 100% [=======================================================] 209 3.0K/s in 00:00:00
INFO: Field func of 209 variants are updated
</code></pre>

<p>The project now has another field <code>variant.func</code> (to differentiate from <code>dbSNP.func</code>),</p>

<pre><code>% select variant 'dbSNP.func = &quot;intron&quot;' --output chr pos ref alt dbSNP.name variant.func  -l 10


1   744045  A   G   rs3131969   intron
1   744055  A   G   rs3131968   intron
1   744197  T   C   rs3131967   intron
1   744366  G   A   rs3115859   intron
1   744827  C   T   rs3131966   intron
1   745750  C   G   rs3131964   intron
1   745753  A   T   rs3115858   intron
1   746131  G   A   rs12567639  intron
1   746243  T   A   rs3131963   intron
1   747503  G   A   rs3115853   intron
</code></pre>

<p>(:exampleend:)</p>

<p>(:toggleexample Examples: Update annotation from position-based data:)</p>

<p>Now, suppose we have another file with only <code>chr</code> and <code>pos</code> information:</p>

<pre><code>% vtools output variant chr pos dbSNP.molType dbSNP.class &gt; moltype.txt
</code></pre>

<p>This file looks like</p>

<pre><code>% head -10 moltype.txt


1   4540    NA  NA
1   5683    genomic single
1   5966    NA  NA
1   6241    genomic single
1   6241    genomic single
1   9992    genomic single
1   9993    genomic single
1   10007   NA  NA
1   10098   genomic single
1   10098   genomic single
</code></pre>

<p>Because this file does not have <code>ref</code> and <code>alt</code> information, we will have to treat it as position-based. That is to say, if there are multiple variants at a chromosomal location, they will updated with the same information. The format to use, is therefore</p>

<pre><code>[format description]
description=A position based format to update variant information
position=chr,pos
variant_info=molType, molClass

[chr]
index=1
type=VARCHAR(20)

[pos]
index=2
type=INTEGER NOT NULL

[molType]
index=3
type=VARCHAR(255)

[molClass]
index=4
type=VARCHAR(255)
</code></pre>

<p>With <code>position.fmt</code> saved in the project directory, the following command can be used to update variants with additional fields <code>molType</code> and <code>molClas</code>:</p>

<pre><code>% vtools update variant --format position --from_file moltype.txt


INFO: Using primary reference genome hg18 of the project.
Getting existing variants: 100% [====================================] 1,611 226.1K/s in 00:00:00
INFO: Updating variants from moltype.txt (1/1)
moltype.txt: 100% [====================================================] 1,679 8.3K/s in 00:00:00
INFO: Fields molType, molClass of 1,689 variants are updated




% vtools output variant chr pos ref alt variant.molType molClass -l 10


1   4540    G   A   NA  NA
1   5683    G   T   genomic single
1   5966    T   G   NA  NA
1   6241    T   C   genomic single
1   9992    C   T   genomic single
1   9993    G   A   genomic single
1   10007   G   A   NA  NA
1   10098   G   A   genomic single
1   14775   G   A   genomic single
1   16862   A   G   genomic single
</code></pre>

<p>(:exampleend:)</p>

<h3 id="calculate-genotype-statistics-for-each-variant-from-stat">Calculate genotype statistics for each variant (<code>--from_stat</code>)</h3>

<p>Option <code>--from_stat</code> adds fields to variant tables with summary statistics from sample genotypes. This parameter accepts expressions in the format of <code>VAR_INFO=FUNC(GENO_INFO)</code> where <code>VAR_INFO</code> is the variant information field to be added or updated, <code>FUNC</code> is a function or expression, and <code>GENO_INFO</code> is genotype or genotype info. If you are only interested in genotype statistics, some specially defined <code>FUNC(GENO_INFO)</code> can be used. A list of acceptable aggregating functions is available <a href="http://www.sqlite.org/lang_aggfunc.html" target="_blank">here</a>.</p>

<p>(:toggleexample Examples: statistics of genotype info fields:) Each genotype in our samples has a field <code>DP</code>, which is the coverage depth. To assess the quality of a variant, it is useful to calculate average depth across all samples as a variant info field. The aggregating function to use is <code>avg</code>:</p>

<pre><code>% vtools update variant --from_stat 'avg_DP=avg(DP)'


Counting variants: 100% [==================================================] 3 110.9/s in 00:00:00
INFO: Adding field avg_DP
Updating variant: 100% [===============================================] 1,611 49.2K/s in 00:00:00
INFO: 1610 records are updated
</code></pre>

<p>Depth of genotypes across three samples are averaged and save to a new variant info field <code>avg_DP</code>. It is different from the <code>DP</code> we saved before because that <code>DP</code> is the depth of one of the genotypes:</p>

<pre><code>% vtools output variant chr pos ref alt DP avg_DP -l5


1   4540    G   A   9   7.5
1   5683    G   T   7   7.0
1   5966    T   G   24  25.3333333333
1   6241    T   C   23  20.5
1   9992    C   T   13  12.0
</code></pre>

<p>(:exampleend:)</p>

<p>A locus has a wildtype allele and can have one or more alternative alleles. Assuming a wildtype allele 0, an alternative allele 1, and an additional alternative allele 2 if exists, a sample can have genotype 0/0, 0/1, 0/2, <sup>1</sup>&frasl;<sub>1</sub>, <sup>2</sup>&frasl;<sub>2</sub>, and <sup>1</sup>&frasl;<sub>2</sub> at a locus on an autosome. <code>vtools update --from_stat</code> counts such genotype for all (or selected) variants using the following special functions:</p>

<ul>
<li><code>#(GT)</code>: the total number of genotypes</li>
<li><code>#(alt)</code>: number of alternative alleles (non-zero alleles 1 or 2)</li>
<li><code>#(hom)</code>: number of homozygote (genotype <sup>1</sup>&frasl;<sub>1</sub> or <sup>2</sup>&frasl;<sub>2</sub>)</li>
<li><code>#(het)</code>: number of heterozygote (genotype 0/1 or 0/2)</li>
<li><code>#(other)</code>: number of heterozygote with two alternative alleles (genotype <sup>1</sup>&frasl;<sub>2</sub>).</li>
<li><code>#(wtGT)</code>: number of wildtype genotypes (<code>#(GT) - #(hom) - #(het) - #(other)</code>)</li>
<li><code>#(mutGT)</code>: number of non-wildtype genotypes (<code>#(hom) + #(het) + #(other)</code>)</li>
<li><code>#(missing)</code>: number of missing genotypes (<code>sample size - #(GT)</code>)</li>
<li><code>maf()</code>: minor allele frequency. The frequency is the smaller of #(alt) / #all_alleles and 1 - #(alt) / #all_alleles. The total number of alleles is determined by type of chromosome, sex of samples, and how to treat missing values. More specifically, if runtime option <code>treat_missing_as_wildtype</code> is set to <code>False</code> (default), <code>#all_alleles</code> is

<ul>
<li><code>2 * #(GT)</code> for variants on autosomes</li>
<li><code># GT for females + #GT for males</code> for variants on chromosome X</li>
<li><code># GT for males</code> for variants on chromosome Y, and</li>
<li><code>#(GT)</code> for variants on other chromosomes.</li>
</ul></li>
</ul>

<p>If <code>treat_missing_as_wildtype</code> is set to true, all missing values are counted as wildtype allele so <code>#all_alleles</code> is
*   <code>2 * number of samples</code> for variants on autosomes
*   <code>2 * number of males + number of females</code> for variants on chromosome X
*   <code>number of males</code> for variants on chromosome Y
*   <code>number of samples</code> for varants on other chromosomes</p>

<p>Note that genotype <sup>1</sup>&frasl;<sub>1</sub> an <sup>2</sup>&frasl;<sub>2</sub> are counted for different variants (0,1) and (0,2), and <code>#(other)</code> is counted for both variants. Therefore, <code>#(alt)</code> equals <code>#(hom)</code> X 2 + <code>#(het)</code> + <code>#(other)</code>*2.</p>

<p>(:toggleexample Examples: sample statistics:) Using these special functions, we can calculate many sample statistics as various variant info fields:</p>

<pre><code>% vtools update variant --from_stat 'total=#(GT)' 'num=#(alt)' 'hom=#(hom)' 'het=#(het)' 'other=#(other)'


Counting variants: 100% [==================================================] 3 238.8/s in 00:00:00
INFO: Adding field num
INFO: Adding field hom
INFO: Adding field het
INFO: Adding field other
INFO: Adding field total
Updating variant: 100% [===============================================] 1,611 30.9K/s in 00:00:00
INFO: 1610 records are updated
</code></pre>

<p>As you can see, genotypes are not available in all samples and the first five variants are all appear as heterozygotes.</p>

<pre><code>% vtools output variant chr pos ref alt total num hom het other -l5


1   4540    G   A   2   2   0   2   0
1   5683    G   T   1   1   0   1   0
1   5966    T   G   3   3   0   3   0
1   6241    T   C   2   2   0   2   0
1   9992    C   T   2   2   0   2   0
</code></pre>

<p>You can use the <code>maf</code> function to calculate minor allele frequency</p>

<pre><code>% vtools update variant --from_stat 'maf=maf()'


INFO: 0, 0, and 0 variants on X, Y and other non-autosome chromosomes are identifield
Counting variants: 100% [============================================] 3 316.5/s in 00:00:00
INFO: Adding variant info field maf with type FLOAT
Updating variant: 100% [=========================================] 1,611 45.8K/s in 00:00:00
INFO: 1611 records are updated
</code></pre>

<p>This case is easy because all variants are on an autosome.</p>

<pre><code>% vtools output variant chr pos ref alt total num hom het maf -l5


1   4540    G   A   2   2   0   2   0.5
1   5683    G   T   1   1   0   1   0.5
1   5966    T   G   3   3   0   3   0.5
1   6241    T   C   2   2   0   2   0.5
1   9992    C   T   2   2   0   2   0.5
</code></pre>

<p>However, because this dataset does not record wildtype alleles, the minior allele frequencies are <code>0.5</code> if all genotypes are heterozygotes. To calculate minor allele frequency for all samples, you should set runtime option <code>treat_missing_as_wildtype</code> as <code>true</code>,</p>

<pre><code>% vtools admin --set_runtime_option treat_missing_as_wildtype=true


INFO: Option treat_missing_as_wildtype is set to True
</code></pre>

<p>In this case, missing genotypes are counted,</p>

<pre><code>$ vtools update variant --from_stat 'maf1=maf()'


Updating variant: 100% [============================] 1,611 5.1K/s in 00:00:00
INFO: 1611 records are updated
</code></pre>

<p>and the results (in field <code>maf1</code>) are number of aleternative alleles devided by <code>6</code>.</p>

<pre><code>% vtools output variant chr pos ref alt num &quot;2*total&quot; maf maf1 -l 5


1   4540    G   A   2   4   0.5 0.333333333333
1   5683    G   T   1   2   0.5 0.166666666667
1   5966    T   G   3   6   0.5 0.5
1   6241    T   C   2   4   0.5 0.333333333333
1   9992    C   T   2   4   0.5 0.333333333333
</code></pre>

<p>(:exampleend:)</p>

<p>All statistics are by default calculated for all samples in the project. But we can limit calculations on subset of samples by <code>--samples</code> and/or <code>--genotypes</code>.</p>

<ul>
<li><p>Option <code>--samples &quot;[condition]&quot;</code> can be used to limit the statistics to a subset of samples, for example, all affected individuals. <code>[condition]</code> should be an SQL expression using one or more columns shown in <code>vtools show samples</code>. For example,</p>

<ul>
<li><code>aff='1'</code> select all samples with affection status equals to 1.</li>
<li><code>some_measure&gt;0.95</code> select all samples with some_measure greater than 0.95.</li>
<li><code>filename like 'MG%'</code> select all samples from files with filename starts with &ldquo;MG&rdquo;.</li>
<li><code>sample_name like 'CEU%'</code> select all samples with sample name starts with &ldquo;CEU&rdquo;.
<br /></li>
</ul></li>

<li><p>Option <code>--genotypes &quot;[condition]&quot;</code> can be used to limit the statistics to a subset of samples having genotypes satisfying a <code>[condition]</code> defined by an SQL expression using one or more columns shown in <code>vtools show genotypes</code>. For example,</p>

<ul>
<li><code>GT=0</code> select only the wildtype genotypes</li>
<li><code>GQ&gt;20</code> select variants having genotype quality greater than 20</li>
</ul></li>
</ul>

<p>(:toggleexample Examples: statistics for subsets of samples and genotypes:) This project has three samples:</p>

<pre><code>% vtools show samples


sample_name filename
SAMP1   V1.vcf
SAMP1   V2.vcf
SAMP1   V3.vcf
</code></pre>

<p>Suppose two samples from files <code>V1.vcf</code> and <code>V2.vcf</code> are affected and we would like to calculate genotype count for only these two samples, we could use parameter <code>--samples</code> to limit our calculation:</p>

<pre><code>% vtools update variant --from_stat &quot;cases_het=#(het)&quot; --samples &quot;filename in ('V1.vcf', 'V2.vcf')&quot;


INFO: 2 samples are selected
Counting variants: 100% [==================================================] 2 233.5/s in 00:00:00
INFO: Adding field cases_het
Updating variant: 100% [===============================================] 1,341 35.3K/s in 00:00:00
INFO: 1340 records are updated
</code></pre>

<p>We can also limit the statistics to genotypes that satisfy certain conditions (e.g. with high coverage):</p>

<pre><code>% vtools update variant --from_stat &quot;cases_het_highDP=#(het)&quot; --samples &quot;filename in ('V1.vcf', 'V2.vcf')&quot; --genotypes 'DP&gt;15'


INFO: 2 samples are selected
Counting variants: 100% [===================================================] 2 80.5/s in 00:00:00
INFO: Adding field cases_het_highDP
Updating variant: 100% [=================================================] 621 42.8K/s in 00:00:00
INFO: 620 records are updated




% vtools output variant chr pos ref alt cases_het cases_het_highDP -l5


1   4540    G   A   1   0
1   5683    G   T   1   0
1   5966    T   G   2   1
1   6241    T   C   1   1
1   9992    C   T   1   0
</code></pre>

<p>(:exampleend:)</p>

<p>(:toggleexample Examples: update variant info of subsets of variants:) If you are interested only in variants in a variant table, you could also only update statistics for variants in specified variant tables. For example, we can select all variants that belong to all three samples and creates a table <code>in_all</code>:</p>

<pre><code>% vtools select variant 'total=3' -t in_all


Running: 1 71.8/s in 00:00:00                                                                      
INFO: 511 variants selected.
</code></pre>

<p>Then we can add a field <code>case_hom</code> to count the number of homozygotes for only these variants:</p>

<pre><code>% vtools update in_all --from_stat 'case_hom=#(hom)' --samples  &quot;filename in ('V1.vcf', 'V2.vcf')&quot;


INFO: 2 samples are selected
Counting variants: 100% [===================================================] 2 50.8/s in 00:00:00
INFO: Adding field case_hom
Updating in_all: 100% [==================================================] 511 67.1K/s in 00:00:00
INFO: 510 records are updated




% vtools output in_all chr pos ref alt case_hom -l 5


1   5966    T   G   0
1   10007   G   A   0
1   20723   G   C   0
1   20786   G   T   2
1   31705   A   G   2
</code></pre>

<p>(:exampleend:)</p>

<h3 id="add-fields-based-on-other-variant-or-annotation-fields-set">Add fields based on other variant or annotation fields (<code>--set</code>)</h3>

<p>Option <code>--set</code> evaluates expressions from existing variant info fields and assign results to a new or existing variant info field. For example, once you have calculated allele count, you could calculate allele frequency based on sample size using expression <code>--set 'maf=m/(n*2.0)'</code> where <em>m</em> is field name calculated as &ldquo;m=#(alt)&rdquo; and <em>n</em> is sample size, which might not be <code>&quot;n=#(GT)&quot;</code> if there are missing data. Note that the denominator should be <code>n*2.0</code>, not <code>n*2</code>, because SQL requires the denominator be a FLOAT type, not INTEGER.</p>

<p>(:toggleexample Examples: set fields from variant info fields:) We can try to calculate the allele frequency using number of aleternative alleles and number of genotypes as follows:</p>

<pre><code>% vtools update variant --set &quot;maf=num/(total*2.0)&quot;


INFO: Adding field maf
</code></pre>

<p>However, because samples from this project is called individually and wildtype alleles are not recorded, there are a lot of missing data. If we assume missing data as non-recorded wildtype alleles, we should add missing data to the denominator of the expression. To do this, we need to first calculate a field for missing genotypes:</p>

<pre><code>% vtools update variant --from_stat 'missing=#(missing)'


Counting variants: 100% [=====================================================================] 3 289.1/s in 00:00:00
INFO: Resetting values at existing field missing
Updating variant: 100% [==================================================================] 1,611 49.5K/s in 00:00:00
INFO: 1610 records are updated
</code></pre>

<p>And then calculate the real allele frequency</p>

<pre><code>% vtools update variant --set &quot;real_maf=num/((total+missing)*2.0)&quot;


INFO: Adding field real_maf
</code></pre>

<p>Two allele frequencies are different</p>

<pre><code>% vtools output variant chr pos ref alt num total missing maf real_maf -l5


1   4540    G   A   2   2   1   0.5 0.333333333333
1   5683    G   T   1   1   2   0.5 0.166666666667
1   5966    T   G   3   3   0   0.5 0.5
1   6241    T   C   2   2   1   0.5 0.333333333333
1   9992    C   T   2   2   1   0.5 0.333333333333
</code></pre>

<p>(:exampleend:)</p>

<p>In addition to variant info fields, annotation fields could also be used in these expressions and set variant info fields. There is usually no such need to copy annotation fields to variant info fields though. Moreoever, because one variant might have more than one annotation value for an annotation field (e.g. a variant might belong to two isoforms of a gene), copying annotation fields to variant info fields might loss information.</p>

<p>(:toggleexample Examples: set variant info field from annotation fields:)</p>

<pre><code>% vtools use refGene-hg18_20110909
% vtools update variant --set refgene=refGene.name


INFO: Adding field refgene
Updating variant: 100% [=================================================================] 11,984 52.6K/s in 00:00:00
</code></pre>

<p>We can select variants that belong to a gene and output it</p>

<pre><code>% vtools select variant 'refGene is not NULL' -t in_gene


Running: 0 0.0/s in 00:00:00                                                                                          
INFO: 144 variants selected.




% vtools output in_gene chr pos ref alt refGene -l 5


1   887188  G   C   NM_198317
1   887427  T   C   NM_198317
1   889872  C   G   NM_198317
1   890115  G   A   NM_198317
1   890593  G   A   NM_198317
</code></pre>

<p>(:exampleend:)</p>


    
    
        <div class="chevrons">
    <div id="navigation">
<a class="nav nav-prev" href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/import/" title="Import"> <i class="fa fa-chevron-left"></i><label>Import</label></a>
    <a class="nav nav-next" href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/phenotype/" title="Phenotype" style="margin-right: 0px;"><label>Phenotype</label><i class="fa fa-chevron-right"></i></a></div>
  </div>

  </section>
</article>

<footer>

<div class="footline">
    

    

  

    <div class="github-link">
      <a href="Documentation/vtools%20Commands/update.md" target="blank"><i class="fa fa-code-fork"></i>
        Improve this page</a>
    </div>
  </div>


	<div>


  
    Create a content/_footer.md file to customize the footer content
  



	</div>
</footer>

<script src="https://vatlab.github.io/vat-docs/js/clipboard.min.js"></script>

<link href="https://vatlab.github.io/vat-docs/css/featherlight.min.css" rel="stylesheet">
<script src="https://vatlab.github.io/vat-docs/js/featherlight.min.js"></script>



<script src="https://vatlab.github.io/vat-docs/theme-flex/script.js"></script>

    

    
    

    
  </body>
</html>