<!DOCTYPE html>
<html>
  <head>
    <title>vatlab</title>
    
      <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
<meta name="generator" content="Hugo 0.36.1" />
<title>track :: vatlab</title>
<link rel="shortcut icon" href="https://vatlab.github.io/vat-docs/images/favicon.png" type="image/x-icon" />
<link href="https://vatlab.github.io/vat-docs/css/font-awesome.min.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/css/nucleus.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/theme-flex/style.css" rel="stylesheet">

<link rel="stylesheet" href="https://vatlab.github.io/vat-docs/css/bootstrap.min.css">
<script src="https://vatlab.github.io/vat-docs/js/jquery-2.x.min.js"></script>
<script type="text/javascript">
      var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
</script>
<meta name="description" content="">


    
  </head>
  <body data-url="/documentation/functions/track/">
    
      <header>
  <div class="logo">
    
	
  
    <a class="baselink" href="https://vatlab.github.io/vat-docs/">vatlab</a>
  

  </div>
  <div class="burger"><a href="javascript:void(0);" style="font-size:15px;">&#9776;</a></div>
</header>
<article>
  <aside>
    <ul class="menu">
          <li data-nav-id="/" class="dd-item">
          <a href="https://vatlab.github.io/vat-docs/">
            <i class="fa fa-fw fa-home"></i>
          </a>
          </li>
    <li data-nav-id="/install/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/install/">Installation</a>
      </div>
    </li>
    <li data-nav-id="/documentation/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/">Documentation</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/keyconcepts/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/">
            Key concepts
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/tutorials/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/tutorials/">
            Tutorials
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/help/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/help/">
            Help
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/vtools-commands/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/">Vtools commands</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools-commands/init/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/init/">
            Init
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/import/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/import/">
            Import
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/update/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/update/">
            Update
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/phenotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/phenotype/">
            Phenotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/liftover/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/liftover/">
            lifttover
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/show/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/show/">
            show
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/use/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/use/">
            use
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/select/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/select/">
            select
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/exclude/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/exclude/">
            exclude
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/compare/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/compare/">
            compare
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/associate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/associate/">
            associate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/output/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/output/">
            output
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/export/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/export/">
            export
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/remove/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/remove/">
            remove
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/execute/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/execute/">
            execute
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/admin/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/admin/">
            admin
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/functions/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/functions/">Functions</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/functions/ref_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/ref_sequence/">
            ref_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/mut_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/mut_sequence/">
            mut_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/track/" class="dd-item active">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/track/">
            track
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/genotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/genotype/">
            Genotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/samples/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/samples/">
            samples
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/in_table/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/in_table/">
            in_table
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/otherfunctions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/otherfunctions/">
            Other functions
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/vtools_report/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/">Vtools_report</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools_report/avg_depth/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/avg_depth/">
            avg_depth
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/discordance_rate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/discordance_rate/">
            discordance_rate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sam_info/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sam_info/">
            sam_info
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sequence/">
            sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/trans_ratio/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/trans_ratio/">
            trans_ratio
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/variant_stat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/variant_stat/">
            variant_stat
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/inbreeding_coefficient/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/inbreeding_coefficient/">
            inbreeding_coefficient
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/meta_analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/meta_analysis/">
            meta_analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_fields/">
            plot_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_geno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_geno_fields/">
            plot_geno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_pheno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_pheno_fields/">
            plot_pheno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_association/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_association/">
            plot_association
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/customization/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/">Customization</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/format/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/">
            Format
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/">Simulation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/custimzedsimu/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/custimzedsimu/">
            Customized simulations
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/models/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/">Models</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/models/peng1/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/peng1/">
            Peng2014_ex1
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/models/pen2/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/pen2/">
            Peng2014_ex2
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/">Pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/customizedpipeline/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/customizedpipeline/">
            Customized pipeline
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/pipelines/variant-calling/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/">Variant calling</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_hg19/">
            bwa_gatk33_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_b37/">
            bat_gatk33_b37
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_hg19/">
            bwa_gatk28_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_b37/">
            bwa_gatk28_b37
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/other-pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/">Other pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/other-pipelines/import_vcf/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/import_vcf/">
            import_vcf
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/anno_utils/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/anno_utils/">
            anno_utils
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/illumina/">
            Illumnina
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/annovar/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/annovar/">
            ANNOVAR
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/snpeff/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/snpeff/">
            snpEff
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/king/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/king/">
            KING
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/utilities/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/utilities/">Utilities</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/utilities/report_progress/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/report_progress/">
            report_progress
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/utilities/_md5sumd/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/_md5sumd/">
            md5sumd
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/">Application</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
    <li data-nav-id="/applications/annotation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/">Annotation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/tutorial/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/tutorial/">
            Tutorial
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/customized/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">
            Customized annotation databases
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/annotation/genes/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/">Genes</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/genes/ccdsgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/ccdsgene/">
            ccdsGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/hugogene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/hugogene/">
            Hugo gene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/entrezgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/entrezgene/">
            EntrezGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/knowngene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/knowngene/">
            knownGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/refgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/refgene/">
            refGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/cancergenecensus/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/cancergenecensus/">
            CancerGeneCensus
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/variants/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/">Variants</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/variants/cosmic/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/cosmic/">
            COSMIC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbnsfp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbnsfp/">
            sbNSFP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbsnp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbsnp/">
            dbSNP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/linkagephysicalmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/linkagephysicalmap/">
            linkage-physical map
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/databaseof/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/databaseof/">
            Database of Genomic Variants
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/exomesequencingpro/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/exomesequencingpro/">
            Exome Sequencing Project
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/gwas/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/gwas/">
            gwasCatalog
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/hapmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/hapmap/">
            HapMap
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/thousand/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/thousand/">
            Thousand Genomes
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/regions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/">Regions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/regions/low-complexity-regions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/low-complexity-regions/">
            Low complexity regions
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/phast-cons/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/phast-cons/">
            Phast cons
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/genomic-super-dups/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/genomic-super-dups/">
            Genomic super dups
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/cytoband/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/cytoband/">
            cytoBand
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/misc/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/">Misc</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/misc/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/illumina/">
            Illumina
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/keggpathway/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/keggpathway/">
            keggPathway
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/encode/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/encode/">
            ENCODE
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
      <li data-nav-id="/applications/discovery/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/discovery/">
            Discovery
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/">Association</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/data-explorationqc/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/data-explorationqc/">
            Data exploration/QC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/association-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/association-analysis/">
            Association analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/post-assoc-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/post-assoc-analysis/">
            Post-assoc analysis
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/basic-data-statistics/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/">Basic data statistics</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/basic-data-statistics/group-statwrite/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/group-statwrite/">
            Group stat&amp;Write
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-variant-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/">Single variant analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-variant-analysis/fisher-exact-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/fisher-exact-test/">
            Fisher exact test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-gene-association-methods/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/">Single gene association methods</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-gene-association-methods/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/cmc-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/cmc-test/">
            CMC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/c_alpha-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/c_alpha-test/">
            C-alpha test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/kbac-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/kbac-test/">
            KBAC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rbt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rbt-test/">
            RBT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rarecover-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rarecover-test/">
            RareCover test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/vt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/vt-test/">
            VT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/wss-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/wss-test/">
            WSS test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/asum-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/asum-test/">
            aSum test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/jointconditional-association-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/">Joint/Conditional association analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/collapsing-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/collapsing-methods/">
            Collapsing methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/aggregation-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/aggregation-methods/">
            Aggregation methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/weighted-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/weighted-methods/">
            Weighted methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/variable-thresholds/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/variable-thresholds/">
            Variable thresholds
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/skat-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/skat-test/">
            SKAT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/metaskat-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/metaskat-analysis/">
            MetaSKAT analysis
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/vat-stacking/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/vat-stacking/">VAT stacking</a>
      </div>
    </li>
    <li data-nav-id="/applications/association/create-your-test/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/">Create your test!</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/create-your-test/running-r-programs/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/running-r-programs/">
            Running R programs
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/create-your-test/programming-in-vat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/programming-in-vat/">
            Programming in VAT
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/view-association-result/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/">View association result</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/view-association-result/qq-plot-and-manhattan-plot/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/qq-plot-and-manhattan-plot/">
            QQ plot and Manhattan plot
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/development/">Development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/development/chagelog/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/chagelog/">
            ChangeLog
          </a>
        </div>
    </li>
      <li data-nav-id="/development/getinvolved/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/getinvolved/">
            Get Involved
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/administration/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/administration/">Administration</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/administration/allrecent/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/allrecent/">
            All recent changes
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/autolink/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/autolink/">
            Autolink
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/stylesheet/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/stylesheet/">
            Style sheet
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/trafficstat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/trafficstat/">
            Traffic statistics
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/under-development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/under-development/">Under development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/under-development/faq/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/under-development/faq/">
            FAQ
          </a>
        </div>
    </li>
        </ul>
    </li>




    </ul>
    <section>
    </section>
  </aside>
  <section class="page">
    
    <div class="nav-select">
      <center>Navigation : 
        <select onchange="javascript:location.href = this.value;">
          
    <option value="/install/" >
   Installation</option>
    <option value="/documentation/" >
   Documentation</option> 
      <option value="/documentation/keyconcepts/" >- Key concepts</option>
      <option value="/documentation/tutorials/" >- Tutorials</option>
      <option value="/documentation/help/" >- Help</option>
    <option value="/documentation/vtools-commands/" >
  - 
   Vtools commands</option>
    <option value="/documentation/functions/" >
  - 
   Functions</option> 
      <option value="/documentation/functions/ref_sequence/" >-- ref_sequence</option>
      <option value="/documentation/functions/mut_sequence/" >-- mut_sequence</option>
      <option value="/documentation/functions/track/"  selected>-- track</option>
      <option value="/documentation/functions/genotype/" >-- Genotype</option>
      <option value="/documentation/functions/samples/" >-- samples</option>
      <option value="/documentation/functions/in_table/" >-- in_table</option>
      <option value="/documentation/functions/otherfunctions/" >-- Other functions</option>
  
    <option value="/documentation/vtools_report/" >
  - 
   Vtools_report</option>
    <option value="/documentation/customization/" >
  - 
   Customization</option>
    <option value="/documentation/pipelines/" >
  - 
   Pipelines</option>
    <option value="/documentation/utilities/" >
  - 
   Utilities</option>
  
    <option value="/applications/" >
   Application</option>
    <option value="/development/" >
   Development</option>
    <option value="/administration/" >
   Administration</option>
    <option value="/under-development/" >
   Under development</option>



        </select>
      </center>
    </div>
      <div>
        <div class="searchbox">
          <input data-search-input id="search-by" type="text" placeholder="Search...">
        </div>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/lunr.min.js"></script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/auto-complete.js"></script>
        <link href="https://vatlab.github.io/vat-docs/css/auto-complete.css" rel="stylesheet">
        <script type="text/javascript">
          
              var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
          
        </script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/search.js"></script>
      </div>
    

    <h1>track</h1>
    
    
    
    

<h1 id="extract-annotation-from-external-files">Extract annotation from external files</h1>

<h2 id="usage">Usage</h2>

<p>Function <code>track(filename, field)</code> returns annotation information at column <code>col</code> (optional) in file <code>filename</code>, at position (<code>chr</code>, <code>pos</code> at primary or alternative reference genome) of each variant. For example, function</p>

<pre><code>% vtools output variant chr pos ref alt &quot;track('1000g.vcf.gz', 'info')&quot;
</code></pre>

<p>single quote (<code>'</code>) should be used for string literals in SQL functions. Double quote (<code>&quot;</code>) should be avoided although it sometimes works.</p>

<p>output the &ldquo;info&rdquo; column of file <code>1000g.vcf.gz</code>, for variants at location <code>chr</code> and <code>pos</code>.</p>

<p>This function currently accepts four types of track files:</p>

<ol>
<li>Local or online <code>bgzipped</code> and <code>tabix</code>-indexed vcf files with extension <code>.vcf.gz</code>.</li>
<li><a href="http://genome.ucsc.edu/goldenPath/help/bigWig.html" target="_blank"><code>bigWig</code></a> files that provides dense and continuous data for a region.</li>
<li><a href="http://genome.ucsc.edu/goldenPath/help/bigBed.html" target="_blank"><code>bigBed</code></a> files with 3 required columns and 9 optional columns. Annotation tracks in BED format can be converted to this format using program <code>bedToBigBed</code>.</li>
<li>Local or online indexed BAM files with extension <code>.bam</code>. This format does not contain &lsquo;columns&rsquo; as other formats do but it provides alignment information for the variants.</li>
</ol>

<p>The allowed and default values of the second option <code>field</code> vary for different file formats. Generally speaking, numeric values such as <code>1</code>, <code>2</code>, &hellip; returns the <code>col</code>-th column of the input data file, with the exception that `<code>returns</code>1<code>(match or not) for all matched records. String values such as</code>&ldquo;chrom&rdquo;<code>,</code>&ldquo;chromStart&rdquo;<code>,</code>&ldquo;info&rdquo;` return values at specified column as strings.</p>

<h2 id="details">Details</h2>

<h3 id="display-information-about-tracks">Display information about tracks</h3>

<p>Before you use each track, it is important to run command</p>

<pre><code>% vtools show track FilenameOrURL
</code></pre>

<p>to get the detailed information about the track file, and the available fields and their types.</p>

<ol>
<li>As a shortcut to enter <code>track</code> function for multiple files, you can use wildcast characters in the first parameter (<code>filename</code>) of the <code>track</code> function. This will result in multiple <code>track()</code> function calls for each matching filename. For example, if you have <code>A01.BAM</code> and <code>A02.BAM</code> in the current directory, function <code>track('A*.BAM', 'calls')</code> is equivalent to <code>track('A01.BAM', 'calls') track('A02.BAM', 'calls')</code>.</li>
<li>The return values of the returned field will be numeric if the column contains numeric data (e.g. flag, score, position). <strong>Only the first record will be returned if a variant matches multiple records in the track file</strong>. If an option <code>all=1</code> is passed to field (e.g. <code>track('my.bam', 'info?all=1')</code>), the <code>track</code> function will output all matching records as string, separated by a delimiter <code>|</code>.</li>
<li>This function automatically chooses correct chromosome name (adding <code>chr</code> to chromosome name if needed), and position (adjust to 0-based position if needed) to match records in the track file.</li>
<li>The return values are not adjusted. That is to say, columns such as <code>pos</code> will be 0-based for 0-based track files (e.g. bigBed files), and 1-based for 1-based track files (e.g. vcf).</li>
</ol>

<p>Please refer to <a href="http://varianttools.sourceforge.net/Vtools/Show#toc10" target="_blank">here</a> for more details of command vtools show, especially a brief description of the BAM header.</p>

<h3 id="tabixed-vcf-tracks">Tabixed vcf tracks</h3>

<p>VCF files that can be used as tracks must be bgzipped and tabix-indexed. Regular vcf files can be converted to this format using commands <code>bgzip my.vcf</code> and <code>tabix -p vcf my.vcf.gz</code>. Parameter <code>col</code> for this format can be <code>1</code> (chrom), <code>2</code> (start, 1-based), <code>3</code> (name), <code>4</code> (ref), <code>5</code> (alt alleles), <code>6</code> (qual), <code>7</code> (filter), <code>8</code> (info), <code>9</code> (format string), <code>10</code> and more (for genotype columns for sample <code>col-9</code>); names of the columns <code>&quot;chrom&quot;</code>, <code>&quot;pos&quot;</code>, <code>&quot;name&quot;</code>, <code>&quot;ref&quot;</code>, <code>&quot;alt&quot;</code>, <code>&quot;qual&quot;</code>, <code>&quot;filter&quot;</code>, <code>&quot;info&quot;</code>, <code>&quot;format&quot;</code>; name of information fields available in the vcf file in the format of <code>info.FIELD</code>; name of samples for genotype columns, and name of genotype info fields in the format of <code>SAMPLE.FIELD</code>. If no <code>col</code> is specified, a default value <code>8</code> is passed to display the full <code>INFO</code> column of the vcf file.</p>

<p>(:toggleexample Examples: Annotate variants using vcf tracks:) Let us get some test data, and index the vcf file using the <code>tabix</code> program</p>

<pre><code>% vtools init track
% vtools admin --load_snapshot vt_testData
% tabix -p vcf CEU.vcf.gz
% vtools import CEU.vcf.gz --build hg18


INFO: Importing variants from CEU.vcf.gz (1/1)
CEU.vcf.gz: 100% [============================================] 300 13.9K/s in 00:00:00
INFO: 288 new variants (288 SNVs) from 300 lines are imported.
Importing genotypes: 100% [=================================] 18,000 9.0K/s in 00:00:02
Copying samples: 100% [=========================================] 75 74.9/s in 00:00:01
</code></pre>

<p>The track information can be displayed using command</p>

<pre><code>% vtools show track CEU.vcf.gz | head -30


Version                 VCF v4.0
Number of fields:       69

Header: (exclude INFO and FORMAT lines)
                        ##reference=human_b36_both.fasta
                        ##rsIDs=dbSNP b129 mapped to NCBI 36.3, August 10, 2009

Available columns (with type VARCHAR if unspecified or all=1):
0 (INTEGER)             1 if matched
chr (1, chrom)          chromosome
pos (2, INTEGER)        position (1-based)
name (3)                name of variant
ref (4)                 reference allele
alt (5)                 alternative alleles
qual (6)                qual
filter (7)              filter
info (8, default)       variant info fields
info.DP (INTEGER)       Total Depth
info.HM2 (INTEGER, flag) HapMap2 membership
info.HM3 (INTEGER, flag) HapMap3 membership
info.AA                 Ancestral Allele, ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/pilot_data/technical/reference/ancestral_alignments/README
info.AC (INTEGER)       Allele count in genotypes
info.AN (INTEGER)       Total number of alleles in called genotypes
format (9)              genotype format
NA06985 (10)            genotype for sample NA06985
NA06985.GT              Genotype for sample NA06985
NA06985.DP (INTEGER)    Read Depth for sample NA06985
NA06985.CB              Called by S(Sanger), M(UMich), B(BI) for sample NA06985
NA06986 (11)            genotype for sample NA06986
NA06986.GT              Genotype for sample NA06986
</code></pre>

<p>We can use the <code>track</code> function to display the info column in the original vcf file,</p>

<pre><code>% vtools output variant chr pos &quot;track('CEU.vcf.gz')&quot; -l 5


1   533 AA=.;AC=6;AN=120;DP=423
1   41342   AA=.;AC=29;AN=120;DP=188
1   41791   AA=.;AC=5;AN=120;DP=192
1   44449   AA=C;AC=2;AN=120;DP=166
1   44539   AA=T;AC=2;AN=120;DP=131
</code></pre>

<p>The default parameter <code>col=8</code> is used to extract the info column of the info file. You can display other tracks such as name</p>

<pre><code>% vtools output variant chr pos &quot;track('CEU.vcf.gz', 'name')&quot; -l 5


1   533 .
1   41342   .
1   41791   .
1   44449   .
1   44539   rs2462492
</code></pre>

<p>Values of individual info fields could be specified by <code>info.FIELD</code> where <code>FIELD</code> is the name of info field.</p>

<pre><code>% vtools output variant chr pos &quot;track('CEU.vcf.gz', 'info.DP')&quot; -l 5


1   533 423
1   41342   188
1   41791   192
1   44449   166
1   44539   131
</code></pre>

<p>If you know the name of the sample (in the vcf file, this example happens to has samples from this file),</p>

<pre><code>% vtools show samples -l 5


sample_name filename
NA06985 CEU.vcf.gz
NA06986 CEU.vcf.gz
NA06994 CEU.vcf.gz
NA07000 CEU.vcf.gz
NA07037 CEU.vcf.gz
(55 records omitted)
</code></pre>

<p>you can get the genotype columns using sample name</p>

<pre><code>% vtools output variant chr pos &quot;track('CEU.vcf.gz', 'NA06986')&quot; -l 5


1   533 0|0:14:SMB
1   41342   0|1:16:SMB
1   41791   0|0:7:SM
1   44449   0|0:6:SM
1   44539   0|0:12:SM
</code></pre>

<p>With the format information abtained from</p>

<pre><code>% vtools output variant chr pos &quot;track('CEU.vcf.gz', 'format')&quot; -l 5


1   533 GT:DP:CB
1   41342   GT:DP:CB
1   41791   GT:DP:CB
1   44449   GT:DP:CB
1   44539   GT:DP:CB
</code></pre>

<p>we can list fields of the genotype columns,</p>

<pre><code>% vtools output variant chr pos &quot;track('CEU.vcf.gz', 'NA06986.GT')&quot; -l 5


1   533 0|0
1   41342   0|1
1   41791   0|0
1   44449   0|0
1   44539   0|0
</code></pre>

<p>(:exampleend:)</p>

<p>A very useful feature of the vcf track is that <strong>you can use vcf files from online</strong> by specifying a URL instead of a local filename.</p>

<p>(:toggleexample Examples: Annotate variants using online vcf files:) We would like to annotate our variants using VCF files from the 1000 genomes project. However, our project uses build <code>hg18</code> of the reference genome and the 1000 genomes project uses <code>hg19</code>. To make use of data from the 1000 genomes project, we need to first lift over our project:</p>

<pre><code>% vtools liftover hg19


INFO: Exporting variants in BED format
Exporting variants: 100% [==========================] 288 67.3K/s in 00:00:00
INFO: Running UCSC liftOver tool
Updating table variant: 100% [======================] 288 21.8K/s in 00:00:00
</code></pre>

<p>To pass the correct coordinates, option <code>--build hg19</code> is needed:</p>

<pre><code>% vtools output variant chr pos &quot;track('http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20110521/ALL.chr1.phase1_release_v3.20101123.snps_indels_svs.genotypes.vcf.gz', 'info')&quot; \
%     -l 5 --build hg19


1   10533   .
1   51479   RSQ=0.7414;AVGPOST=0.9085;AA=T;AN=2184;THETA=0.0131;AC=235;VT=SNP;LDAF=0.1404;SNPSOURCE=LOWCOV;ERATE=0.0012;AF=0.11;ASN_AF=0.0035;AMR_AF=0.16;AFR_AF=0.03;EUR_AF=0.22
1   51928   .
1   54586   .
1   54676   LDAF=0.1528;RSQ=0.6989;AA=T;AN=2184;AC=267;VT=SNP;AVGPOST=0.8998;SNPSOURCE=LOWCOV;THETA=0.0110;ERATE=0.0037;AF=0.12;ASN_AF=0.02;AMR_AF=0.20;AFR_AF=0.09;EUR_AF=0.18
</code></pre>

<p>(:exampleend:)</p>

<p>Available variant and genotype info fields are determined from the header of input vcf file. Columns such as <code>info.AA</code> is unacceptable if <code>AA</code> is not defined in the header.</p>

<h3 id="bigwig-tracks">bigWig tracks</h3>

<p>The bigWig tracks contains numeric values for locations (ranges). The default <code>col</code> value for this format is <code>4</code> (the value column), but you can specify <code>1</code> (chrom), <code>2</code> (start, 0-based), <code>3</code> (end, 1-based), <code>4</code> (value), and <code>&quot;chrom&quot;</code>, <code>&quot;chromStart&quot;</code>, <code>&quot;chromEnd&quot;</code>, and <code>&quot;value&quot;</code>.</p>

<p>(:toggleexample Examples: Use bigWig tracks to annotate and select variants:) Let us create a project in hg19, import some data, and download a bigWig track from the UCSC ENCODE website:</p>

<pre><code>% wget http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeOpenChromFaire/wgEncodeOpenChromFaireFrontalcortexocSig.bigWig
% vtools init track -f
% vtools import indels.vcf  --build hg19


INFO: Importing variants from indels.vcf (1/1)
indels.vcf: 100% [============================================] 184 12.3K/s in 00:00:00
INFO: 137 new variants (1 SNVs, 77 insertions, 58 deletions, 7 complex variants) from 184 lines are imported.
Importing genotypes: 0 0.0/s in 00:00:00
Copying samples: 0 0.0/s in 00:00:00
</code></pre>

<p>The detailed information about this track can be obtained by</p>

<pre><code>% vtools show track wgEncodeOpenChromFaireFrontalcortexocSig.bigWig


Version:                4
Primary data size       1320909131
Zoom levels:            10
Chrom count:            23
Chrom size:
    chr1                249250621
    chr10               135534747
    chr11               135006516
    chr12               133851895
    chr13               115169878
    chr14               107349540
    chr15               102531392
    chr16               90354753
    chr17               81195210
    chr18               78077248
    chr19               59128983
    chr2                243199373
    chr20               63025520
    chr21               48129895
    chr22               51304566
    chr3                198022430
    chr4                191154276
    chr5                180915260
    chr6                171115067
    chr7                159138663
    chr8                146364022
    chr9                141213431
    chrX                155270560
Bases covered:          2951253637
Mean:                   0.004807
Min:                    0.000000
Max:                    0.592400
std:                    0.004469
Number of fields:       4

Available columns (with type VARCHAR if unspecified or all=1):
0 (INTEGER)             1 if matched
chrom (1)               chromosome
chromStart (2, INTEGER) start position (0-based)
chromEnd (3, INTEGER)   end position (1-based)
value (4, FLOAT)        value
</code></pre>

<p>and we can show the track values for each variant using command</p>

<pre><code>% vtools output variant chr pos ref alt &quot;track('wgEncodeOpenChromFaireFrontalcortexocSig.bigWig')&quot; -l 5


1   10434   -   C   0.00089999998454
1   10440   C   -   0.00089999998454
1   54789   C   -   0.00719999987632
1   54790   -   T   0.00719999987632
1   63738   ACT -   0.00710000004619
</code></pre>

<p>In addition to output, the track can also be used to select variants,</p>

<pre><code>% vtools select variant &quot;track('wgEncodeOpenChromFaireFrontalcortexocSig.bigWig') &gt; 0.001&quot; \
     --output chr pos ref alt &quot;track('wgEncodeOpenChromFaireFrontalcortexocSig.bigWig')&quot; -l 5


1   54789   C   -   0.00719999987632
1   54790   -   T   0.00719999987632
1   63738   ACT -   0.00710000004619
1   63738   ACT CTA 0.00710000004619
1   81963   -   AA  0.0120000001043
</code></pre>

<p>(:exampleend:)</p>

<h3 id="bigbed-tracks">bigBed tracks</h3>

<p>BigBed is a compressed indexed BED format that contains three mandatory columns and nine optional columns. The default <code>col</code> value for this format is `<code>(return 1 for matched records), but you can be specify items such as</code>1<code>(chrom) and</code>chromStart<code>(start, 0-based) according to output of command</code>vtools show track BIGBEDFILE`.</p>

<p>(:toggleexample Examples: Use bigBed tracks to annotate variants:) Let us create a project in hg19, import some data, and download a bigWig track from the UCSC ENCODE website:</p>

<pre><code>% wget http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeDukeAffyExon/wgEncodeDukeAffyExonUrothelUt189SimpleSignalRep2.bigBed
% vtools init track
% vtools import indels.vcf  --build hg19


INFO: Importing variants from indels.vcf (1/1)
indels.vcf: 100% [============================================] 184 12.3K/s in 00:00:00
INFO: 137 new variants (1 SNVs, 77 insertions, 58 deletions, 7 complex variants) from 184 lines are imported.
Importing genotypes: 0 0.0/s in 00:00:00
Copying samples: 0 0.0/s in 00:00:00
</code></pre>

<p>This tracks provides the following information:</p>

<pre><code>% vtools show track wgEncodeDukeAffyExonUrothelUt189SimpleSignalRep2.bigBed


Version:                4
Item count:             38378
Primary data size:      798350
Zoom levels:            7
Chrom count:            24
Chrom size:
    chr1                249250621
    chr10               135534747
    chr11               135006516
    chr12               133851895
    chr13               115169878
    chr14               107349540
    chr15               102531392
    chr16               90354753
    chr17               81195210
    chr18               78077248
    chr19               59128983
    chr2                243199373
    chr20               63025520
    chr21               48129895
    chr22               51304566
    chr3                198022430
    chr4                191154276
    chr5                180915260
    chr6                171115067
    chr7                159138663
    chr8                146364022
    chr9                141213431
    chrX                155270560
    chrY                59373566
Bases covered           1143378960
Mean depth:             1.055693
Min depth:              1.000000
Max depth:              18.000000
Std of depth:           0.310857
Number of fields:       9

Available columns (with type VARCHAR if unspecified or all=1):
chrom (1)               Chromosome (or contig, scaffold, etc.)
chromStart (2, INTEGER) Start position in chromosome
chromEnd (3, INTEGER)   End position in chromosome
name (4)                Name of item
score (5, INTEGER)      Score from 0-1000. Capped number of reads
strand (6)              + or -
signalValue (7, FLOAT)  Measurement of expression value of the gene
exonCount (8, INTEGER)  Number of exons used to estimate expression value
constituitiveExons (9, INTEGER) Number of constituitive exons used to estimate the
                        expression value
</code></pre>

<p>The track provides provides numeric annotation for each variant,</p>

<pre><code>% vtools output variant chr pos ref alt &quot;track('wgEncodeDukeAffyExonUrothelUt189SimpleSignalRep2.bigBed')&quot; -l5


1   10434   -   C   .
1   10440   C   -   .
1   54789   C   -   .
1   54790   -   T   .
1   63738   ACT -   .
</code></pre>

<p>The first five variant does not overlap with any regions in the bigBed file, but we can select variants using track membership:</p>

<pre><code>% vtools select variant &quot;track('wgEncodeDukeAffyExonUrothelUt189SimpleSignalRep2.bigBed') = 1&quot; -t encode


Running: 0 0.0/s in 00:00:00
INFO: 28 variants selected.
</code></pre>

<p>and lists fields from the bigBed file for these variants</p>

<pre><code>% vtools output encode chr pos ref alt &quot;track('wgEncodeDukeAffyExonUrothelUt189SimpleSignalRep2.bigBed', 4)&quot; -l5


1   761958  -       T       LINC00115
1   768117  GTTTT   -       RP11-206L10.11
1   768117  -       GTTTT   RP11-206L10.11
1   768118  -       TT      RP11-206L10.11
1   768625  -       A       RP11-206L10.11
</code></pre>

<p>and</p>

<pre><code>% vtools output encode chr pos ref alt &quot;track('wgEncodeDukeAffyExonUrothelUt189SimpleSignalRep2.bigBed', 'score')&quot; -l5


1   761958  -       T       692
1   768117  GTTTT   -       659
1   768117  -       GTTTT   659
1   768118  -       TT      659
1   768625  -       A       659
</code></pre>

<p>(:exampleend:)</p>

<h3 id="indexed-bam-tracks">Indexed BAM tracks</h3>

<p>Tracks in BAM format provides information regarding aligments, namely the reads that cover the starting position of each variant. If the variant is called from the provided BAM file, the BAM track provides information regarding the reads from which variants are called.</p>

<p><em>variant tools</em> currently only use the starting location of variants so it ignores reads that overlap but do not cover the starting position of a variant (e.g. an insertion).</p>

<p>A BAM track accepts the following fields,</p>

<ol>
<li><code>coverage</code>: number of reads that cover the starting position of each variant. This is the default field.</li>
<li><code>calls</code>: nucleotide of the reads at the variant location. This allows you to show the number of reference and alternative alleles for SNV variants, but not so informative for indels.</li>
<li><code>reads</code>: display a small piece of nucleotide sequence around the variant location (default to 5, namely the variant location and 4 base after it), separated by <code>|</code>. <code>_</code> will be displayed if the position goes beyond the end of a read. A parameter can be specified in the form of <code>reads?start=-5&amp;width=10</code> to change the starting point and width of displayed sequence.</li>
<li><code>qual</code>: A list of phred base quality of reads at the location.</li>
<li><code>avg_qual</code>: Average qual scores of all reads.</li>
<li><code>mapq</code>: A list of phred-scaled quality of alignment at the location.</li>
<li><code>avg_mapq</code>: Average map qual scores of all reads.</li>
<li>Any tag values listed at the end of command <code>vtools show track</code>.</li>
</ol>

<p>Parameters can be used to limit the reads to count or display, and change the way reads are displayed. The <code>bam</code> track currently supports the following options:</p>

<ol>
<li><code>type</code>: If set to `<code>(matched to reference sequence),</code>1<code>(unmatched single nucleotide),</code>2<code>(insertion) and</code>3` (deletion), only reads that are matched, unmatched (single nucleotide), insertion or deletion at the variant location is counted or outputted. Note that nucleotide before the insertion will be matched to reference genome, but they are not counted as matched.</li>
<li><code>min_mapq</code>: limits the reads to those with <code>mapq</code> scores that exceed the specified value.</li>
<li><code>min_qual</code>: limits the reads to those with <code>qual</code> scores that execeed the specified value.</li>
<li><code>color=[1|0]</code>: display variant location in blue, and insertions in green for <code>reads</code> field.</li>
<li><code>limit</code>: limit the number of reads or calls to display if the depth of coverage is high.</li>
<li><code>delimiter</code>: character (e.g. <code>\t</code> to separate reads in the <code>reads</code> output (<code>|</code> is used by default).</li>
<li><code>show_seq=[1|0]</code>: A <code>.</code> is used by default when the nucleotide matches the reference genome at the location. The actual nucleotide sequence will be displayed if this option is set to <code>1</code>.</li>
</ol>

<p>You can count the number of reads that match (or unmatch) the reference genome using parameter <code>coverage</code>type=0@@.</p>

<p>(:toggleexample Examples: Use BAM files to check the details of variant calls. :) Now suppose that we have a project with a list of variants (due to the size of BAM files, original data is not provided), we select the variants based on the sample from which they are called:</p>

<pre><code>% vtools select variant --samples &quot;sample_name = 'WGS4_9'&quot; -t ex49


INFO: 1 samples are selected by condition: sample_name = 'WGS4_9'
Running: 3,959 164.5/s in 00:00:24
INFO: 1191 variants selected.
</code></pre>

<p>We first need to check the available information that can be retrived</p>

<pre><code>$ vtools show track LP6005253-DNA_A02.bam


Header:
@HD VN:1.0  SO:coordinate
@PG ID:CASAVA   VN:CASAVA-1.9.0a1_110909    CL:/illumina/development/casava/CASAVA-VariantCalling-2.12a_gVCF/bin/configureBuild.pl --targets all bam --inSampleDir=/illumina/build/services/Projects/MDAnderson_Thompson2/LP6005253-DNA_A02/Aligned/D1LTMACXX_Aligned_MDAnderson_Thompson2_LP6005253-DNA_A02_121222_SN1012_0268_BD1LTMACXX_CE_L5/Sample_LP6005253-DNA_A02 --inSampleDir=/illumina/build/services/Projects/MDAnderson_Thompson2/LP6005253-DNA_A02/Aligned/D1LTMACXX_Aligned_MDAnderson_Thompson2_LP6005253-DNA_A02_121222_SN1012_0268_BD1LTMACXX_CE_L6/Sample_LP6005253-DNA_A02 --inSampleDir=/illumina/build/services/Projects/MDAnderson_Thompson2/LP6005253-DNA_A02/Aligned/D1LTMACXX_Aligned_MDAnderson_Thompson2_LP6005253-DNA_A02_121222_SN1012_0268_BD1LTMACXX_CE_L7/Sample_LP6005253-DNA_A02 --outDir=/scratch/LP6005253-DNA_A02 --samtoolsRefFile=/illumina/scratch/services/Genomes/FASTA_UCSC/HumanNCBI37_UCSC/HumanNCBI37_UCSC_XX.fa --indelsSaveTempFiles --variantsConsensusVCF --jobsLimit=12 --variantsPrintUsedAlleleCounts --variantsWriteRealigned --sortKeepAllReads --bamChangeChromLabels=OFF --sgeQueue=prod-s.q
@SQ SN:chr1 LN:249250621
@SQ SN:chr2 LN:243199373
@SQ SN:chr3 LN:198022430
@SQ SN:chr4 LN:191154276
@SQ SN:chr5 LN:180915260
@SQ SN:chr6 LN:171115067
@SQ SN:chr7 LN:159138663
@SQ SN:chrX LN:155270560
@SQ SN:chr8 LN:146364022
@SQ SN:chr9 LN:141213431
@SQ SN:chr10    LN:135534747
@SQ SN:chr11    LN:135006516
@SQ SN:chr12    LN:133851895
@SQ SN:chr13    LN:115169878
@SQ SN:chr14    LN:107349540
@SQ SN:chr15    LN:102531392
@SQ SN:chr16    LN:90354753
@SQ SN:chr17    LN:81195210
@SQ SN:chr18    LN:78077248
@SQ SN:chr20    LN:63025520
@SQ SN:chr19    LN:59128983
@SQ SN:chr22    LN:51304566
@SQ SN:chr21    LN:48129895
@SQ SN:chrM LN:16571

Chrom size:             24
    chr1                249250621
    chr2                243199373
    chr3                198022430
    chr4                191154276
    chr5                180915260
    chr6                171115067
    chr7                159138663
    chrX                155270560
    chr8                146364022
    chr9                141213431
    chr10               135534747
    chr11               135006516
    chr12               133851895
    chr13               115169878
    chr14               107349540
    chr15               102531392
    chr16               90354753
    chr17               81195210
    chr18               78077248
    chr20               63025520
    chr19               59128983
    chr22               51304566
    chr21               48129895
    chrM                16571

Available fields (with type VARCHAR if unspecified or all=1):
0 (INTEGER)             1 if depth is over 0, NULL otherwise
coverage (INTEGER)      Number of reads that cover the starting position of the variant
calls                   nucleotide of the reads at the variant location
reads                   nucleotide sequence around the variant location
qual                    A list of phred base quality of reads at the location
avg_qual (FLOAT)        Average qual score of all reads
mapq                    A list of phred base quality of alignment at the location
avg_mapq (FLOAT)        Average mapq score of all reads

Tags and flag that can be outputed or used in filters, with values from the 1st record:
AM                      C (int)    : 0
BC                      Z (string) : 0
XD                      Z (string) : 49A12AC19C11C4
SM                      i (int32)  : 0
AS                      i (int32)  : 511
flag                    int flag   : 0x63 (paired, unmapped according to bits 1 &amp; 3)

Parameters start (default to 0), width (default to 5) and color (default to 0) can be used with reads to adjust the window around variant, and use colors for insertions and variant allele, with syntax reads?start=-5&amp;width=10&amp;color=1. min_qual, min_mapq and TAG=VAL (or &gt;, &gt;=, &lt;, &lt;=, !=) can be used for all fields to limit the reads to the ones with mapq and qual scores that exceed the specified value, and tag satisfying specified conditions. Parameter limit limits the number of reads or calls to display if the depth of coverage is high.
</code></pre>

<p>The depth of coverage of these variants could be obtained using the BAM track,</p>

<pre><code>% vtools output ex49 chr pos ref alt &quot;track('LP6005253-DNA_A02.bam')&quot; -l5


1   1138963 C   T   26
1   1470808 G   A   37
1   6161109 C   T   27
1   6314785 T   C   32
1   9990112 A   G   43
</code></pre>

<p>The quality of reads and alignment can be displayed using fields <code>qual</code> and <code>mapq</code>,</p>

<pre><code>% vtools output ex49 chr pos ref alt &quot;track('LP6005253-DNA_A02.bam', 'qual')&quot; -l5


1   1138963 C   T   34,34,32,30,33,39,40,41,31,34,23,25,37,33,34,40,40,2,11,31,33,24,2,40,39,35
1   1470808 G   A   31,2,37,35,35,33,33,35,33,29,41,35,35,33,33,2,35,5,35,35,36,40,31,40,31,26,23,38,33,39,31,41,40,30,35,34,34
1   6161109 C   T   10,31,32,39,31,39,41,41,35,2,22,40,38,28,39,40,39,35,41,20,40,35,39,38,35,30,34
1   6314785 T   C   2,34,37,2,33,2,31,27,37,10,24,39,33,36,31,35,35,36,33,33,38,41,41,29,38,38,39,23,35,35,31,35
1   9990112 A   G   34,34,37,37,35,36,36,33,36,36,41,31,37,39,36,40,38,36,41,38,37,41,35,25,38,40,40,40,36,41,41,39,37,34,30,36,36,41,41,36,39,37,37




% vtools output ex49 chr pos ref alt &quot;track('LP6005253-DNA_A02.bam', 'mapq')&quot; -l5


1   1138963 C   T   254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,241,254,254,254,254,254,254,254,254
1   1470808 G   A   254,194,254,254,254,254,254,254,254,254,254,254,254,254,254,149,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254
1   6161109 C   T   254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254
1   6314785 T   C   254,254,254,254,254,254,254,254,254,231,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254
1   9990112 A   G   254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254,254
</code></pre>

<p>We can exclude some reads depending on quality scores, using parameters <code>min_qual</code> or <code>min_mapq</code>,</p>

<pre><code>% vtools output ex49 chr pos ref alt &quot;track('LP6005253-DNA_A02.bam', 'coverage?min_qual=30')&quot; -l5


1   1138963 C   T   20
1   1470808 G   A   31
1   6161109 C   T   22
1   6314785 T   C   24
1   9990112 A   G   42
</code></pre>

<p>Read TAGs can also be outputed or used in filter conditions. For example, this bam file has tags <code>AM</code>, <code>BC</code>, <code>XD</code>, <code>SM</code> and <code>AS</code>, you can list the <code>AS</code> values of all reads using command</p>

<pre><code>% vtools output ex49 chr pos ref alt 'ref_sequence(chr, pos-3, pos+3)' &quot;track('LP6005253-DNA_A02.bam', 'AS?min_qual=35')&quot; -l5


1   1138963 C   T   AGCCTCC 1007|1001|1001|966|941|1006|816|946|1002
1   1470808 G   A   GGCGGCC 1007|475|783|951|998|878|967|968|1004|967|967|962|962|935|1004|1008|963|998
1   6161109 C   T   TACCGTG 897|922|927|830|967|965|936|997|997|832|774|961|966|922|937|949|964
1   6314785 T   C   CGATGGG 939|517|0|925|968|965|962|912|968|967|855|0|924|962|923|919
1   9990112 A   G   ATCATTA 964|966|968|1007|965|1003|1008|1008|991|963|989|963|906|872|1008|1007|965|962|968|1002|963|1007|963|966|1006|937|912|1008|966|1008|962|1008|1008|1005|1008|1008
</code></pre>

<p>or its values to filter reads:</p>

<pre><code>% vtools output ex49 chr pos ref alt 'ref_sequence(chr, pos-3, pos+3)' &quot;track('LP6005253-DNA_A02.bam', 'coverage?AS&gt;1000')&quot; -l5


1   1138963 C   T   AGCCTCC 5
1   1470808 G   A   GGCGGCC 7
1   6161109 C   T   TACCGTG 0
1   6314785 T   C   CGATGGG 1
1   9990112 A   G   ATCATTA 18
</code></pre>

<p>(:exampleend:)</p>

<p>(:toggleexample Count and display calls and reads for reads of different types:)</p>

<p>The <code>track</code> function can also be used to display calls (nucleotide at the variant site) and reads (nucleotide sequences around the variant site). To demonstrate the features more clearly, we will use a project with more types of variants.</p>

<p>First, we can display the nucleotide at the variant site using the <code>calls</code> parameter,</p>

<pre><code>% bam=/path/to/a/bam/file
% vtools output exon1 chr pos ref alt &quot;track('${bam}', 'calls?limit=20')&quot;


1   118420020   -               T   IIIII.I.I.I..III.I.I
1   159023386   G               A   .AAAAA..A.AA..A..A.A
1   160398161   G               A   A.A.AA....A..AAA..A.
1   180772617   C               T   .T..TTT..TT.T.TTTT.T
3   12581722    T               C   C.C.......C.C....CC.
4   1945715     A               T   .N..T..TTTTT.T..T.TT
5   137089865   C               G   .G..G.G..GGGG...G...
8   42878531    TCCT            -   ....................
12  65449852    C               A   AA.A.A..AA.AAA..AAAA
16  11929051    T               C   CC..CCC.........C...
16  17197814    G               A   AA..A.AAAAA..AA.AAAA
17  78938525    G               A   A......AAA....AAAAAA
17  79525590    C               G   .GG.GGG..GG...G.G...
17  79687655    C               T   ....T.TTT.TTT..TTN.T
19  34843754    CCCCACCCCAGC    -   ..........**.*......
</code></pre>

<p>The output shows</p>

<ol>
<li>Nucleotides that match the reference sequence are displayed as <code>.</code>.</li>
<li>Deletions are displayed as <code>*</code>.</li>
<li>Insertions are displayed as <code>I</code>.</li>
</ol>

<p>You can display the reads around the variant site, using the <code>reads</code> parameter:</p>

<pre><code>% vtools output exon1 chr pos ref alt &quot;track('${bam}', 'reads?limit=5')&quot;


1   118420020   -               T   T.....|T.....|T.....|T.....|T.....
1   159023386   G               A   ..   |A... |A....|A....|A....
1   160398161   G               A   A    |..   |A..  |...  |A....
1   180772617   C               T   .    |T..  |.... |.....|T....
3   12581722    T               C   C    |..   |C..  |.... |.....
4   1945715     A               T   .... |N....|.....|.....|T....
5   137089865   C               G   .....|G....|.....|.....|G....
8   42878531    TCCT            -   .    |.... |.... |.....|.....
12  65449852    C               A   A    |A..  |.... |A....|.....
16  11929051    T               C   C.   |C....|.....|.....|C....
16  17197814    G               A   A.   |A..  |.... |.....|A....
17  78938525    G               A   A..  |.... |.... |.....|.....
17  79525590    C               G   ..   |G... |G... |.....|G....
17  79687655    C               T   .    |.....|.....|.....|T....
19  34843754    CCCCACCCCAGC    -   ..   |..   |...  |.....|.....
</code></pre>

<p>Parameter <code>limit-5</code> is used to avoid lengthy output.</p>

<p>Parameters <code>start</code> and <code>width</code> can be used to specify the window of sequences to display:</p>

<pre><code>% vtools output exon1 chr pos ref alt &quot;track('${bam}', 'reads?limit=5&amp;start=-5&amp;width=8&amp;color=1&amp;show_seq=1')&quot;


1   118420020   -               T   GTTACTTTT|GTTACTTTT|GTTACTTTT|GTTACTTTT|GTTACTTTT
1   159023386   G               A   AAGTGGT |AAGTGATG|AAGTGATG|AAGTGATG|AAGTGATG
1   160398161   G               A   CTTCCA  |CTTCCGT |CTTCCATG|CTTCCGTG|CTTCCATG
1   180772617   C               T   TACTAC  |TACTATGC|TACTACGC|TACTACGC|TACTATGC
3   12581722    T               C   GGTTGC  |GGTTGTG |GGTTGCGC|GGTTGTGC|GGTTGTGC
4   1945715     A               T   AAATGACC|AAANNNCC|AAATGACC|AAATGACC|AAATGTCC
5   137089865   C               G   TGGAGCCA|TGGAGGCA|TGGAGCCA|TGGAGCCA|TGGAGGCA
8   42878531    TCCT            -   CTTCCT  |CTTCCTCC|CTTCCTCC|CTTCCTCC|CTTCCTCC
12  65449852    C               A   ACTTAA  |ACTTAAAT|ACTTACAT|ACTTAAAT|ACTTACAT
16  11929051    T               C   TTTTTCT |TTTTTCTC|TTTTTTTC|TTTTTTTC|TTTTTCTC
16  17197814    G               A   GAGAGAA |GAGAGAAA|GAGAGGAA|GAGAGGAA|GAGAGAAA
17  78938525    G               A   CAGGCACT|CAGGCGCT|CAGGCGCT|CAGGCGCT|CAGGCGCT
17  79525590    C               G   CTCCCCT |CTCCCGTT|CTCCCGTT|CTCCCCTT|CTCCCGTT
17  79687655    C               T   CAGACC  |CAGACCAC|CAGACCAC|CAGACCAC|CAGACTAC
19  34843754    CCCCACCCCAGC    -   GCAGACC |GCAGACC |GCAGACCC|GCAGACCC|GCAGACCC
</code></pre>

<p>Parameter <code>color=1</code> will make the insertion displayed in green, and nucleotide at variant site displayed in blue on terminal. Parameter <code>show_seq</code> displays real sequence instead of <code>.</code> for matched nucleotides.</p>

<p>You can also specify the types of reads so that you can count or display just a subsets of reads. For example, you can display all reads on the forward strand</p>

<pre><code>% vtools output exon1 chr pos ref alt &quot;track('${bam}', 'reads?limit=5&amp;start=-5&amp;width=8&amp;color=1&amp;show_seq=1&amp;strand=+')&quot;


1   118420020   -               T   GTTACTTTT|GTTACTTTT|GTTACTTTT|GTTACTTTT|GTTACTTTT
1   159023386   G               A   AAGTGATG|AAGTGATG|AAGTGATG|AAGTGATG|AAGTGGTG
1   160398161   G               A   CTTCCATG|CTTCCATG|CTTCCATG|CTTCCATG|CTTCCATG
1   180772617   C               T   TACTACGC|TACTACGC|TACTATGC|TACTATGC|TACTACGC
3   12581722    T               C   GGTTGTG |GGTTGCGC|GGTTGTGC|GGTTGTGC|GGTTGTGC
4   1945715     A               T   AAANNNCC|AAATGACC|AAATGACC|AAATGTCC|AAATGACC
5   137089865   C               G   TGGAGCCA|TGGAGGCA|TGGAGCCA|TGGAGCCA|TGGAGGCA
8   42878531    TCCT            -   CTTCCTCC|CTTCCTCC|CTTCCTCC|CTTCCTCC|CTTCCTCC
12  65449852    C               A   ACTTACAT|ACTTAAAT|ACTTAAAT|ACTTAAAT|ACTTAAAT
16  11929051    T               C   TTTTTCTC|TTTTTTTC|TTTTTTTC|TTTTTTTC|TTTTTTTC
16  17197814    G               A   GAGAGAA |GAGAGAAA|GAGAGGAA|GAGAGAAA|GAGAGAAA
17  78938525    G               A   CAGGCACT|CAGGCGCT|CAGGCGCT|CAGGCGCT|CAGGCGCT
17  79525590    C               G   CTCCCGTT|CTCCCGTT|CTCCCGTT|CTCCCCTT|CTCCCCTT
17  79687655    C               T   CAGACCAC|CAGACCAC|CAGACCAC|CAGACCAC|CAGACTAC
19  34843754    CCCCACCCCAGC    -   GCAGACCC|GCAGACCC|GCAGACCC|GCAGACCC|GCAGACCC
</code></pre>

<p>Or display just the mismatch single-nucleotides</p>

<pre><code>% vtools output exon1 chr pos ref alt &quot;track('${bam}', 'reads?limit=5&amp;start=-5&amp;width=8&amp;type=1')&quot;


1   118420020   -               T
1   159023386   G               A   .....A..|.....A..|.....A..|.....A..|.....A..
1   160398161   G               A   .....A  |.....A..|.....A..|.....A..|.....A..
1   180772617   C               T   .....T..|.....T..|.....T..|.....T..|.....T..
3   12581722    T               C   .....C  |.....C..|.....C..|.....C..|.....C..
4   1945715     A               T   ...NNN..|.....T..|.....T..|.....T..|.....T..
5   137089865   C               G   .....G..|.....G..|.....G..|.....G..|.....G..
8   42878531    TCCT            -
12  65449852    C               A   .....A  |.....A..|.....A..|G....A..|.....A..
16  11929051    T               C   .....C. |.....C..|.....C..|.....C..|.....C..
16  17197814    G               A   .....A. |.....A..|.....A..|.....A..|.....A..
17  78938525    G               A   .....A..|.....A..|.....A..|.....A..|.....A..
17  79525590    C               G   .....G..|.....G..|.....G..|.....G..|.....G..
17  79687655    C               T   .....T..|.....T..|.....T..|.....T..|.....T..
19  34843754    CCCCACCCCAGC    -
</code></pre>

<p>For example, we can output the number of reads that match (type 0), mismatch (type 1), insert before (type 2), or delete (type 3) the nucleotide sequence at the variant site:</p>

<pre><code>vtools output exon1 chr pos ref alt &quot;track('${bam}')&quot; \
    &quot;track('${bam}', 'coverage?type=0')&quot; \
    &quot;track('${bam}', 'coverage?type=1')&quot; \
    &quot;track('${bam}', 'coverage?type=2')&quot; \
    &quot;track('${bam}', 'coverage?type=3')&quot; \
    &quot;track('${bam}', 'coverage?type=3&amp;strand=+')&quot; \
    &quot;track('${bam}', 'coverage?type=3&amp;strand=-')&quot;


1   118420020   -               T   25  9   0   16  0   0   0
1   159023386   G               A   43  18  25  0   0   0   0
1   160398161   G               A   50  23  27  0   0   0   0
1   180772617   C               T   40  17  23  0   0   0   0
3   12581722    T               C   107 63  44  0   0   0   0
4   1945715     A               T   64  29  35  0   0   0   0
5   137089865   C               G   81  41  40  0   0   0   0
8   42878531    TCCT            -   50  27  0   14  9   3   6
12  65449852    C               A   65  30  35  0   0   0   0
16  11929051    T               C   69  37  32  0   0   0   0
16  17197814    G               A   57  19  38  0   0   0   0
17  78938525    G               A   88  47  41  0   0   0   0
17  79525590    C               G   58  31  27  0   0   0   0
17  79687655    C               T   95  54  41  0   0   0   0
19  34843754    CCCCACCCCAGC    -   64  30  0   0   34  16  18
</code></pre>

<p>The last two functions are interesting as it shows the number of reads on forward and reverse strands that shows the deletion. This information can be usful because the variant might not be real if it exists mostly on one of the strands.</p>

<p>(:exampleend:)</p>

<p>An option <code>color=1</code> can be used with the <code>read</code> field to display insertions and variant allele in color (green and blue respectively). This is very helpful if you have long reads and reads that contain indels.</p>

<p>Online BAM tracks can also be used so you do not have to download large BAM files in order to use them.</p>

<p>(:toggleexample Examples: obtain depth of coverage of variants using online BAM files:)</p>

<pre><code>% vtools output variant chr pos &quot;track('ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data/HG00096/alignment/HG00096.mapped.ILLUMINA.bwa.GBR.low_coverage.20120522.bam')&quot; -l5


[bam_index_load] attempting to download the remote index file.
1   533 0
1   41342   1
1   41791   4
1   44449   7
1   44539   12
</code></pre>

<p>(:exampleend:)</p>


    
    
        <div class="chevrons">
    <div id="navigation">
<a class="nav nav-prev" href="https://vatlab.github.io/vat-docs/documentation/functions/mut_sequence/" title="mut_sequence"> <i class="fa fa-chevron-left"></i><label>mut_sequence</label></a>
    <a class="nav nav-next" href="https://vatlab.github.io/vat-docs/documentation/functions/genotype/" title="Genotype" style="margin-right: 0px;"><label>Genotype</label><i class="fa fa-chevron-right"></i></a></div>
  </div>

  </section>
</article>

<footer>

<div class="footline">
    

    

  

    <div class="github-link">
      <a href="Documentation/Functions/track.md" target="blank"><i class="fa fa-code-fork"></i>
        Improve this page</a>
    </div>
  </div>


	<div>


  
    Create a content/_footer.md file to customize the footer content
  



	</div>
</footer>

<script src="https://vatlab.github.io/vat-docs/js/clipboard.min.js"></script>

<link href="https://vatlab.github.io/vat-docs/css/featherlight.min.css" rel="stylesheet">
<script src="https://vatlab.github.io/vat-docs/js/featherlight.min.js"></script>



<script src="https://vatlab.github.io/vat-docs/theme-flex/script.js"></script>

    

    
    

    
  </body>
</html>