<!DOCTYPE html>
<html>
  <head>
    <title>vatlab</title>
    
      <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
<meta name="generator" content="Hugo 0.36.1" />
<title>Import :: vatlab</title>
<link rel="shortcut icon" href="https://vatlab.github.io/vat-docs/images/favicon.png" type="image/x-icon" />
<link href="https://vatlab.github.io/vat-docs/css/font-awesome.min.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/css/nucleus.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/theme-flex/style.css" rel="stylesheet">

<link rel="stylesheet" href="https://vatlab.github.io/vat-docs/css/bootstrap.min.css">
<script src="https://vatlab.github.io/vat-docs/js/jquery-2.x.min.js"></script>
<script type="text/javascript">
      var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
</script>
<meta name="description" content="">


    
  </head>
  <body data-url="/documentation/vtools-commands/import/">
    
      <header>
  <div class="logo">
    
	
  
    <a class="baselink" href="https://vatlab.github.io/vat-docs/">vatlab</a>
  

  </div>
  <div class="burger"><a href="javascript:void(0);" style="font-size:15px;">&#9776;</a></div>
</header>
<article>
  <aside>
    <ul class="menu">
          <li data-nav-id="/" class="dd-item">
          <a href="https://vatlab.github.io/vat-docs/">
            <i class="fa fa-fw fa-home"></i>
          </a>
          </li>
    <li data-nav-id="/install/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/install/">Installation</a>
      </div>
    </li>
    <li data-nav-id="/documentation/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/">Documentation</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/keyconcepts/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/">
            Key concepts
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/tutorials/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/tutorials/">
            Tutorials
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/help/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/help/">
            Help
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/vtools-commands/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/">Vtools commands</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools-commands/init/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/init/">
            Init
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/import/" class="dd-item active">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/import/">
            Import
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/update/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/update/">
            Update
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/phenotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/phenotype/">
            Phenotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/liftover/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/liftover/">
            lifttover
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/show/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/show/">
            show
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/use/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/use/">
            use
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/select/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/select/">
            select
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/exclude/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/exclude/">
            exclude
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/compare/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/compare/">
            compare
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/associate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/associate/">
            associate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/output/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/output/">
            output
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/export/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/export/">
            export
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/remove/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/remove/">
            remove
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/execute/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/execute/">
            execute
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/admin/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/admin/">
            admin
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/functions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/functions/">Functions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/functions/ref_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/ref_sequence/">
            ref_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/mut_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/mut_sequence/">
            mut_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/track/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/track/">
            track
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/genotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/genotype/">
            Genotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/samples/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/samples/">
            samples
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/in_table/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/in_table/">
            in_table
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/otherfunctions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/otherfunctions/">
            Other functions
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/vtools_report/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/">Vtools_report</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools_report/avg_depth/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/avg_depth/">
            avg_depth
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/discordance_rate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/discordance_rate/">
            discordance_rate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sam_info/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sam_info/">
            sam_info
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sequence/">
            sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/trans_ratio/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/trans_ratio/">
            trans_ratio
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/variant_stat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/variant_stat/">
            variant_stat
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/inbreeding_coefficient/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/inbreeding_coefficient/">
            inbreeding_coefficient
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/meta_analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/meta_analysis/">
            meta_analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_fields/">
            plot_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_geno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_geno_fields/">
            plot_geno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_pheno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_pheno_fields/">
            plot_pheno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_association/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_association/">
            plot_association
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/customization/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/">Customization</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/format/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/">
            Format
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/">Simulation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/custimzedsimu/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/custimzedsimu/">
            Customized simulations
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/models/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/">Models</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/models/peng1/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/peng1/">
            Peng2014_ex1
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/models/pen2/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/pen2/">
            Peng2014_ex2
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/">Pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/customizedpipeline/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/customizedpipeline/">
            Customized pipeline
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/pipelines/variant-calling/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/">Variant calling</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_hg19/">
            bwa_gatk33_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_b37/">
            bat_gatk33_b37
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_hg19/">
            bwa_gatk28_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_b37/">
            bwa_gatk28_b37
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/other-pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/">Other pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/other-pipelines/import_vcf/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/import_vcf/">
            import_vcf
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/anno_utils/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/anno_utils/">
            anno_utils
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/illumina/">
            Illumnina
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/annovar/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/annovar/">
            ANNOVAR
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/snpeff/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/snpeff/">
            snpEff
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/king/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/king/">
            KING
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/utilities/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/utilities/">Utilities</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/utilities/report_progress/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/report_progress/">
            report_progress
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/utilities/_md5sumd/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/_md5sumd/">
            md5sumd
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/">Application</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
    <li data-nav-id="/applications/annotation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/">Annotation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/tutorial/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/tutorial/">
            Tutorial
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/customized/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">
            Customized annotation databases
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/annotation/genes/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/">Genes</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/genes/ccdsgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/ccdsgene/">
            ccdsGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/hugogene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/hugogene/">
            Hugo gene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/entrezgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/entrezgene/">
            EntrezGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/knowngene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/knowngene/">
            knownGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/refgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/refgene/">
            refGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/cancergenecensus/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/cancergenecensus/">
            CancerGeneCensus
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/variants/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/">Variants</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/variants/cosmic/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/cosmic/">
            COSMIC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbnsfp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbnsfp/">
            sbNSFP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbsnp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbsnp/">
            dbSNP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/linkagephysicalmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/linkagephysicalmap/">
            linkage-physical map
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/databaseof/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/databaseof/">
            Database of Genomic Variants
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/exomesequencingpro/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/exomesequencingpro/">
            Exome Sequencing Project
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/gwas/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/gwas/">
            gwasCatalog
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/hapmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/hapmap/">
            HapMap
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/thousand/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/thousand/">
            Thousand Genomes
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/regions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/">Regions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/regions/low-complexity-regions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/low-complexity-regions/">
            Low complexity regions
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/phast-cons/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/phast-cons/">
            Phast cons
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/genomic-super-dups/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/genomic-super-dups/">
            Genomic super dups
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/cytoband/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/cytoband/">
            cytoBand
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/misc/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/">Misc</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/misc/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/illumina/">
            Illumina
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/keggpathway/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/keggpathway/">
            keggPathway
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/encode/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/encode/">
            ENCODE
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
      <li data-nav-id="/applications/discovery/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/discovery/">
            Discovery
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/">Association</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/data-explorationqc/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/data-explorationqc/">
            Data exploration/QC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/association-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/association-analysis/">
            Association analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/post-assoc-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/post-assoc-analysis/">
            Post-assoc analysis
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/basic-data-statistics/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/">Basic data statistics</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/basic-data-statistics/group-statwrite/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/group-statwrite/">
            Group stat&amp;Write
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-variant-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/">Single variant analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-variant-analysis/fisher-exact-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/fisher-exact-test/">
            Fisher exact test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-gene-association-methods/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/">Single gene association methods</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-gene-association-methods/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/cmc-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/cmc-test/">
            CMC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/c_alpha-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/c_alpha-test/">
            C-alpha test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/kbac-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/kbac-test/">
            KBAC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rbt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rbt-test/">
            RBT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rarecover-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rarecover-test/">
            RareCover test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/vt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/vt-test/">
            VT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/wss-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/wss-test/">
            WSS test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/asum-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/asum-test/">
            aSum test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/jointconditional-association-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/">Joint/Conditional association analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/collapsing-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/collapsing-methods/">
            Collapsing methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/aggregation-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/aggregation-methods/">
            Aggregation methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/weighted-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/weighted-methods/">
            Weighted methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/variable-thresholds/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/variable-thresholds/">
            Variable thresholds
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/skat-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/skat-test/">
            SKAT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/metaskat-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/metaskat-analysis/">
            MetaSKAT analysis
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/vat-stacking/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/vat-stacking/">VAT stacking</a>
      </div>
    </li>
    <li data-nav-id="/applications/association/create-your-test/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/">Create your test!</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/create-your-test/running-r-programs/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/running-r-programs/">
            Running R programs
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/create-your-test/programming-in-vat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/programming-in-vat/">
            Programming in VAT
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/view-association-result/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/">View association result</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/view-association-result/qq-plot-and-manhattan-plot/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/qq-plot-and-manhattan-plot/">
            QQ plot and Manhattan plot
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/development/">Development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/development/chagelog/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/chagelog/">
            ChangeLog
          </a>
        </div>
    </li>
      <li data-nav-id="/development/getinvolved/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/getinvolved/">
            Get Involved
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/administration/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/administration/">Administration</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/administration/allrecent/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/allrecent/">
            All recent changes
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/autolink/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/autolink/">
            Autolink
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/stylesheet/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/stylesheet/">
            Style sheet
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/trafficstat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/trafficstat/">
            Traffic statistics
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/under-development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/under-development/">Under development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/under-development/faq/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/under-development/faq/">
            FAQ
          </a>
        </div>
    </li>
        </ul>
    </li>




    </ul>
    <section>
    </section>
  </aside>
  <section class="page">
    
    <div class="nav-select">
      <center>Navigation : 
        <select onchange="javascript:location.href = this.value;">
          
    <option value="/install/" >
   Installation</option>
    <option value="/documentation/" >
   Documentation</option> 
      <option value="/documentation/keyconcepts/" >- Key concepts</option>
      <option value="/documentation/tutorials/" >- Tutorials</option>
      <option value="/documentation/help/" >- Help</option>
    <option value="/documentation/vtools-commands/" >
  - 
   Vtools commands</option> 
      <option value="/documentation/vtools-commands/init/" >-- Init</option>
      <option value="/documentation/vtools-commands/import/"  selected>-- Import</option>
      <option value="/documentation/vtools-commands/update/" >-- Update</option>
      <option value="/documentation/vtools-commands/phenotype/" >-- Phenotype</option>
      <option value="/documentation/vtools-commands/liftover/" >-- lifttover</option>
      <option value="/documentation/vtools-commands/show/" >-- show</option>
      <option value="/documentation/vtools-commands/use/" >-- use</option>
      <option value="/documentation/vtools-commands/select/" >-- select</option>
      <option value="/documentation/vtools-commands/exclude/" >-- exclude</option>
      <option value="/documentation/vtools-commands/compare/" >-- compare</option>
      <option value="/documentation/vtools-commands/associate/" >-- associate</option>
      <option value="/documentation/vtools-commands/output/" >-- output</option>
      <option value="/documentation/vtools-commands/export/" >-- export</option>
      <option value="/documentation/vtools-commands/remove/" >-- remove</option>
      <option value="/documentation/vtools-commands/execute/" >-- execute</option>
      <option value="/documentation/vtools-commands/admin/" >-- admin</option>
  
    <option value="/documentation/functions/" >
  - 
   Functions</option>
    <option value="/documentation/vtools_report/" >
  - 
   Vtools_report</option>
    <option value="/documentation/customization/" >
  - 
   Customization</option>
    <option value="/documentation/pipelines/" >
  - 
   Pipelines</option>
    <option value="/documentation/utilities/" >
  - 
   Utilities</option>
  
    <option value="/applications/" >
   Application</option>
    <option value="/development/" >
   Development</option>
    <option value="/administration/" >
   Administration</option>
    <option value="/under-development/" >
   Under development</option>



        </select>
      </center>
    </div>
      <div>
        <div class="searchbox">
          <input data-search-input id="search-by" type="text" placeholder="Search...">
        </div>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/lunr.min.js"></script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/auto-complete.js"></script>
        <link href="https://vatlab.github.io/vat-docs/css/auto-complete.css" rel="stylesheet">
        <script type="text/javascript">
          
              var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
          
        </script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/search.js"></script>
      </div>
    

    <h1>Import</h1>
    
    
    
    

<h1 id="import-variants-genotypes-and-related-information-fields">Import variants, genotypes and related information fields</h1>

<h2 id="usage">Usage</h2>

<pre><code>% vtools import -h


usage: vtools import [-h] [--build BUILD] [--format FORMAT]
                     [--sample_name [SAMPLE_NAME [SAMPLE_NAME ...]]] [-f]
                     [-j N] [-v {0,1,2}]
                     input_files [input_files ...]
x
Import variants and related sample genotype from one or more delimiter
separated files (e.g. VCF and a number of indel formats).

positional arguments:
  input_files           A list of failes that will be imported. The file should
                        be delimiter separated with format described by
                        parameter --format. Gzipped files are acceptable. If a
                        preprocessor is defined in the format, input files
                        will be processed by the preprocessor before they are
                        imported.

optional arguments:
  -h, --help            show this help message and exit
  --build BUILD         Build version of the reference genome (e.g. hg18) of
                        the input data. If unspecified, it is assumed to be
                        the primary reference genome of the project. If a
                        reference genome that is different from the primary
                        reference genome of the project is specified, it will
                        become the alternative reference genome of the
                        project. The UCSC liftover tool will be automatically
                        called to map input coordinates between the primary
                        and alternative reference genomes. If you are
                        uncertain about the reference genome used in the data,
                        use a recent standard reference genome (e.g. hg19) and
                        validate it after the data is imported (c.f. &quot;vtools
                        admin --validate_build&quot;).
  --format FORMAT       Format of the input text file. It can be one of the
                        variant tools supported file types such as VCF (cf.
                        'vtools show formats'), or a local format
                        specification file (with extension .fmt). If
                        unspecified, variant tools will try to guess format
                        from file extension. Some file formats accept
                        parameters (cf. 'vtools show format FMT') and allow
                        you to import additional or alternative fields defined
                        for the format.
  --sample_name [SAMPLE_NAME [SAMPLE_NAME ...]]
                        Name of the samples imported by the input files. The
                        same names will be used for all files if multiple
                        files are imported. If unspecified, headers of the
                        genotype columns of the last comment line (line starts
                        with #) of the input files will be used (and thus
                        allow different sample names for input files). If
                        sample names are specified for input files without
                        genotype, samples will be created without genotype. If
                        sample names cannot be determined from input file and
                        their is no ambiguity (only one sample is imported), a
                        sample with NULL sample name will be created.
  -f, --force           Import files even if the files have been imported
                        before. This option can be used to import from updated
                        file or continue disrupted import, but will not remove
                        wrongfully imported variants from the master variant
                        table.
  -j N, --jobs N        Number of processes to import input file. Variant
                        tools by default uses four processes to import
                        variants and samples genotypes in parallel, and you
                        can use more or less processes by adjusting this
                        parameter. Due to the overhead of inter-process
                        communication, more jobs do not automatically lead to
                        better performance.
  -v {0,1,2}, --verbosity {0,1,2}
                        Output error and warning (0), info (1) and debug (2)
                        information to standard output (default to 1).
</code></pre>

<h2 id="general-usages-and-options">General usages and options</h2>

<p>Command <code>`vtools import</code> imports variants, sample genotypes and related information fields. The imported variants are saved to the master variant table <code>variant</code> of the project, along with their information fields. Samples are saved to a genotype database <code>$proj_genotype</code> as separate sample tables.</p>

<ol>
<li><p>Variant tools can import SNVs, Indels and complex variants with reference and alternative alleles explicitly listed in the source files. It cannot yet handle structural variants such as large indels listed in vcf file as <code>&lt;INS&gt;</code> or <code>DUP:TANDEM&gt;</code>. For details about how different types of variants are imported into <em>variant tools</em>, please refer to <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Details.Variants?action=edit" target="_blank">here</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Details.Variants?action=edit" target="_blank">?</a></p></li>

<li><p>Although vcf files produced by different variant calling software have different fields, <em>variant tools</em> provides a growing number of pipelines to import data in particular formats. You can have a look at available pipelines (<code>vtools show pipelines</code>) and details of a particular pipeline (<code>vtools show pipeline PIPELINE</code>) to see if you can find a pipeline to import your data.</p></li>

<li><p>It is sometimes useful to import only variants to a project. The variant info could be added later using command <code>vtools update</code>, or built into an annotation database to reduce the size of the project.</p></li>
</ol>

<p>(:toggleexample Examples: create a project and load some sample data:) Let us first create a project and download a sample project with a bunch of test datasets:</p>

<pre><code>% vtools init import --parent vt_testData


INFO: variant tools 1.0.7svn : Copyright (c) 2011 - 2012 Bo Peng
INFO: San Lucas FA, Wang G, Scheet P, Peng B (2012) Bioinformatics 28(3):421-422
INFO: Please visit http://varianttools.sourceforge.net for more information.
INFO: Creating a new project import
INFO: Extracting snapshot vt_testData to .
Downloading snapshot vt_testData.tar.gz from online repository
</code></pre>

<p>You can use command <code>vtools show snapshots</code> to get a list of available snapshots</p>

<p>(:exampleend:)</p>

<h3 id="file-formats-and-format-specification-files-fmt-option-format">File formats and format specification files (.fmt) (option <code>--format</code>)</h3>

<p><code>`vtools import</code> can handle input file in many different formats (e.g. <code>.vcf</code>) and their gzipped or bzipped versions (e.g. <code>.vcf.gz</code>). variant tools relies on <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.New?action=edit" target="_blank">format specification files</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.New?action=edit" target="_blank">?</a> to describe a file format. These files (with extension <code>.fmt</code>) tell variant tools how to read from an input file. They are available online and will be downloaded automatically to the local resource directory of variant tools. Please refer to the variant tools <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.HomePage?action=edit" target="_blank">input file format</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.HomePage?action=edit" target="_blank">?</a> for a list of supported formats, or use command <code>vtools show formats</code> to get a list of formats, and <code>vtools show format FMT</code> for details of a format.</p>

<p>A format specification file defines how to import variants (fields <code>chr</code>, <code>pos</code>, <code>ref</code>, and <code>alt</code>), variant info fields, genotypes, and genotype infor fields from input files. It basically tells command <code>vtools import</code> what fields are available from input file, from which column each field should be read, how to post-process input (e.g convert 0-based positions to 1-based), and how to store the data (data type).</p>

<p>Although variants consists of chromosome, position, reference and alternative alleles, your input does not have to contain all such information. For example, using format <code>rsname</code>, variant tools can import variants from a list of dbSNP IDs.</p>

<p>(:toggleexample Examples: Show supported file formats and details of a format:) Command <code>vtools show formats</code> lists all supported file formats with a short description.</p>

<pre><code>% vtools show formats


CASAVA18_snps           Input format illumina snps.txt file, created by CASAVA
                        version 1.8
                        (http://www.illumina.com/support/documentation.ilmn).
                        This format imports chr, pos, ref, alt of most likely
                        genotype, and a Q score for the most likely genotype.
plink                   Input format for PLINK dataset. Currently only PLINK
                        binary PED file format is supported (*.bed, *.bim &amp;
                        *.fam)
ANNOVAR                 Input format of ANNOVAR. No genotype is defined.
pileup_indel            Input format for samtools pileup indel caller. This
                        format imports chr, pos, ref, alt and genotype.
ANNOVAR_exonic_variant_function Output from ANNOVAR, generated from command
                        &quot;path/to/annovar/annotate_variation.pl annovar.txt
                        path/to/annovar/humandb/&quot;. This format imports chr,
                        pos, ref, alt and ANNOVAR annotations. For details
                        please refer to http://www.openbioinformatics.org/anno
                        var/annovar_gene.html
ANNOVAR_variant_function Output from ANNOVAR for files of type
                        &quot;*.variant_function&quot;, generated from command
                        &quot;path/to/annovar/annotate_variation.pl annovar.txt
                        path/to/annovar/humandb/&quot;. This format imports chr,
                        pos, ref, alt and ANNOVAR annotations. For details
                        please refer to http://www.openbioinformatics.org/anno
                        var/annovar_gene.html
CGA                     Input format from Complete Genomics Variant file
                        masterVarBeta-[ASM-ID].tsv.bz2, created by Complete
                        Genomcis Analysis Tools (GSA Tools 1.5 or eariler,
                        http://www.completegenomics.com/sequence-
                        data/cgatools/, http://media.completegenomics.com/docu
                        ments/DataFileFormats+Standard+Pipeline+2.0.pdf). This
                        format imports chr, pos, ref, alt of only variants
                        that have been fully called and are not equals to ref.
                        (E.g. records with zygosity equal to no-call and half,
                        and varType equal to ref are discarded.)
map                     This input format imports variants from files in MAP
                        format (with columns chr, name gen_dist, pos), or any
                        delimiter-separated format with columns chr and pos.
                        Because these input files do not contain reference and
                        alternative alleles of variants, this format queries
                        such information from the dbSNP database using chr and
                        pos. Records that does not exist in dbSNP will be
                        discarded. Records with multiple alternative alleles
                        will lead to multiple records.
polyphen2               To be used to export variants in a format that is
                        acceptable by the polyphen2 server
                        http://genetics.bwh.harvard.edu/pph2/bgi.shtml, and to
                        import the FULL report returned by this server.
basic                   A basic variant input format with four columns: chr,
                        pos, ref, alt.
vcf                     Import vcf
CASAVA18_indels         Input format illumina indels.txt file, created by
                        CASAVA version 1.8
                        (http://www.illumina.com/support/documentation.ilmn).
                        This format imports chr, pos, ref, alt of most likely
                        genotype, and a Q score for the most likely genotype.
csv                     Import variants (chr, pos, ref, alt) in csv format, or
                        output arbitrary specified fields in csv format
tped                    Output to TPED format with the first four columns chr
                        name gen_pos pos, and the rest for genotypes. Variant
                        tools cannot import from this format because it does
                        not contain information about reference genome.
</code></pre>

<p>You can suppress descriptions to formats using option <code>-v0</code> to command @@vtools show formats</p>

<p>If you are interested in a particular format, you can</p>

<pre><code>% vtools show format basic


A basic variant input format with four columns: chr, pos, ref, alt.

Columns:
  None defined, cannot export to this format

variant:
  chr                   Chromosome
  pos                   1-based position
  ref                   Reference allele, '-' for insertion.
  alt                   Alternative allele, '-' for deletion.

Format parameters:
  chr_col               Column index for the chromosome field (default: 1)
  pos_col               Column index for the position field (default: 2)
  ref_col               Column index for the reference field (default: 3)
  alt_col               Column index for the alternative field (default: 4)
  pos_adj               Set to 1 if the input position is zero-based.
                        (default: 0)
</code></pre>

<p>From the description, we can that format <code>basic</code> import variants from a text file, with fields <code>chr</code>, <code>pos</code>, <code>ref</code>, and <code>alt</code> import from columns 1, 2, 3 and 4. The input columns could be adjust by format parameters <code>chr_col</code>, <code>pos_col</code>, <code>ref_col</code>, and <code>alt_col</code>. This format assumes a 1-based position. If the input data uses 0-based position, parameter <code>pos_adj</code> can be used to adjust input. (:exampleend:)</p>

<p>If your input file matches the description of a format, you can use this format to import data. The format should be specified by parameter <code>--format</code>, although this parameter could be ignored if file format can be detected automatically from file extension (e.g. format <code>vcf</code> will be used automatically for files with extension <code>.vcf</code> and <code>.vcf.gz</code>).</p>

<p>(:toggleexample Examples: Import variants using format <code>basic</code>:) File <code>variants.txt</code> has a list of variants as follows</p>

<pre><code>% head variants.txt 


1   203148112   T   -
1   203148168   G   A
1   203148202   G   C
1   203148224   G   A
1   203148265   GG  T
1   203148284   T   C
1   203148294   G   T
1   203148359   C   A
1   203148360   G   A
1   203148360   G   C
</code></pre>

<p>You can import variants using format <code>basic</code> as follows:</p>

<pre><code>% vtools import variants.txt --format basic --build hg19


INFO: Importing variants from variants.txt (1/1)
variants.txt: 100% [===============================================] 20 9.9K/s in 00:00:00
INFO: 20 new variants (17 SNVs, 1 insertions, 1 deletions, 1 complex variants) from 20 lines are imported.
WARNING: Sample information is not recorded for a file without genotype and sample name.
Importing genotypes: 0 0.0/s in 00:00:00                                                                                                        
Copying genotype: 0 0.0/s in 00:00:00 
</code></pre>

<p>The variants can be displayed using command <code>vtools output</code>,</p>

<pre><code>% vtools output variant chr pos ref alt -l 10


1   203148112   T   -
1   203148168   G   A
1   203148202   G   C
1   203148224   G   A
1   203148265   GG  T
1   203148284   T   C
1   203148294   G   T
1   203148359   C   A
1   203148360   G   A
1   203148360   G   C
</code></pre>

<p>(:exampleend:)</p>

<p>Format parameters can be used to adjust how data are imported using a particular format. For example, fields <code>chr</code>, <code>pos</code>, <code>ref</code>, and <code>alt</code> are by default imported from columns 1, 2, 3 and 4 using format <code>basic</code>. The input columns could be adjust by format parameters <code>chr_col</code>, <code>pos_col</code>, <code>ref_col</code>, and <code>alt_col</code>. If the input data uses 0-based position, parameter <code>pos_adj</code> can be used to adjust input.</p>

<p>(:toggleexample Examples: Use format parameters to adjust how data are imported:) If the position used in <code>variants.txt</code> is zero-based (like all data downloaded from UCSC), you can use format parameter <code>--pos_adj 1</code> to add <code>1</code> to import positions:</p>

<pre><code>% vtools init import -f
% vtools import variants.txt --format basic --pos_adj 1 --build hg19 


vtools import variants.txt --format basic --pos_adj 1 --build hg19
INFO: Importing variants from variants.txt (1/1)
variants.txt: 100% [========================================] 20 758.8/s in 00:00:00
INFO: 20 new variants (17 SNVs, 1 insertions, 1 deletions, 1 complex variants) from 20 lines are imported.
WARNING: Sample information is not recorded for a file without genotype and sample name.
Importing genotypes: 0 0.0/s in 00:00:00
Copying genotype: 0 0.0/s in 00:00:00




% vtools output variant chr pos ref alt -l 10


1   203148113   T   -
1   203148169   G   A
1   203148203   G   C
1   203148225   G   A
1   203148266   GG  T
1   203148285   T   C
1   203148295   G   T
1   203148360   C   A
1   203148361   G   A
1   203148361   G   C
</code></pre>

<p>(:exampleend:)</p>

<h3 id="sample-genotypes-sample-names-and-samples-without-genotype-option-sample-name">Sample genotypes, sample names, and samples without genotype (option <code>--sample_name</code>)</h3>

<p>An input file can have genotype data for zero (e.g. a list of variants), one, or more than one samples (many <code>.vcf</code> files), and genotypes for a sample might be stored in more than one files, it can be confusing how samples are handled in <em>variant tools</em>.</p>

<ol>
<li>A list of variants without genotype is by default imported without creating a sample. The variants will be imported to the master variant table and there is no way to trace the origin of the variants if variants from multiple files are imported.</li>
<li>A list of variants can be imported as a sample without genotype if a sample name is given. The variants will be imported to the master variant table. A sample will be created with these variants. You can use <code>vtools select variant --samples CONDITION</code> to locate variants belonging to this sample.</li>
<li>One or more samples will be created for an input file with one or more samples. Variant tools will try to use sample names specified by parameter <code>--sample_name</code>, or names from header of input file. A NULL sample name will be used if no sample name could be determined.</li>
</ol>

<ul>
<li>Sample names could be changed using command <code>vtools admin --rename_samples</code> after they are imported.</li>
<li>If genotypes for samples are stored in separate files (e.g. chromosome by chromosome), they will be imported as separate samples. If proper sample names are provided, samples with the same names (and belong to the same physical sample) could be merged using command <code>vtools admin --merge_samples</code>.</li>
</ul>

<p>(:toggleexample Examples: import a list variants and create a sample without genotype:) If there is no genotype in the input file, no sample is created by default. This is the case for the previous example where only variants are imported:</p>

<pre><code>% vtools show samples


sample_name filename
</code></pre>

<p>However, if you would like to trace what variants have been imported from each input file, you can create a <code>sample</code> (without genotype information) by providing parameter <code>--sample_name</code>.</p>

<pre><code>% vtools init import -f
% vtools import variants.txt --format basic --build hg19 --sample_name noGT


INFO: Importing variants from variants.txt (1/1)
variants.txt: 100% [==============================================] 20 9.1K/s in 00:00:00
INFO: 20 new variants (17 SNVs, 1 insertions, 1 deletions, 1 complex variants) from 20 lines are imported.
Importing genotypes: 100% [=======================================] 20 10.0/s in 00:00:02
Copying genotype: 100% [===========================================] 1 6.8K/s in 00:00:00
</code></pre>

<p>There is a sample called <code>noGT</code>,</p>

<pre><code>% vtools show samples


sample_name filename
noGT    variants.txt
</code></pre>

<p>although this sample does not have any genotype fields:</p>

<pre><code>% vtools show genotypes


sample_name filename    num_genotypes   sample_genotype_fields
noGT    variants.txt    20
</code></pre>

<p>(:exampleend:)</p>

<p>If your data is provided in VCF format, sample names are retrieved automatically from the header of input file and you usually do not have to specify names to samples.</p>

<p>(:toggleexample Examples: Import sample from a vcf files with multiple samples:)</p>

<pre><code>% vtools import CEU.vcf.gz --build hg19


INFO: Importing variants from CEU.vcf.gz (1/1)
CEU.vcf.gz: 100% [==========================================================================] 300 14.1K/s in 00:00:00
INFO: 288 new variants (288 SNVs) from 300 lines are imported.
Importing genotypes: 100% [===============================================================] 18,000 9.0K/s in 00:00:02
Copying genotype: 100% [====================================================================] 60 504.3K/s in 00:00:00
</code></pre>

<p>This file contains genotype information for 60 samples,</p>

<pre><code>% vtools show genotypes -l 10


sample_name filename    num_genotypes   sample_genotype_fields
NA06985 CEU.vcf.gz  287 GT
NA06986 CEU.vcf.gz  287 GT
NA06994 CEU.vcf.gz  287 GT
NA07000 CEU.vcf.gz  287 GT
NA07037 CEU.vcf.gz  287 GT
NA07051 CEU.vcf.gz  287 GT
NA07346 CEU.vcf.gz  288 GT
NA07347 CEU.vcf.gz  287 GT
NA07357 CEU.vcf.gz  287 GT
NA10847 CEU.vcf.gz  287 GT
(50 records omitted)
</code></pre>

<p>(:exampleend:)</p>

<p>However, when the samples are provided in multiple vcf files, the pipeline that produce these files might assign samples the same names. In this case, you can override the default name with proper sample names.</p>

<p>(:toggleexample Examples: override sample names specified in .vcf file:) If we import three samples from three files</p>

<pre><code>% vtools init import -f
% vtools import V*.vcf --build hg18


INFO: Importing variants from V1.vcf (1/3)
V1.vcf: 100% [============================================================================] 1,000 12.6K/s in 00:00:00
INFO: 989 new variants (989 SNVs) from 1,000 lines are imported.
INFO: Importing variants from V2.vcf (2/3)
V2.vcf: 100% [============================================================================] 1,000 17.9K/s in 00:00:00
INFO: 352 new variants (352 SNVs) from 995 lines are imported.
INFO: Importing variants from V3.vcf (3/3)
V3.vcf: 100% [============================================================================] 1,000 18.7K/s in 00:00:00
INFO: 270 new variants (270 SNVs) from 1,000 lines are imported.
Importing genotypes: 100% [================================================================] 2,995 1.5K/s in 00:00:02
Copying genotype: 100% [======================================================================] 3 19.8K/s in 00:00:00
INFO: 1,611 new variants (1,611 SNVs) from 2,995 lines (3 samples) are imported.
</code></pre>

<p>These samples will have the same sample name</p>

<pre><code>% vtools show samples


sample_name filename
SAMP1   V1.vcf
SAMP1   V2.vcf
SAMP1   V3.vcf
</code></pre>

<p>This can be fixed using command <code>vtools admin --rename_samples</code>, but can also be corrected during the import process</p>

<pre><code>vtools init import -f
vtools import V1.vcf --sample_name V1 --build hg18
vtools import V2.vcf --sample_name V2 --build hg18
vtools import V3.vcf --sample_name V3 --build hg18
</code></pre>

<p>The samples now have correct names</p>

<pre><code>% vtools show genotypes


sample_name filename    num_genotypes   sample_genotype_fields
V1  V1.vcf  989 GT
V2  V2.vcf  990 GT
V3  V3.vcf  988 GT
</code></pre>

<p>(:exampleend:)</p>

<h3 id="primary-and-alternative-reference-genomes-option-build">Primary and alternative reference genomes (option <code>--build</code>)</h3>

<p>Sometimes you can different batches of data that use different reference genomes. For example, your project might involve multiple batches of data over the years and variants from the old and new batches are processed using different versions of pipelines with different reference genome. In addition, you might want to a previous collection of samples or some public dataset as controls to your project, but they might use a different reference genome than your project. <em>variant tools</em> allows you to import data in two reference genomes. All you need to do is to specify the correct build of reference genome for each batch of data using option <code>--build</code>.</p>

<p>(:toggleexample Import datasets in hg18 and hg19:) Variants from <code>CEU.vcf.gz</code> uses build <code>hg18</code> of the reference genome,</p>

<pre><code>% vtools init import -f
% vtools import vcf/CEU.vcf.gz --build hg18


INFO: Importing variants from CEU.vcf.gz (1/1)
CEU.vcf.gz: 100% [==========================================] 300 6.7K/s in 00:00:00
INFO: 288 new variants (288 SNVs) from 300 lines are imported.
Importing genotypes: 100% [==============================] 18,000 8.9K/s in 00:00:02
Copying genotype: 100% [=====================================] 60 30.0/s in 00:00:02
</code></pre>

<p>Data from <code>variants.txt</code> are in <code>hg19</code>, so we use option <code>--build hg19</code> to impor them,</p>

<pre><code>% vtools import variants.txt --format basic --build hg19


WARNING: The new files uses a different reference genome (hg19) from the primary reference genome (hg18) of the project.
INFO: Adding an alternative reference genome (hg19) to the project.
INFO: Downloading liftOver tool from UCSC
liftOver: 100% [====================================] 3,383,268.0 1.3M/s in 00:00:02
INFO: Downloading liftOver chain file from UCSC
hg18ToHg19.over.chain.gz: 100% [====================] 140,346.0 132.7K/s in 00:00:01
INFO: Exporting variants in BED format
Exporting variants: 100% [=================================] 288 19.5K/s in 00:00:00
INFO: Running UCSC liftOver tool
Updating table variant: 100% [==============================] 288 6.6K/s in 00:00:00
Getting existing variants: 100% [==========================] 288 97.4K/s in 00:00:00
INFO: Importing variants from variants.txt (1/1)
variants.txt: 100% [========================================] 20 500.9/s in 00:00:00
INFO: 20 new variants (17 SNVs, 1 insertions, 1 deletions, 1 complex variants) from 20 lines are imported.
WARNING: Sample information is not recorded for a file without genotype and sample name.
Importing genotypes: 0 0.0/s in 00:00:00
Copying genotype: 0 0.0/s in 00:00:00
INFO: Mapping new variants at 20 loci from hg19 to hg18 reference genome
INFO: Downloading liftOver chain file from UCSC
hg19ToHg18.over.chain.gz: 100% [====================] 231,197.0 503.6K/s in 00:00:00
INFO: Running UCSC liftOver tool
Updating coordinates: 100% [================================] 20 563.1/s in 00:00:00
INFO: Coordinates of 20 (20 total, 0 failed to map) new variants are updated.
</code></pre>

<p>When you execute the second command, <em>variant tools</em> will</p>

<ol>
<li>Liftover the existing project to hg19 using the UCSC liftover tool</li>
<li>Get all coordinates from the hg19 input files and map them to hg18.</li>
<li>Import the hg19 data files using mapped coordinates, while keeping their hg19 coordinates as alternative coordinates.</li>
</ol>

<p>variants inputted in this way can be accessed using both reference genomes.</p>

<pre><code>% vtools output variant chr pos ref alt --build hg18 -l 10


1   533 G   C
1   41342   T   A
1   41791   G   A
1   44449   T   C
1   44539   C   T
1   44571   G   C
1   45162   C   T
1   52066   T   C
1   53534   G   A
1   75891   T   C




% vtools output variant chr pos ref alt --build hg19 -l 10


1   10533   G   C
1   51479   T   A
1   51928   G   A
1   54586   T   C
1   54676   C   T
1   54708   G   C
1   55299   C   T
1   62203   T   C
1   63671   G   A
1   86028   T   C
</code></pre>

<p>(:exampleend:)</p>

<ul>
<li><p>Because the UCSC liftover tool does not guarantee all coordinates can be mapped between reference genomes, variants that cannot be mapped back to the primary reference genome will have missing primary coordinates. <strong>These variants (with missing primary coordinates) can only be retrieved and annotated through the alternative reference genome</strong>.</p></li>

<li><p>Different variants in one reference genome might be mapped to the same variant in another (e.g. variant 90460203 and 91680930 in hg19 are both mapped to 91044656 in hg18). If two (or more) variants imported from the alternative reference genome are mapped to the same coordinates in the primary reference genome, <strong>duplicate entries will appear in the primary reference genome</strong>. If you have data in both reference genome (e.g. hg18, hg19), it is suggested that you <strong>use the more recent reference genome (e.g. hg19) as the primary reference genome</strong>, because latter reference genomes have more variants and less probability of coordinate collision.</p></li>
</ul>

<h3 id="validate-build-of-reference-genome-and-variant-positions">Validate build of reference genome and variant positions</h3>

<p>If you are uncertain about the reference genome of your input data, or if the variant positions are 0- or 1-based, you can use command <code>vtools admin --validate_build</code> to compare reference alleles with the alleles at the corresponding locations on the reference genome. You will notice a large number of mismatch variants if you have used incorrect reference genome or failed to adjust positions of variants from 0-based positions to 1-based.</p>

<p>(:toggleexample Examples: validate if the correct reference genome has been specified :) Let us import variants from <code>variants.txt</code> using hg18,</p>

<pre><code>% vtools init import -f
% vtools import variants.txt --format basic --build hg18
% vtools admin --validate_build


Validate reference alleles: 100% [=========================] 20/13 2.3K/s in 00:00:00
INFO: 19 non-insertion variants are checked. 13 mismatch variants found.
</code></pre>

<p>13 out of 19 non-insertion variants mismatch so we let us try using the <code>hg19</code> build of reference genome,</p>

<pre><code>% vtools init import -f
% vtools import variants.txt --format basic --build hg19
% vtools admin --validate_build


Validate reference alleles: 100% [=========================] 20/1 976.6/s in 00:00:00
INFO: 19 non-insertion variants are checked. 1 mismatch variants found.
</code></pre>

<p>We still have 1 mismatch variant but the variants are more likely created using <code>hg19</code>. The mismatch variants can be found in the project log file, or be listed with option <code>-v2</code>. It should be verified or removed from the project. (:exampleend:)</p>

<h3 id="performance-optimization-option-jobs-and-other-techniques">Performance optimization (option <code>--jobs</code> and other techniques)</h3>

<p><em>variant tools</em> by default uses multiple processes to load data (<code>--jobs 4</code>), which works well for most datasets under most computing environments. Compared to importing data using a single process (<code>--jobs 1</code>), it performs slightly worse for small datasets, but is significantly faster for large files, especially when multiple files are imported. If you have high speed harddrive (e.g. disk array, SSD), you can use more processes to import data although more processes do not automatically translate to better performance due to the overhead of multi-processing and disk I/O scheduling.</p>

<p>Depending on your computing environment (amount of RAM, speed of harddrive), <em>vtools import</em> could be optimized by setting appropriate runtime options that are compatible with your computer hardware environment. One optimization is to <strong>temporarily</strong> set an <em>in-ram journal mode</em>, and in-ram cache if the computer has large RAM:</p>

<pre><code>% vtools admin --set_runtime_option 'sqlite_pragma=synchronous=OFF,journal_mode=MEMORY,cachesize=10000'
</code></pre>

<p>The in-ram journal mode will lessen the disk I/O burden, but compromises the database&rsquo;s failure recovery capability. It is therefore recommended to revert the setting after the import is done.</p>

<p>Another optimization is to move the temporary directory to a large, separate physical disk. By default the temporary directory locates in one of the system&rsquo;s temporary folders (e.g., <code>/tmp</code> or <code>/var/tmp</code> for Linux). &ldquo;&lsquo;Moving <code>$temp_dir</code> to a different physical disk will greatly improve the performance of <code>vtools import</code> (and <code>vtools associate</code>) because of significantly less movement of reading head of harddrives.</p>

<pre><code>% vtools admin --set_runtime_option 'temp_dir=/home/HD1/tmp_some_random_name'
</code></pre>

<p>The folder&rsquo;s name is arbitrary. It will be created each time a command starts and deleted upon completion of the command.</p>

<p>It is very important to make sure that for import and associate commands there is sufficient disk space in the temporary directory, since potentially large temp files will be generated by these commands.</p>

<p>After data import is finished, we will set <em>journal_mode</em> back to default, for reasons previously discussed.</p>

<pre><code>% vtools admin --reset_runtime_option sqlite_pragma
</code></pre>

<p>Finally, if you need to import a large amount of data from multiple files, it can be helpful to create multiple project, import files one by one (or group by group), and create a parent project from these children projects.</p>

<p><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Tutorial.1000Genomes?action=edit" target="_blank">This tutorial</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Tutorial.1000Genomes?action=edit" target="_blank">?</a> demonstrates how to import all genotype data from the 1000 genomes project using different techniques. It contains some tips on how to import a large amount of data (the uncompressed 1000 genome data exceed 1 Tb).</p>

<h2 id="importing-data-in-particular-formats">Importing data in particular formats</h2>

<h3 id="importing-vcf-files-format-vcf">Importing VCF files (format <code>vcf</code>)</h3>

<p>VCF is the most commonly used format to store genetic variants and sample genotypes from next-gen sequencing studies. <em>variant tools</em> can import data from plain (<code>.vcf</code>) or compressed (<code>.vcf.gz</code>) vcf format (version 4.0 or higher). Depending on the number of samples a vcf file stores, <em>variant tools</em> can import only variants (even if the vcf file contains sample genotypes, see an example below), import variants with a sample without genotype, and import all samples in a vcf file using specified sample names or sample names obtained from the meta information (see examples above). A limitation is that <em>variant tools</em> ignores phase information of genotypes.</p>

<h4 id="import-variant-and-genotype-info-fields">Import variant and genotype info fields</h4>

<p>The vcf format can store arbitary variant and genotype information fields. <em>variant tools</em> by default does not import any variant and info fields. However, the default configuration file does provide a variant info field <code>DP</code> and a genotype info field <code>DP_geno</code>, which you may import using option <code>--var_info</code> and <code>--geno_info</code>. To import variant or genotype fields from a vcf file, you need to</p>

<ol>
<li>know what fields are available from the input vcf file. That is to say, you need to open a file, read its header, and determine what fields are provided.</li>
<li>use command <code>vtools show format vcf</code> to see what fields are available. If the field you would like to import is listed, you can add it to parameter <code>--var_info</code> or <code>--geno_info</code> to import it. Otherwise, you can <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.New?action=edit" target="_blank">customize your format file</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.New?action=edit" target="_blank">?</a> to add the field.</li>

<li><p>use command <code>vtools import input_file.vcf --var_info var_fields --geno_info geno_fields</code> to import variants, genotypes and related fields.</p></li>

<li><p>If your vcf file is bgzipped and tabix indexed (you can run compress and index your vcf file using commands <code>bgzip</code> and <code>tabix</code>), you can use command <code>vtools show track FILENAME.vcf.gz</code> to get details of the vcf file. The <a href="http://varianttools.sourceforge.net/Vtools/Commands#toc10" target="_blank"><code>track</code></a> function can also be used to retrieve such information when needed so you do not have to import variant info fields into the project.</p></li>

<li><p>If your vcf file contains novel variant and/or geno info fields, or if you would like to import all variant and genotype info fields into the project, you can create a customized <code>.fmt</code> file to import these. This process can be simplied using pipeline <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Pipeline.ImportVcf?action=edit" target="_blank">import_vcf</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Pipeline.ImportVcf?action=edit" target="_blank">?</a>. The command to use is similar to <code>vtools execute import_vcf --input my_file.vcf --output myvcf.fmt --build hg19</code>.</p></li>
</ol>

<p>(:toggleexample Examples: importing variant and genotype info fields:) If we have a look at the header of <code>CEU.vcf.gz</code>, we can see</p>

<pre><code>% gzcat CEU.vcf.gz | head -14


##fileformat=VCFv4.0
##INFO=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Total Depth&quot;&gt;
##INFO=&lt;ID=HM2,Number=0,Type=Flag,Description=&quot;HapMap2 membership&quot;&gt;
##INFO=&lt;ID=HM3,Number=0,Type=Flag,Description=&quot;HapMap3 membership&quot;&gt;
##INFO=&lt;ID=AA,Number=1,Type=String,Description=&quot;Ancestral Allele, ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/pilot_data/technical/reference/ancestral_alignments/README&quot;&gt;
##reference=human_b36_both.fasta
##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
##FORMAT=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Read Depth&quot;&gt;
##FORMAT=&lt;ID=CB,Number=1,Type=String,Description=&quot;Called by S(Sanger), M(UMich), B(BI)&quot;&gt;
##rsIDs=dbSNP b129 mapped to NCBI 36.3, August 10, 2009
##INFO=&lt;ID=AC,Number=.,Type=Integer,Description=&quot;Allele count in genotypes&quot;&gt;
##INFO=&lt;ID=AN,Number=1,Type=Integer,Description=&quot;Total number of alleles in called genotypes&quot;&gt;
#CHROM  POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  NA06985 NA06986 NA06994 NA07000 NA07037 NA07051 NA07346 NA07347 NA07357 NA10847 NA10851 NA11829 NA11830 NA11831 NA11832 NA11840 NA11881 NA11894 NA11918 NA11919 NA11920 NA11931 NA11992 NA11993 NA11994 NA11995 NA12003 NA12004 NA12005 NA12006 NA12043 NA12044 NA12045 NA12144 NA12154 NA12155 NA12156 NA12234 NA12249 NA12287 NA12414 NA12489 NA12716 NA12717 NA12749 NA12750 NA12751 NA12760 NA12761 NA12762 NA12763 NA12776 NA12812 NA12813 NA12814 NA12815 NA12828 NA12872 NA12873 NA12874
1   533 .   G   C   .   PASS    AA=.;AC=6;AN=120;DP=423 GT:DP:CB    0|0:6:SMB   0|0:14:SMB  0|0:4:SMB   0|0:3:SMB   0|0:7:SMB   0|0:4:SMB   1|0:6:MB    0|0:3:SMB   0|0:13:SMB  0|0:1:SMB   0|0:14:SMB  0|0:10:SMB  0|0:6:SB    0|0:2:SMB   0|0:6:SMB   0|0:4:SMB   0|0:2:SMB   0|0:15:SMB  0|0:2:SMB   0|0:1:SMB   0|0:26:SMB  0|0:6:SMB   0|1:14:MB   0|0:5:SMB   0|0:3:SMB   0|0:20:SMB  0|0:3:SMB   0|0:2:SMB   0|0:4:SMB   0|0:12:SMB  0|0:1:SMB   0|0:7:SMB   0|0:2:SMB   0|0:25:SMB  0|0:9:SMB   0|1:1:MB    0|0:9:SMB   0|0:1:SMB   0|0:6:SMB   0|0:12:SMB  0|0:7:SMB   0|0:18:SMB  0|0:2:SMB   0|0:2:SM    0|0:38:SMB  0|0:3:SM    0|0:3:SMB   0|0:5:SMB   0|0:5:SMB   0|0:3:SMB   0|0:0:MB    0|0:5:SMB   0|0:7:SMB   0|0:0:SMB   0|0:6:SMB   1|0:5:SMB   0|0:4:MB    0|0:5:SMB   1|0:5:MB    0|1:9:SMB
</code></pre>

<p>we can see that this file contains genotype info <code>AA</code>, <code>AC</code>, <code>AN</code>, and <code>DP</code>, and genotype info <code>DP</code> and <code>CB</code>. If you check the output of command <code>vtools show format vcf</code>, you could find fields <code>AA</code>, <code>AC</code>, <code>AN</code>, and <code>DP</code> that parse the INFO column of the input file, and field <code>DP_geno</code> that parses the genotype columns.</p>

<pre><code>% vtools init import -f
% vtools import CEU.vcf.gz --build hg18 --var_info AA AC AN DP --geno_info DP_geno


INFO: Importing variants from CEU.vcf.gz (1/1)
CEU.vcf.gz: 100% [====================================================] 300 11.4K/s in 00:00:00
INFO: 288 new variants (288 SNVs) from 300 lines are imported.
Importing genotypes: 100% [=========================================] 18,000 9.0K/s in 00:00:02
Copying genotype: 100% [==============================================] 60 473.0K/s in 00:00:00
</code></pre>

<p>The imported data now has variant info field <code>AA</code>, <code>AC</code>, <code>AN</code>, and <code>DP</code>,</p>

<pre><code>% vtools show fields


variant.chr
variant.pos
variant.ref
variant.alt
variant.AA
variant.AC
variant.AN
variant.DP
</code></pre>

<p>and genotype info field DP_genw</p>

<pre><code>% vtools show genotypes -l 5


sample_name filename    num_genotypes   sample_genotype_fields
NA06985 CEU.vcf.gz  287 GT,DP_geno
NA06986 CEU.vcf.gz  287 GT,DP_geno
NA06994 CEU.vcf.gz  287 GT,DP_geno
NA07000 CEU.vcf.gz  287 GT,DP_geno
NA07037 CEU.vcf.gz  287 GT,DP_geno
(55 records omitted)
</code></pre>

<p>(:exampleend:)</p>

<p>The above example uses fields <code>DP</code> and <code>DP_geno</code> to extract information from the <code>INFO</code> and genotype columns of the input vcf file. However, it is not always clear whether a field should be imported as variant info or genotype info field. For example,</p>

<ul>
<li>If genotypes for multiple samples are saved in separate vcf files, each has a <code>DP</code> (depth) field as variant info, the variant info (field <code>DP</code>) is actually describing depth of sample genotypes and should be imported as <code>--geno_info DP</code>.</li>
<li>If variant info <code>DP</code> lists the total depth of the genotype of all samples, it should be imported as <code>--var_info DP</code>.</li>
<li>If <code>DP</code> appear in the <code>FORMAT</code> column, it should be imported using field <code>DP_geno</code> and used as <code>--geno_info DP_geno</code>.</li>
</ul>

<p>This is demonstrated in the following example:</p>

<p>(:toggleexample Examples: import variant info field <code>DP</code> as genotype info:) If we have a look at the meta information of <code>V1.vcf</code>,</p>

<pre><code>% head -14 V1.vcf


##fileformat=VCFv4.0
##INFO=&lt;ID=NS,Number=1,Type=Integer,Description=&quot;Number of Samples With Data&quot;&gt;
##INFO=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Total Depth&quot;&gt;
##INFO=&lt;ID=AF,Number=.,Type=Float,Description=&quot;Allele Frequency&quot;&gt;
##INFO=&lt;ID=AA,Number=1,Type=String,Description=&quot;Ancestral Allele&quot;&gt;
##INFO=&lt;ID=DB,Number=0,Type=Flag,Description=&quot;dbSNP membership, build 129&quot;&gt;
##INFO=&lt;ID=H2,Number=0,Type=Flag,Description=&quot;HapMap2 membership&quot;&gt;
##FILTER=&lt;ID=q10,Description=&quot;Quality below 10&quot;&gt;
##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
##FORMAT=&lt;ID=GQ,Number=1,Type=Integer,Description=&quot;Genotype Quality&quot;&gt;
##FORMAT=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Read Depth&quot;&gt;
##FORMAT=&lt;ID=HQ,Number=2,Type=Integer,Description=&quot;Haplotype Quality&quot;&gt;
#CHROM  POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  SAMP1
1   4540    .   G   A   32  PASS    DP=6;NS=1   GT  0/1
</code></pre>

<p><code>DP</code> is listed as a INFO field, along with <code>NS</code>, we can import <code>V1.vcf</code>, <code>V2.vcf</code> and <code>V3.vcf</code> using command</p>

<pre><code>% vtools init import -f
% vtools import V*.vcf --build hg19 --var_info DP NS


INFO: Importing variants from V1.vcf (1/3)
V1.vcf: 100% [======================================================] 1,000 11.3K/s in 00:00:00
INFO: 989 new variants (989 SNVs) from 1,000 lines are imported.
INFO: Importing variants from V2.vcf (2/3)
V2.vcf: 100% [=======================================================] 1,000 9.5K/s in 00:00:00
INFO: 352 new variants (990 SNVs) from 1,000 lines are imported.
INFO: Importing variants from V3.vcf (3/3)
V3.vcf: 100% [=======================================================] 1,000 9.1K/s in 00:00:00
INFO: 270 new variants (988 SNVs) from 1,000 lines are imported.
Importing genotypes: 100% [==========================================] 3,000 1.5K/s in 00:00:02
Copying genotype: 100% [================================================] 3 20.0K/s in 00:00:00
INFO: 1,611 new variants (2,967 SNVs) from 3,000 lines (3 samples) are imported.
</code></pre>

<p>but <code>DP</code> and <code>NS</code> are imported for each <strong>variant</strong>, not <em>&lsquo;genotype</em>, so only the first occurance of <code>DP</code> and <code>NS</code> will be imported. The correct way to import these two fields is to treat them as genotype fields,</p>

<pre><code>% vtools init import -f
% vtools import V*.vcf --build hg19 --geno_info DP NS


INFO: Importing variants from V1.vcf (1/3)
V1.vcf: 100% [======================================================] 1,000 11.1K/s in 00:00:00
INFO: 989 new variants (989 SNVs) from 1,000 lines are imported.
INFO: Importing variants from V2.vcf (2/3)
V2.vcf: 100% [======================================================] 1,000 15.8K/s in 00:00:00
INFO: 352 new variants (352 SNVs) from 995 lines are imported.
INFO: Importing variants from V3.vcf (3/3)
V3.vcf: 100% [======================================================] 1,000 17.0K/s in 00:00:00
INFO: 270 new variants (270 SNVs) from 1,000 lines are imported.
Importing genotypes: 100% [==========================================] 2,995 1.5K/s in 00:00:02
Copying genotype: 100% [================================================] 3 21.0K/s in 00:00:00
INFO: 1,611 new variants (1,611 SNVs) from 2,995 lines (3 samples) are imported.
</code></pre>

<p><code>DP</code> and <code>NS</code> will be imported as genotype info for each genotype,</p>

<pre><code>% vtools show fields


variant.chr
variant.pos
variant.ref
variant.alt




% vtools show genotypes


sample_name filename    num_genotypes   sample_genotype_fields
SAMP1   V1.vcf  989 GT,DP,NS
SAMP1   V2.vcf  990 GT,DP,NS
SAMP1   V3.vcf  988 GT,DP,NS
</code></pre>

<p>If you need total depth, you can obtained from the corresponding genotype field</p>

<pre><code>% vtools update variant --from_stat 'DP=sum(DP)' 'NS=sum(NS)'


Counting variants: 100% [===============================================] 3 145.1/s in 00:00:00
INFO: Adding variant info field DP
INFO: Adding variant info field NS
Updating variant: 100% [============================================] 1,611 47.8K/s in 00:00:00
INFO: 1611 records are updated
</code></pre>

<p>Here <code>DP=sum(DP)</code> looks strange but the first <code>DP</code> is a new variant info and the second <code>DP</code> is the existing genotype info. Now, variant info field <code>DP</code> stands for total depth and <code>NS</code> stands for number of times a variant shows up in the samples.</p>

<pre><code>% output variant chr pos ref alt DP NS -l 5


1   4540    G   A   15  2
1   5683    G   T   7   1
1   5966    T   G   76  3
1   6241    T   C   41  2
1   9992    C   T   24  2
</code></pre>

<p>(:exampleend:)</p>

<h4 id="using-customized-or-alternative-fields-to-import-genotypes">Using customized or alternative fields to import genotypes</h4>

<p>The vcf format accepts a number of parameters to customize the way how genotypes are imported. For example, the <code>--geno</code> option accepts a genotype field. The default <code>GT</code> field import all genotypes assuming that <code>GT</code> is the first field in the genotype columns. However,</p>

<ul>
<li>You can use option <code>--geno</code> without value to specify an empty genotype field so that variant tools will not process any genotype.</li>
<li>In <code>vcf.fmt</code> we import wild-type genotypes as numeric value 0. This is important for the purpose of association studies. If you are only interested in exploring variants of your sample, and you do not care to discriminate wild-type genotypes from missing genotype calls, you may set <code>--wildtype_value None</code>, which will leave out all wild-type genotypes, resulting in much quicker import and smaller database size.</li>
<li>The <code>GT</code> field assumes that <code>GT</code> is the first field in the <code>FORMAT</code> column. However, some variant-calling software produce non-standard vcf files that do not follow this standard. In this case, an alternative field <code>safe_GT</code> can be used (e.g. <code>--geno safe_GT</code>) to import genotypes according to the location of <code>GT</code> in the <code>FORMAT</code> string.</li>
</ul>

<p>(:toggleexample Examples: Ignore genotypes of input vcf file:)</p>

<pre><code>% vtools init import -f
% vtools import CEU.vcf.gz --build hg18 --var_info AA AC AN DP --geno


INFO: Importing variants from CEU.vcf.gz (1/1)
CEU.vcf.gz: 100% [====================================================] 300 11.2K/s in 00:00:00
INFO: 288 new variants (288 SNVs) from 300 lines are imported.
WARNING: Sample information is not recorded for a file without genotype and sample name.
Importing genotypes: 0 0.0/s in 00:00:00
Copying genotype: 0 0.0/s in 00:00:00
</code></pre>

<p>With this command, no genotype is imported</p>

<pre><code>% vtools show genotypes 


filename    sample_name num_genotypes   sample_genotype_fields
</code></pre>

<p>(:exampleend:)</p>

<h4 id="using-a-customized-format-file-to-import-additional-fields">Using a customized format file to import additional fields</h4>

<p>If you have a field that you would like to import, but does not exist in <code>vcf.fmt</code>, you can add it to a customized format file and use this format file to import the data.</p>

<p>(:toggleexample Examples: Using a customized .fmt file to import additional fields:) For example, the <code>CB</code> genotype field is commonly used and is not defined in <code>vcf.fmt</code>, according to the meta information, this field is a string. You can then copy <code>~/.variant_tools/format/vcf.fmt</code> as <code>my_vcf.fmt</code> and add</p>

<pre><code>[CB_geno]
index=9, 10:
adj=FieldFromFormat('CB', ':')
type=VARCHAR(3)
comment=Called by S(Sanger), M(UMich), B(BI)
</code></pre>

<p>at the end of this file, and import this field using</p>

<pre><code>% vtools init import -f
% vtools import CEU.vcf.gz --format my_vcf.fmt --build hg18 --var_info AA AC AN DP --geno_info DP_geno CB_geno


INFO: Importing variants from CEU.vcf.gz (1/1)
CEU.vcf.gz: 100% [====================================================] 300 12.4K/s in 00:00:00
INFO: 288 new variants (288 SNVs) from 300 lines are imported.
Importing genotypes: 100% [=========================================] 18,000 9.0K/s in 00:00:02
Copying genotype: 100% [==============================================] 60 484.0K/s in 00:00:00




% vtools show genotypes -l5


sample_name filename    num_genotypes   sample_genotype_fields
NA06985 CEU.vcf.gz  287 GT,DP_geno,CB_geno
NA06986 CEU.vcf.gz  287 GT,DP_geno,CB_geno
NA06994 CEU.vcf.gz  287 GT,DP_geno,CB_geno
NA07000 CEU.vcf.gz  287 GT,DP_geno,CB_geno
NA07037 CEU.vcf.gz  287 GT,DP_geno,CB_geno
(55 records omitted)
</code></pre>

<p>(:exampleend:)</p>

<h3 id="import-data-in-other-formats">Import data in other formats</h3>

<p><em>variant tools</em> can import variant and genotypes data in a variety of input formats. Due to the flexibility of the format specification system, it can import variants and genotypes from data that have non-standard (e.g. 0-based index), unclear or even missing information on fields <code>chr</code>, <code>pos</code>, <code>ref</code>, and <code>alt</code>. For example, variant tools can import variants with just rsname by querying the dbSNP database for variant information, and variants with only observed slleles by querying the reference genome for reference and alternative alleles. Please check <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.HomePage?action=edit" target="_blank">here</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.HomePage?action=edit" target="_blank">?</a> for a list of supported file formats. If you cannot find a format that fits your need, you can write a <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.New?action=edit" target="_blank">format description file</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Format.New?action=edit" target="_blank">?</a> by yourself. We can assist you in this process if you could send us a sample data file.</p>


    
    
        <div class="chevrons">
    <div id="navigation">
<a class="nav nav-prev" href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/init/" title="Init"> <i class="fa fa-chevron-left"></i><label>Init</label></a>
    <a class="nav nav-next" href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/update/" title="Update" style="margin-right: 0px;"><label>Update</label><i class="fa fa-chevron-right"></i></a></div>
  </div>

  </section>
</article>

<footer>

<div class="footline">
    

    

  

    <div class="github-link">
      <a href="Documentation/vtools%20Commands/import.md" target="blank"><i class="fa fa-code-fork"></i>
        Improve this page</a>
    </div>
  </div>


	<div>


  
    Create a content/_footer.md file to customize the footer content
  



	</div>
</footer>

<script src="https://vatlab.github.io/vat-docs/js/clipboard.min.js"></script>

<link href="https://vatlab.github.io/vat-docs/css/featherlight.min.css" rel="stylesheet">
<script src="https://vatlab.github.io/vat-docs/js/featherlight.min.js"></script>



<script src="https://vatlab.github.io/vat-docs/theme-flex/script.js"></script>

    

    
    

    
  </body>
</html>