<!DOCTYPE html>
<html>
  <head>
    <title>vatlab</title>
    
      <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
<meta name="generator" content="Hugo 0.36.1" />
<title>ChangeLog :: vatlab</title>
<link rel="shortcut icon" href="https://vatlab.github.io/vat-docs/images/favicon.png" type="image/x-icon" />
<link href="https://vatlab.github.io/vat-docs/css/font-awesome.min.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/css/nucleus.css" rel="stylesheet">
<link href="https://vatlab.github.io/vat-docs/theme-flex/style.css" rel="stylesheet">

<link rel="stylesheet" href="https://vatlab.github.io/vat-docs/css/bootstrap.min.css">
<script src="https://vatlab.github.io/vat-docs/js/jquery-2.x.min.js"></script>
<script type="text/javascript">
      var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
</script>
<meta name="description" content="">


    
  </head>
  <body data-url="/development/changelog/">
    
      <header>
  <div class="logo">
    
	
  
    <a class="baselink" href="https://vatlab.github.io/vat-docs/">vatlab</a>
  

  </div>
  <div class="burger"><a href="javascript:void(0);" style="font-size:15px;">&#9776;</a></div>
</header>
<article>
  <aside>
    <ul class="menu">
          <li data-nav-id="/" class="dd-item">
          <a href="https://vatlab.github.io/vat-docs/">
            <i class="fa fa-fw fa-home"></i>
          </a>
          </li>
    <li data-nav-id="/install/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/install/">Installation</a>
      </div>
    </li>
    <li data-nav-id="/documentation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/">Documentation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/keyconcepts/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/keyconcepts/">
            Key concepts
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/tutorials/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/tutorials/">
            Tutorials
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/help/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/help/">
            Help
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/vtools-commands/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/">Vtools commands</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools-commands/init/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/init/">
            Init
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/import/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/import/">
            Import
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/update/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/update/">
            Update
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/phenotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/phenotype/">
            Phenotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/liftover/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/liftover/">
            lifttover
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/show/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/show/">
            show
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/use/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/use/">
            use
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/select/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/select/">
            select
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/exclude/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/exclude/">
            exclude
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/compare/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/compare/">
            compare
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/associate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/associate/">
            associate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/output/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/output/">
            output
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/export/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/export/">
            export
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/remove/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/remove/">
            remove
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/execute/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/execute/">
            execute
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools-commands/admin/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools-commands/admin/">
            admin
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/functions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/functions/">Functions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/functions/ref_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/ref_sequence/">
            ref_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/mut_sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/mut_sequence/">
            mut_sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/track/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/track/">
            track
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/genotype/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/genotype/">
            Genotype
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/samples/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/samples/">
            samples
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/in_table/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/in_table/">
            in_table
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/functions/otherfunctions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/functions/otherfunctions/">
            Other functions
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/vtools_report/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/">Vtools_report</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/vtools_report/avg_depth/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/avg_depth/">
            avg_depth
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/discordance_rate/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/discordance_rate/">
            discordance_rate
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sam_info/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sam_info/">
            sam_info
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/sequence/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/sequence/">
            sequence
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/trans_ratio/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/trans_ratio/">
            trans_ratio
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/variant_stat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/variant_stat/">
            variant_stat
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/inbreeding_coefficient/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/inbreeding_coefficient/">
            inbreeding_coefficient
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/meta_analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/meta_analysis/">
            meta_analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_fields/">
            plot_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_geno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_geno_fields/">
            plot_geno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_pheno_fields/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_pheno_fields/">
            plot_pheno_fields
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/vtools_report/plot_association/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/vtools_report/plot_association/">
            plot_association
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/customization/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/">Customization</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/format/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/format/">
            Format
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/">Simulation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/custimzedsimu/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/custimzedsimu/">
            Customized simulations
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/customization/simulation/models/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/">Models</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/customization/simulation/models/peng1/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/peng1/">
            Peng2014_ex1
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/customization/simulation/models/pen2/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/customization/simulation/models/pen2/">
            Peng2014_ex2
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/">Pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/customizedpipeline/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/customizedpipeline/">
            Customized pipeline
          </a>
        </div>
    </li>
    <li data-nav-id="/documentation/pipelines/variant-calling/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/">Variant calling</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_hg19/">
            bwa_gatk33_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk33_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk33_b37/">
            bat_gatk33_b37
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_hg19/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_hg19/">
            bwa_gatk28_hg19
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/variant-calling/bwa_gatk28_b37/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/variant-calling/bwa_gatk28_b37/">
            bwa_gatk28_b37
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/pipelines/other-pipelines/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/">Other pipelines</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/pipelines/other-pipelines/import_vcf/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/import_vcf/">
            import_vcf
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/anno_utils/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/anno_utils/">
            anno_utils
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/transmission/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/transmission/">
            transmission
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/illumina/">
            Illumnina
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/annovar/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/annovar/">
            ANNOVAR
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/snpeff/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/snpeff/">
            snpEff
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/pipelines/other-pipelines/king/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/pipelines/other-pipelines/king/">
            KING
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/documentation/utilities/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/documentation/utilities/">Utilities</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/documentation/utilities/report_progress/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/report_progress/">
            report_progress
          </a>
        </div>
    </li>
      <li data-nav-id="/documentation/utilities/_md5sumd/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/documentation/utilities/_md5sumd/">
            md5sumd
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/">Application</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
    <li data-nav-id="/applications/annotation/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/">Annotation</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/tutorial/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/tutorial/">
            Tutorial
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/customized/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/customized/">
            Customized annotation databases
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/annotation/genes/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/">Genes</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/genes/ccdsgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/ccdsgene/">
            ccdsGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/hugogene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/hugogene/">
            Hugo gene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/entrezgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/entrezgene/">
            EntrezGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/knowngene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/knowngene/">
            knownGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/refgene/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/refgene/">
            refGene
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/genes/cancergenecensus/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/genes/cancergenecensus/">
            CancerGeneCensus
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/variants/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/">Variants</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/variants/cosmic/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/cosmic/">
            COSMIC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbnsfp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbnsfp/">
            sbNSFP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/dbsnp/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/dbsnp/">
            dbSNP
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/linkagephysicalmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/linkagephysicalmap/">
            linkage-physical map
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/databaseof/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/databaseof/">
            Database of Genomic Variants
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/exomesequencingpro/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/exomesequencingpro/">
            Exome Sequencing Project
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/gwas/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/gwas/">
            gwasCatalog
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/hapmap/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/hapmap/">
            HapMap
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/variants/thousand/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/variants/thousand/">
            Thousand Genomes
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/regions/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/">Regions</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/regions/low-complexity-regions/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/low-complexity-regions/">
            Low complexity regions
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/phast-cons/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/phast-cons/">
            Phast cons
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/genomic-super-dups/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/genomic-super-dups/">
            Genomic super dups
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/regions/cytoband/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/regions/cytoband/">
            cytoBand
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/annotation/misc/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/">Misc</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/annotation/misc/illumina/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/illumina/">
            Illumina
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/keggpathway/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/keggpathway/">
            keggPathway
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/annotation/misc/encode/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/annotation/misc/encode/">
            ENCODE
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
      <li data-nav-id="/applications/discovery/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/discovery/">
            Discovery
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/">Association</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/data-explorationqc/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/data-explorationqc/">
            Data exploration/QC
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/association-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/association-analysis/">
            Association analysis
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/post-assoc-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/post-assoc-analysis/">
            Post-assoc analysis
          </a>
        </div>
    </li>
    <li data-nav-id="/applications/association/basic-data-statistics/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/">Basic data statistics</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/basic-data-statistics/group-statwrite/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/basic-data-statistics/group-statwrite/">
            Group stat&amp;Write
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-variant-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/">Single variant analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-variant-analysis/fisher-exact-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-variant-analysis/fisher-exact-test/">
            Fisher exact test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/single-gene-association-methods/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/">Single gene association methods</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/single-gene-association-methods/introduction/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/introduction/">
            Introduction
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/cmc-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/cmc-test/">
            CMC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/c_alpha-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/c_alpha-test/">
            C-alpha test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/kbac-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/kbac-test/">
            KBAC test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rbt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rbt-test/">
            RBT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/rarecover-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/rarecover-test/">
            RareCover test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/vt-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/vt-test/">
            VT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/wss-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/wss-test/">
            WSS test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/single-gene-association-methods/asum-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/single-gene-association-methods/asum-test/">
            aSum test
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/jointconditional-association-analysis/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/">Joint/Conditional association analysis</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/collapsing-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/collapsing-methods/">
            Collapsing methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/aggregation-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/aggregation-methods/">
            Aggregation methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/weighted-methods/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/weighted-methods/">
            Weighted methods
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/variable-thresholds/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/variable-thresholds/">
            Variable thresholds
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/skat-test/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/skat-test/">
            SKAT test
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/jointconditional-association-analysis/metaskat-analysis/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/jointconditional-association-analysis/metaskat-analysis/">
            MetaSKAT analysis
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/vat-stacking/" class="dd-item
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/vat-stacking/">VAT stacking</a>
      </div>
    </li>
    <li data-nav-id="/applications/association/create-your-test/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/">Create your test!</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/create-your-test/running-r-programs/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/running-r-programs/">
            Running R programs
          </a>
        </div>
    </li>
      <li data-nav-id="/applications/association/create-your-test/programming-in-vat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/create-your-test/programming-in-vat/">
            Programming in VAT
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/applications/association/view-association-result/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/">View association result</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/applications/association/view-association-result/qq-plot-and-manhattan-plot/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/applications/association/view-association-result/qq-plot-and-manhattan-plot/">
            QQ plot and Manhattan plot
          </a>
        </div>
    </li>
        </ul>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li data-nav-id="/development/" class="dd-item parent haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/development/">Development</a>
            <i class="fa fa-angle-down fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/development/changelog/" class="dd-item active">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/changelog/">
            ChangeLog
          </a>
        </div>
    </li>
      <li data-nav-id="/development/getinvolved/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/development/getinvolved/">
            Get Involved
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/administration/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/administration/">Administration</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/administration/allrecent/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/allrecent/">
            All recent changes
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/autolink/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/autolink/">
            Autolink
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/stylesheet/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/stylesheet/">
            Style sheet
          </a>
        </div>
    </li>
      <li data-nav-id="/administration/trafficstat/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/administration/trafficstat/">
            Traffic statistics
          </a>
        </div>
    </li>
        </ul>
    </li>
    <li data-nav-id="/under-development/" class="dd-item haschildren
        ">
      <div>
      <a href="https://vatlab.github.io/vat-docs/under-development/">Under development</a><i class="fa fa-angle-right fa-lg category-icon"></i>
      </div>
        <ul>
      <li data-nav-id="/under-development/faq/" class="dd-item">
        <div>
          <a href="https://vatlab.github.io/vat-docs/under-development/faq/">
            FAQ
          </a>
        </div>
    </li>
        </ul>
    </li>




    </ul>
    <section>
    </section>
  </aside>
  <section class="page">
    
    <div class="nav-select">
      <center>Navigation : 
        <select onchange="javascript:location.href = this.value;">
          
    <option value="/install/" >
   Installation</option>
    <option value="/documentation/" >
   Documentation</option>
    <option value="/applications/" >
   Application</option>
    <option value="/development/" >
   Development</option> 
      <option value="/development/changelog/"  selected>- ChangeLog</option>
      <option value="/development/getinvolved/" >- Get Involved</option>
  
    <option value="/administration/" >
   Administration</option>
    <option value="/under-development/" >
   Under development</option>



        </select>
      </center>
    </div>
      <div>
        <div class="searchbox">
          <input data-search-input id="search-by" type="text" placeholder="Search...">
        </div>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/lunr.min.js"></script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/auto-complete.js"></script>
        <link href="https://vatlab.github.io/vat-docs/css/auto-complete.css" rel="stylesheet">
        <script type="text/javascript">
          
              var baseurl = "https:\/\/vatlab.github.io\/vat-docs\/";
          
        </script>
        <script type="text/javascript" src="https://vatlab.github.io/vat-docs/js/search.js"></script>
      </div>
    

    <h1>ChangeLog</h1>
    
    
    
    

<h1 id="change-log-of-variant-tools">Change Log of variant tools</h1>

<h3 id="version-2-7-0-released-on-jan-20-2016">Version 2.7.0 (Released on Jan 20, 2016)</h3>

<p>MAJOR NEW FEATURES:</p>

<ul>
<li>Normalize variants using reference sequence information before importing variants.</li>
<li>New file format for spec file of variant pipeline tools.</li>
<li>Support arbitrary reference genome.</li>
</ul>

<p>BUG FIXES:</p>

<ul>
<li>Fix importing wildtype genotypes in some cases.</li>
</ul>

<h3 id="version-2-6-1-released-on-jan-15-2015">Version 2.6.1 (Released on Jan 15, 2015)</h3>

<ul>
<li>Fix a few small bugs introduced in 2.6.0.</li>
<li>Fix compatibility issues with Python 3.</li>
</ul>

<h3 id="version-2-6-0-released-on-dec-15-2014">Version 2.6.0 (Released on Dec 15, 2014)</h3>

<p>NEW FEATURES:</p>

<ul>
<li>Major cleanup of the pipeline code (<code>vtools execute</code> and <code>vtools simulate</code>), especially for the pipeline action interface.</li>
<li>Add command <code>vtools show actions</code> and <code>vtools show action ACTION</code>.</li>
<li>Select annotation databases automatically (the latest version for project annotation database) for command <code>vtools use annoDB</code> when no version information is given.</li>
</ul>

<h3 id="version-2-5-1-released-on-nov-10-2014">Version 2.5.1 (Released on Nov 10, 2014)</h3>

<p>BUG FIX:</p>

<ul>
<li>Nov 1st: Fix the handling of runtime option <code>temp_dir</code></li>
</ul>

<p>NEW FEATURES:</p>

<ul>
<li>Nov 3rd: Add function <code>in_table</code>.</li>
</ul>

<h3 id="version-2-5-0-release-on-oct-15-2014">Version 2.5.0 (Release on Oct 15, 2014)</h3>

<p>MAJOR NEW FEATURE:</p>

<ul>
<li>Oct 2nd: Implement a mirrored file distribution system so that users can download resources from multiple servers.</li>
<li>Oct 5th: Add <code>site_options.py</code> to allow system administrator to set up site-wide resource directory for variant tools users.</li>
</ul>

<p>NEW FEATURES:</p>

<ul>
<li>Oct 5th: Add md5 signature of database files to <code>.ann</code> files and re-decompress <code>.DB.gz</code> files if the decompressed <code>.DB</code> files are corrupted.</li>
</ul>

<h3 id="version-2-4-1-released-on-august-20-2014">Version 2.4.1 ((Released on August 20, 2014))</h3>

<p>BUG FIX:</p>

<ul>
<li>Fix a python2/3 compatibility bug that prevents vtools from working properly under python2.</li>
</ul>

<h3 id="version-2-4-0-released-on-august-15-2014">Version 2.4.0 (Released on August 15, 2014)</h3>

<p>MAJOR NEW FEATURE:</p>

<ul>
<li>June 10: Variant Simulation Tools is introduced to simulate genetic samples with common and rare variants.</li>
</ul>

<p>NEW FEATURE:</p>

<ul>
<li>Mar 13: Allow the specification of a header using option <code>--header</code> for command <code>vtools phenotype --from_file</code>.</li>
<li>May 15: Progress bar for saving and loading snapshots.</li>
<li>May 15: Use faster compression for local compressed snapshots.</li>
<li>May 20: Add option <code>--translate</code> to command to output protein sequence of genes in specified region.</li>
<li>Jun 5: Allow the execution of many information-based commands such as <code>vtools show annotations</code> without a project.</li>
<li>Jun 26: New features in pipeline allows definitions of multiple steps using one section.</li>
<li>July 5th: Add option <code>--build</code> to command <code>vtools init</code></li>
</ul>

<p>BUGS:</p>

<ul>
<li>June 5: Allow the use of function <code>genotype()</code> when one or more samples do not have genotype.</li>
</ul>

<h3 id="version-2-3-0-released-on-feb-27-2014">Version 2.3.0 (Released on Feb 27, 2014)</h3>

<p>BUGS:</p>

<ul>
<li>Jan 26: Fix a bug with pipeline functor <code>DecompressFiles</code> when the input tar file contains directories.</li>
<li>Feb 22: Fix a bug with wildcard match with table names having special characters such as <code>(</code> or <code>)</code>.</li>
</ul>

<p>NEW FEATURE:</p>

<ul>
<li>Feb 19: Add option <code>--expression</code> to command <code>vtools compare</code>.</li>
<li>Feb 20: Add pipelines <code>filtering</code> to identify recessive and de novo mutations in families with unaffected parents and an affected offspring.</li>
<li>Feb 20: Allow the use of snapshot files (or name of online snapshot name) as <code>--child</code> or <code>--parent</code> in command <code>vtools init</code>.</li>
<li>Feb 22: Keep a copy of original variant tables when merging projects.</li>
</ul>

<h3 id="version-2-2-0-released-on-jan-16-2014">Version 2.2.0 (Released on Jan 16, 2014)</h3>

<p>MAJOR NEW FEATURES:</p>

<ul>
<li>Jan 12, 2014: Add parameters <code>type</code>, <code>show_seq</code>, <code>delimiter</code>, <code>limit</code> and <code>strand</code> to <code>track()</code> function for bam tracks.</li>
<li>Dec 4nd: Add command <code>vtools admin --validate_sex</code> to check sample sex using genotypes on sex chromosomes.</li>
<li>Dec 3nd: Add command <code>vtools_report inbreeding_coefficient</code> to compute individual level inbreeding coefficient, the F-statistic.</li>
<li>Nov 15: Add sub-commands <code>plot_fields</code>, <code>plot_geno_fields</code> and <code>plot_pheno_fields</code> to <code>vtools_report</code>. These commands generate summary plots for specified variant/genotype/sample information.</li>
<li>Dec 2nd: Add special function <code>maf()</code> to command <code>vtools update --from_stat</code> to calculate minor allele frequency.</li>
</ul>

<p>NEW FEATURES:</p>

<ul>
<li>Jan 16, 2014: Update gwas catalogue annotation database.</li>
<li>Jan 16, 2014: Add annotation databases for Database of Genomic Variation (DGV).</li>
<li>Jan 10, 2014: Add population-specific hapmap frequency annotations.</li>
<li>Jan 6th, 2014: Add pipeline <code>import_vcf</code> to import all variant and genotype info from vcf file.</li>
<li>Jan 6th, 2014: Add pipeline <code>anno_utils.annFileFromVcf</code> to make it easier to create an annotation database from vcf files.</li>
<li>Nov 22: Add an annotation database for expanded exome regions of the Illumina Nextera Rapid Capture Expanded Exome Enrichment Kits.</li>
<li>Nov 29: Add KING pipeline to perform global ancestry and kinship analysis.</li>
</ul>

<p>BUG FIXES:</p>

<ul>
<li>Jan 8: Do not abort <code>vtools remove genotypes</code> if genotypes from one of the samples fail to remove.</li>
<li>Jan 4: Fix a multi-processing bug with functor <code>FieldFromDb</code>.</li>
<li>Dec 2: Fix a regression bug for online vcf track.</li>
<li>Nov 14: Fix the use of annotation field if the annotation database has the same name of a variant table.</li>
<li>Nov 18: Fix import PLINK format to allow for arbitrary coding for unknown &ldquo;sex&rdquo; status in <code>fam</code> file.</li>
<li>Nov 20: Properly handle meta analysis input with trailing white space in column names.</li>
<li>Nov 27: Properly handle missing data in external weight for $vtools associate command.</li>
</ul>

<h3 id="version-2-1-0-released-on-nov-6th">Version 2.1.0 (Released on Nov. 6th)</h3>

<p>MAJOR NEW FEATURES:</p>

<ul>
<li>Nov 4: Add option <code>--as</code> to command <code>vtools use</code>, which allows the use of multiple versions of the same annotation databases.</li>
<li>Oct 25: Add SQL function <code>samples()</code> to output samples that contain the variants.</li>
<li>Oct 21: Add SQL function <code>genotype()</code> to output genotypes that contains the variants.</li>
</ul>

<p>NEW FEATURES:</p>

<ul>
<li>Nov 6: Add option <code>all=1</code> to parameter <code>field</code> of <code>track()</code> function and deprecate the third option.</li>
<li>Oct 30: Add pipelines <code>ANNOVAR</code> and <code>snpEff</code> to facilitate the use of <code>ANNOVAR</code> and <code>snpEff</code> to get variant effect estimate.</li>
<li>Oct 25: Allow the use of wildcast characters in the first (<code>filename</code>) parameter of the <code>track</code> function.</li>
<li>Oct 22: Add descriptions to fields added to the project using command <code>vtools update --from_stat</code> and <code>vtools update --set</code>.</li>
<li>Oct 22: Add option <code>--delimiter</code> to command <code>vtools associate</code>.</li>
<li>Oct 17: Handle missing phenotypes more cleverly.</li>
<li>Oct 10: Allow the output read tags and use of tags for read filtering in the track function of BAM tracks.</li>
</ul>

<p>BUGS:</p>

<ul>
<li>Oct 24: Fix a bug that caused misalignment of reads in the output of BAM track when the reads are clipped in CIGAR string.</li>
<li>Oct 11: Fix a bug related to exporting a large number of samples (&gt; 60) in vcf format.</li>
</ul>

<h3 id="version-2-0-1-released-on-oct-7th-2013">Version 2.0.1 (Released on Oct 7th, 2013)</h3>

<p>NEW FEATURES:</p>

<ul>
<li>Oct 5: Update dbNSFP to version 2.1.</li>
<li>Oct 3: align columns of output of command <code>vtools output</code> and <code>vtools show genotypes</code> using variable spaces.</li>
<li>Sep 23: Save ${local_resource} in annoDB to increase portability of projects.</li>
<li>Sep 2: Expand command <code>vtool compare</code> to compare location and genotypes as well.</li>
</ul>

<p>BUGS:</p>

<ul>
<li>Oct 5: Fix the vcf exporter for multiple variants at the same location.</li>
<li>Oct 4: Fix indel track mismatch for vcf track files because position adjustment for indels.</li>
<li>Oct 2: Fix a deadlock bug for flag info when using vcf track.</li>
<li>Sep 18: Fix a bug in <code>vtools admin --merge_samples</code>.</li>
<li>Sep 23: Fix a crash caused by unrecognized chromosome name or out of range positions in function <code>ref_sequence</code>.</li>
<li>Sep 25: Fix a bug related to the use of arbitrary table name in command <code>vtools update</code>.</li>
</ul>

<h3 id="version-2-0-0-released-on-aug-27-2013">Version 2.0.0 (Released on Aug. 27, 2013)</h3>

<p>MAJOR NEW FEATURE:</p>

<ul>
<li>Aug. 15: Add function <code>track</code> to annotate variants using vcf, bigWig and bigBed tracks.</li>
<li>Jun. 28: Add command <code>vtools execute pipeline</code> to execute variant tools pipelines.</li>
<li>July 13: Add option <code>--all</code> to <code>vtools output</code>, <code>vtools select --output</code> and <code>vtools export</code> to remove duplicated lines without sorting output.</li>
</ul>

<p>NEW FEATURE:</p>

<ul>
<li>Aug. 14: Add item <code>fmt</code> to field definitions used to output fields.</li>
<li>Aug. 14: Add functors <code>InfoFormatter</code> and <code>FlagFormatter</code> to output fields in vcf format.</li>
<li>Aug. 13: Add SQL functions <code>ref_sequence</code>.</li>
<li>Aug. 12: Add runtime option check_update.</li>
<li>Jul. 11: Allow .ann files to use preprocessor to process files before importing.</li>
<li>Jul. 11: Add a preprocessor <code>Dos2Unix</code> to convert files with <code>\r</code> as newline character to unix format.</li>
<li>Jul. 10: Change the output of <code>vtools show table TABLE</code> to make it more informative.</li>
<li>Jul. 5: Allow exporting genotypes in format csv.</li>
<li>Jun. 29: Add a feature that allows <code>vtools admin --rename_samples COND NAME1 NAME2</code> to rename samples by replacing the first incidence of <code>NAME1</code> in selected samples to <code>NAME2</code>.</li>
<li>Jun. 29: Add annotation database <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Annotation.GenomicSuperDups?action=edit" target="_blank">genomicSuperDups</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Annotation.GenomicSuperDups?action=edit" target="_blank">?</a> and <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Annotation.PhastCons?action=edit" target="_blank">phast cons elements</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Annotation.PhastCons?action=edit" target="_blank">?</a></li>
<li>May 24: Check duplicate genotypes after samples are imported.</li>
<li>Jun. 25: Allow importing csv files for command <code>vtools phenotype --from_file</code>.</li>
<li>Jun. 25: Add command <code>vtools show phenotypes P1 P2</code>.</li>
<li>Jun. 25: Allow command <code>vtools show samples</code> to show a selected list of samples.</li>
</ul>

<h3 id="version-1-0-6-released-on-may-16th-2013">Version 1.0.6 (Released on May 16th, 2013)</h3>

<p>NEW FEATURE:</p>

<ul>
<li>May 16: Allow the merge of projects with different phenotypes.</li>
<li>May 14: Adding a few fields to vcf.fmt for Illumina data.</li>
<li>May 6: Extend <code>vtools update --from_file</code> to update genotype info from non-original input file, and without genotype.</li>
<li>Apr. 15: Allow the use of wildcard characters in command vtools compare</li>
<li>Apr. 22: Allow the merge of project with different variant tables and fields</li>
</ul>

<p>BUG FIXES:</p>

<ul>
<li>May 16: Allow the use of non-ascii table name in commands <code>vtools export</code> and <code>output</code>/</li>
<li>May 15: Remove statistics missing from vtools phenotype because it is not meaningful.</li>
<li>May 10: Fix runtime variable <code>local_resource</code></li>
<li>May 10: Fix importing data in binary plink format.</li>
</ul>

<h3 id="version-1-0-5-released-on-mar-20-2013">Version 1.0.5 (Released on Mar. 20, 2013)</h3>

<p>NEW FEATURES:</p>

<ul>
<li>March, 2013: Allow the use of arbitrary characters in name of variant tables.</li>
<li>March, 2013: Add command vtools admin &ndash;update_resource to download all relevant resources.</li>
<li>March 18, 2013: Add command vtools_report sequence to output nucleotide sequence at specified chromosome region.</li>
</ul>

<p>BUG FIXES:</p>

<ul>
<li>Feb 26, Fix a bug with command <code>vtools select --samples</code> when there are more than 50 specified samples.</li>
</ul>

<h3 id="version-1-0-4-released-on-feb-22-2013">Version 1.0.4 (Released on Feb 22, 2013)</h3>

<p>NEW FEATURES:</p>

<ul>
<li>Oct 20, 2012: Allow users to list and download online snapshots, which are used for training and documentation purposes.</li>
<li>Oct 20, 2012: Use colors to differentiate debug, info, warning and error message.</li>
<li>Nov 2, 2012: Define runtime option local_resource and use resources from this directory first.</li>
<li>Nov 3, 2012: Add command <code>vtools admin --validate_build</code>.</li>
</ul>

<h3 id="version-1-0-3-released-on-sep-21-2012">Version 1.0.3 (Released on Sep 21, 2012)</h3>

<p>NEW FEATURES:</p>

<ul>
<li>Jul 11, 2012: Add command <code>vtools admin --reset_runtime_option</code> and <code>vtools show runtime_options</code>.</li>
<li>Jul 4, 2012: Reorganize the association test to improve the efficiency of <code>vtools associate</code>.</li>
<li>Jun 16, 2012: Reorganize the import process to substantially improve the efficiency of <code>vtools import</code> when it is used to import files with a large number of samples.</li>
<li>Jun 15, 2012: Add command <code>vtools admin --rename_table</code> and <code>vtools admin --describe_table</code> to change the name and meta information of variant tables.</li>
<li>Jun 12, 2012: Allow a comment string when a variant table is created from command <code>vtools select</code>, <code>vtools exclude</code>, and <code>vtools compare</code>.</li>
<li>May 21, 2012: Add <a href="http://localhost/~iceli/wiki/pmwiki.php?n=Vtools.Update?action=edit" target="_blank">Exact Tests of Hardy-Weinberg Equilibrium</a><a href="http://localhost/~iceli/wiki/pmwiki.php?n=Vtools.Update?action=edit" target="_blank">?</a> and Fisher&rsquo;s exact test for case/ctrl associations to <code>vtools update</code></li>
<li>May 12, 2012: Add command <code>vtools admin</code>.</li>
<li>Jan 27, 2012: Analyze project database automatically (if needed) to improve the performance of queries.</li>
<li>Feb 29, 2012: Completion of the interface of command <code>vtools associate</code></li>
<li>Mar 1, 2012: add output formatting options <code>--header</code>, <code>--na</code>, <code>--delimiter</code> and <code>--limit</code> to <code>vtools phenotype</code></li>
<li>Mar 5, 2012: add parameter <code>--genotypes</code> to command <code>vtools init --parent</code>. (This option was defined but not implemented).</li>
</ul>

<p>BUGS:</p>

<ul>
<li>Jan 27, 2012: Correctly handle missing data when exporting genotype in TPEF format</li>
<li>Feb 9, 2012: Allow the use of single field name (e.g. ccdsid instead of dbNSFP.ccdsid) for parameter <code>--linked_by</code> of command <code>vtools use</code></li>
<li>Feb 17, 2012: Fix a bug with exporting deletion in vcf format, for python2 version of variant tools.</li>
</ul>

<h3 id="version-1-0-2-released-on-jan-24-2012">Version 1.0.2 (released on Jan 24, 2012)</h3>

<p>NEW FEATURES:</p>

<ul>
<li>Jan 4, 2012: add parameter <code>--style</code> to format tped to allow output of genotype in numeric style</li>
<li>Jan 4, 2012: allow command <code>vtools phenotype --from_stat</code> to use conditions that involve variant tables in the project.</li>
<li>Jan 23, 2012: Add a new import format map. A new functor is created to retrieve reference and alternative alleles from dbSNP.</li>
</ul>

<p>BUGS:</p>

<ul>
<li>Jan 11, 2012: Fix a bug that prevents the removal of genotype in command <code>vtools remove variant</code> if the genotype tables are not indexed.</li>
<li>Jan 11-13, 2012: Fix the use of vtools_report under python 3.</li>
<li>Jan 17, 2012: Fix the use of annotation database when exporting in TPED format.</li>
<li>Jan 23, 2012: Remove the use of COLLATE in a query to avoid potential compatibility problems.</li>
<li>Jan 23, 2012: Fix a bug that causes duplicate output in commands <code>vtools output</code> and <code>vtools export</code>.</li>
<li>Jan 23, 2012: Fix a bug for the detection of the existence of indexes.</li>
</ul>

<h3 id="version-1-0-1-released-on-jan-2nd-2012">Version 1.0.1 (released on Jan 2nd, 2012)</h3>

<p>NEW FEATURES:</p>

<ul>
<li>Nov 20, 2011: associate command can now be executed under python 3</li>
<li>Nov 27, 2011: Add entry <code>encoding</code> to <code>.ann</code> and <code>.fmt</code> in order to import files with non-ascii characters</li>
<li>Dec 06, 2011: Give a warning when duplicated records are outputted</li>
<li>Dec 13, 2011: Stop altering user-provided headers for command <code>vtools output</code>.</li>
<li>Dec 13, 2011: Add a parameter <code>--order_by</code> to order output from commands <code>select</code>, <code>exclude</code> and <code>output</code>.</li>
<li>Dec 13, 2011: Add a new format <code>csv.fmt</code> in order to output fields in csv format (with properly quoted fields)</li>
<li>Dec 14, 2011: Add a keyword <code>sort_output_by</code> to file format specification in order to sort variants by specified fields.</li>
<li>Dec 14, 2011: Allow command <code>vtools output</code> to read a header from standard input using option <code>--header -</code>.</li>
<li>Dec 15, 2011: Add parameters <code>anno_type</code> and <code>linked_fields</code> to command <code>vtools use</code> in order to allow the flexibility of how an annotation database it linked.</li>
<li>Dec 15, 2011: Beautify output of <code>vtools show</code> when there are multiple variant tables and annotation databases.</li>
<li>Dec 15, 2011: Adding a new interface for command <code>vtools compare</code> and allow the comparison of more than two tables, using new parameters <code>--intersection</code>, <code>--difference</code>, and <code>--union</code>. The old interface still exits and functional, but has been marked as deprecated.</li>
<li>Dec 28, 2011: Allow the use of wildcard characters <code>?</code> and <code>*</code> in table names in command <code>vtools remove tables</code>.</li>
</ul>

<p>DEPRECATED:</p>

<ul>
<li>Dec 14, 2011: Parameter <code>filename</code> for command <code>vtools export</code> is deprecated. Using a pipe to save output to a file is recommended.</li>
<li>Dec 15, 2011: Parameter <code>--A_and_B</code>, <code>--A_or_B</code>, <code>--A_diff_B</code>, and <code>--B_diff_A</code> are deprecated. Parameters <code>--intersection</code>, <code>--difference</code>, and <code>--union</code> should be used instead.</li>
<li>Dec 29, 2011: Use of name &lsquo;field annotation database&rsquo; instead of &lsquo;attribute annotation database&rsquo; for consistency considerations. Both names are however allowed in .ann file for backward compatibility considerations.</li>
</ul>

<p>BUGS:</p>

<ul>
<li>Dec 15, 2011: Fix a bug when number of genotype is counted when one or more samples do not have genotype field <code>GT</code>.</li>
<li>Dec 15, 2011: Fix a bug where <code>vtools update table --set FIELD=NULL</code> fails because variant tools cannot determine the type of the RHS value. This should be allowed when the LHS is an existing field.</li>
<li>Jan 1, 2012: Fix <code>vtools export --format vcf</code> for the output of genotype when there are two different alternative alleles.</li>
</ul>

<p>REGRESSIONS:</p>

<ul>
<li>Dec 14, 2011: parameter <code>--header</code> of command <code>vtools export</code> is redesigned to work similarly to parameter <code>--header</code> of the <code>vtools output</code> command.</li>
</ul>

<p>INTERNAL IMPROVEMENTS:</p>

<ul>
<li>Nov 17, 2011: Improve efficiency of commands <code>vtools compare</code> by using direct SQL query in the case of <code>--verbosity 0</code></li>
<li>Nov 17, 2011: Improve efficiency of <code>vtools remove variants</code> by creating indexes for genotype tables before removing variants.</li>
<li>Nov 25, 2011: Improve efficiency of <code>vtools select</code> by creating indexes of relevant fields of the annotation database.</li>
<li>Dec 28, 2011: Improve efficiency of commands <code>vtools select</code> for ranged based queries by creating auxiliary binning tables.</li>
</ul>


    
    
        <div class="chevrons">
    <div id="navigation">
<a class="nav nav-prev" href="https://vatlab.github.io/vat-docs/development/" title="Development"> <i class="fa fa-chevron-left"></i><label>Development</label></a>
    <a class="nav nav-next" href="https://vatlab.github.io/vat-docs/development/getinvolved/" title="Get Involved" style="margin-right: 0px;"><label>Get Involved</label><i class="fa fa-chevron-right"></i></a></div>
  </div>

  </section>
</article>

<footer>

<div class="footline">
    

    

  

    <div class="github-link">
      <a href="Development/changelog.md" target="blank"><i class="fa fa-code-fork"></i>
        Improve this page</a>
    </div>
  </div>


	<div>


  
    Create a content/_footer.md file to customize the footer content
  



	</div>
</footer>

<script src="https://vatlab.github.io/vat-docs/js/clipboard.min.js"></script>

<link href="https://vatlab.github.io/vat-docs/css/featherlight.min.css" rel="stylesheet">
<script src="https://vatlab.github.io/vat-docs/js/featherlight.min.js"></script>



<script src="https://vatlab.github.io/vat-docs/theme-flex/script.js"></script>

    

    
    

    
  </body>
</html>